From 1756fc2611e763b2a315486afb601434dfd785a6 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sun, 8 Apr 2012 13:52:02 -0400
Subject: [PATCH 0002/1936] Documentation: asciidoc's latex backend -> dblatex

- Fixes bug ID: 3515891
---
 doc/Makefile | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/doc/Makefile b/doc/Makefile
index 9f22257a..cef9a8d2 100644
--- a/doc/Makefile
+++ b/doc/Makefile
@@ -29,13 +29,7 @@ chunked:
 		 "./zynaddsubfx.xml" \
 
 pdf:
-	asciidoc -b latex --conf-file=myconf.conf zynaddsubfx.txt
-	grep -v figure zynaddsubfx.tex | \
-	sed -e 's~^\\includegraphics~\\par\\noindent\{\} \\includegraphics~' | \
-	sed -e 's~includegraphics\[\]~includegraphics[scale=0.5]~' > tmp.tex
-	mv tmp.tex zynaddsubfx.tex
-	pdflatex zynaddsubfx.tex
-	pdflatex zynaddsubfx.tex
+	a2x -f pdf zynaddsubfx.txt
 
 clean:
 	rm -f *~ *html *pdf *xml *tex *log zynaddsubfx.chunked/images
-- 
2.47.0

