From 008bfc290d88c3fec1f1591dbcb86bc189dd8a1e Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Thu, 19 Apr 2012 16:00:38 -0400
Subject: [PATCH 0007/1936] UI: removing unneeded lock of master mutex

---
 src/UI/MasterUI.fl | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/UI/MasterUI.fl b/src/UI/MasterUI.fl
index 19e312f6..268c4db5 100644
--- a/src/UI/MasterUI.fl
+++ b/src/UI/MasterUI.fl
@@ -183,9 +183,9 @@ if ((maxdbr>MIN_DB-20)){
 int ox=x(); int oy=y(); int lx=w(); int ly=h();
 
 if (!active_r()){
-  pthread_mutex_lock(&master->mutex);
+  pthread_mutex_lock(&master->vumutex);
    int fakedb=master->fakepeakpart[npart];
-  pthread_mutex_unlock(&master->mutex);
+  pthread_mutex_unlock(&master->vumutex);
   fl_rectf(ox,oy,lx,ly,140,140,140);
   if (fakedb>0){
     fakedb=(int)(fakedb/255.0*ly)+4;
@@ -196,9 +196,9 @@ if (!active_r()){
 };
 
 //draw the vu lines
-pthread_mutex_lock(&master->mutex);
+pthread_mutex_lock(&master->vumutex);
  float db=rap2dB(master->vuoutpeakpart[npart]);
-pthread_mutex_unlock(&master->mutex);
+pthread_mutex_unlock(&master->vumutex);
 
 db=(MIN_DB-db)/MIN_DB; 
 if (db<0.0) db=0.0;
-- 
2.47.0

