From 291fa0641b63bd09d66c4abf21ed5d9a3659c301 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Mon, 5 Nov 2012 22:54:38 -0500
Subject: [PATCH 0039/1936] Adding Header Guards to PresetsStore.h

---
 src/Params/PresetsStore.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/Params/PresetsStore.h b/src/Params/PresetsStore.h
index 355ed522..a2f48fd2 100644
--- a/src/Params/PresetsStore.h
+++ b/src/Params/PresetsStore.h
@@ -20,6 +20,9 @@
 
 */
 
+#ifndef PRESETSTORE_H
+#define PRESETSTORE_H
+
 #include <string>
 #include <vector>
 #include "../Misc/XMLwrapper.h"
@@ -64,3 +67,4 @@ class PresetsStore
 };
 
 extern PresetsStore presetsstore;
+#endif
-- 
2.47.0

