From 77d6481128127fd58db755c230aba5ebc1d01ff3 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Wed, 21 Nov 2012 22:28:57 -0500
Subject: [PATCH 0051/1936] UI: reuses VST code for embedding

---
 src/UI/MasterUI.fl | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/UI/MasterUI.fl b/src/UI/MasterUI.fl
index 57ed0166..f625b1e2 100644
--- a/src/UI/MasterUI.fl
+++ b/src/UI/MasterUI.fl
@@ -437,8 +437,8 @@ class MasterUI {open
   } {
     Fl_Window masterwindow {
       label zynaddsubfx
-      callback {\#ifdef VSTAUDIOOUT
-fl_alert("ZynAddSubFX could not be closed this way, because it's a VST plugin. Please use the host aplication to close it.");
+      callback {\#ifdef PLUGINVERSION
+fl_alert("ZynAddSubFX could not be closed this way, because it's a plugin. Please use the host aplication to close it.");
 \#else
 if ((
 \#if USE_NSM 
@@ -1121,8 +1121,8 @@ updatepanel();}
     }
     Fl_Window simplemasterwindow {
       label ZynAddSubFX
-      callback {\#ifdef VSTAUDIOOUT
-fl_alert("ZynAddSubFX could not be closed this way, because it's a VST plugin. Please use the host aplication to close it.");
+      callback {\#ifdef PLUGINVERSION
+fl_alert("ZynAddSubFX could not be closed this way, because it's a plugin. Please use the host aplication to close it.");
 \#else
 if (fl_choice("Exit and leave the unsaved data?","No","Yes",NULL)) {
 	config.save();
-- 
2.47.0

