From a78e09a8713fc5d4c181a84c122bbc97f5d98376 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sat, 20 Apr 2013 21:37:24 -0400
Subject: [PATCH 0074/1936] Merge cleanup

---
 src/Nio/EngineMgr.cpp | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/Nio/EngineMgr.cpp b/src/Nio/EngineMgr.cpp
index a109805a..79373c19 100644
--- a/src/Nio/EngineMgr.cpp
+++ b/src/Nio/EngineMgr.cpp
@@ -43,7 +43,6 @@ EngineMgr::EngineMgr()
 #endif
 #if JACK
     engines.push_back(new JackEngine());
-#endif
     engines.push_back(new JackMultiEngine());
 #endif
 #if PORTAUDIO
-- 
2.47.0

