From ca84e38f90801675ae7dd26f4aa22ba44f5ed29c Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sat, 18 May 2013 11:31:43 -0400
Subject: [PATCH 0084/1936] Cmake: removes assembly for non x86 platforms

---
 src/CMakeLists.txt | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 7ee11f20..68fe7f5a 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -29,6 +29,10 @@ mark_as_advanced(DSSI_LIBRARIES)
 pkg_search_module(LIBLO liblo>=0.26)
 mark_as_advanced(LIBLO_LIBRARIES)
 
+execute_process(COMMAND echo fistpl 0
+                COMMAND as -
+                ERROR_VARIABLE AVOID_ASM)
+
 ######### Settings ###########
 # NOTE: These cache variables should normally not be changed in this
 # file, but either in in CMakeCache.txt before compile, or by passing
@@ -180,11 +184,14 @@ endif()
 include_directories(${ZLIB_INCLUDE_DIRS} ${MXML_INCLUDE_DIRS})
 
 add_definitions(
-	 -DASM_F2I_YES
 	 -g #TODO #todo put in a better location
 	 -Wall
 	 -Wextra
 	 )
+if(NOT AVOID_ASM)
+	message(STATUS "Compiling with x86 assembly support")
+    add_definitions(-DASM_F2I_YES)
+endif()
 
 if (BuildForDebug)
 	set (CMAKE_BUILD_TYPE "Debug")
-- 
2.47.0

