From f7e4a8e146a12b462bda468167f73484b8aebd27 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sat, 18 May 2013 11:51:33 -0400
Subject: [PATCH 0085/1936] Cmake: minor change

---
 src/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 68fe7f5a..b6f543e1 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -189,7 +189,7 @@ add_definitions(
 	 -Wextra
 	 )
 if(NOT AVOID_ASM)
-	message(STATUS "Compiling with x86 assembly support")
+	message(STATUS "Compiling with x86 opcode support")
     add_definitions(-DASM_F2I_YES)
 endif()
 
-- 
2.47.0

