From 7739c0a4742f5082b3b8203518e10d9400acffd7 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Mon, 29 Jul 2013 21:50:12 -0400
Subject: [PATCH 0095/1936] Update Old Submodule URL

---
 .gitmodules | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/.gitmodules b/.gitmodules
index dee185d2..c582dbb7 100644
--- a/.gitmodules
+++ b/.gitmodules
@@ -1,6 +1,6 @@
 [submodule "instruments"]
 	path = instruments
-	url = git://zynaddsubfx.git.sourceforge.net/gitroot/zynaddsubfx/instruments
+	url = git://git.code.sf.net/p/zynaddsubfx/instruments
 [submodule "rtosc"]
 	path = rtosc
 	url = https://github.com/fundamental/rtosc
-- 
2.47.0

