From 8add4978257e23e98cc9eccacfa1661cfb543193 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sun, 11 Aug 2013 17:01:36 -0400
Subject: [PATCH 0117/1936] Min/Max key Callbacks

---
 src/Misc/Part.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/Misc/Part.cpp b/src/Misc/Part.cpp
index 2587b7ab..c44a9650 100644
--- a/src/Misc/Part.cpp
+++ b/src/Misc/Part.cpp
@@ -62,6 +62,11 @@ static Ports partPorts = {
     rToggle(Ppolymode,  "Polyphoney mode"),
     rToggle(Plegatomode, "Legato enable"),
     rParam(Pkeylimit,   "Key limit per part"),
+    {"captureMin:", NULL, NULL, [](const char *, RtData &r)
+        {Part *p = (Part*)r.obj; p->Pminkey = p->lastnote;}},
+    {"captureMax:", NULL, NULL, [](const char *, RtData &r)
+        {Part *p = (Part*)r.obj; p->Pmaxkey = p->lastnote;}},
+
     //{"kit#16::T:F", "::Enables or disables kit item", 0,
     //    [](const char *m, RtData &d) {
     //        Part *p = (Part*)d.obj;
-- 
2.47.0

