From 80c0ef553f5cb98a1477949b6121f17adb88dc98 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Tue, 18 Mar 2014 13:18:15 -0400
Subject: [PATCH 0151/1936] Fix Uninitialized Var

---
 src/Synth/ADnote.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Synth/ADnote.cpp b/src/Synth/ADnote.cpp
index 2d76cea0..2b7b3970 100644
--- a/src/Synth/ADnote.cpp
+++ b/src/Synth/ADnote.cpp
@@ -40,7 +40,7 @@ ADnote::ADnote(ADnoteParameters *pars,
                int portamento_,
                int midinote_,
                bool besilent)
-    :SynthNote(freq, velocity, portamento_, midinote, besilent)
+    :SynthNote(freq, velocity, portamento_, midinote_, besilent)
 {
     tmpwavel = new float [synth->buffersize];
     tmpwaver = new float [synth->buffersize];
-- 
2.47.0

