From 645f6be4bc9e05a310425dc3c3fc632986472d49 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Tue, 18 Mar 2014 15:13:53 -0400
Subject: [PATCH 0154/1936] NTK: Fix Hard to See Checkbox

---
 src/UI/PartUI.fl | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/UI/PartUI.fl b/src/UI/PartUI.fl
index 8e591ac4..556ae13e 100644
--- a/src/UI/PartUI.fl
+++ b/src/UI/PartUI.fl
@@ -1028,6 +1028,11 @@ master=master_;
 ninseff=0;
 
 make_window();
+\#ifdef NTK_GUI
+adsynenabledcheck->selection_color(55);
+subsynenabledcheck->selection_color(55);
+padsynenabledcheck->selection_color(55);
+\#endif
 partgroup->position(this->parent()->x()+2,this->parent()->y()+2);
 partgroup->show();
 end();
-- 
2.47.0

