From d5ef4e73d93d3d7afed0f81cede5122e8af20b92 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Tue, 18 Mar 2014 19:12:39 -0400
Subject: [PATCH 0159/1936] UI: Add part name back to panel window

---
 src/UI/MasterUI.fl | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/UI/MasterUI.fl b/src/UI/MasterUI.fl
index cdda76cd..9f2269b6 100644
--- a/src/UI/MasterUI.fl
+++ b/src/UI/MasterUI.fl
@@ -137,6 +137,8 @@ class Panellistitem {open : {public Fl_Osc_Group}
 };
 bankui->show();}
           xywh {5 27 60 30} box THIN_DOWN_BOX down_box FLAT_BOX labelfont 1 labelsize 10 align 208
+          code0 {o->ext = "Pname";o->oscRegister("Pname");}
+          class PartNameButton
         }
         Fl_Slider partvolume {
           xywh {10 65 30 110} type {Vert Knob} box NO_BOX minimum 127 maximum 0 step 1 value 127
@@ -198,7 +200,7 @@ panellistitem->show();
 end();} {}
   }
   Function {refresh()} {} {
-    code {partname->label("NAME TODO");
+    code {
 
 if ((int)bankui->cbwig->value()!=(npart+1))
    panellistitemgroup->color(fl_rgb_color(160,160,160));
-- 
2.47.0

