From 3b76a035e7c32bf6b2c04f5f23c59abab7798409 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Fri, 18 Apr 2014 18:35:13 -0400
Subject: [PATCH 0168/1936] Minor ifdef for Carla Embedding

---
 src/UI/VirKeyboard.fl | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/UI/VirKeyboard.fl b/src/UI/VirKeyboard.fl
index c07a74ea..6bbaa870 100644
--- a/src/UI/VirKeyboard.fl
+++ b/src/UI/VirKeyboard.fl
@@ -76,10 +76,17 @@ rndvelocity=0;} {}
     code {int ox=x(),oy=y(),lx=w(),ly=h()-1,i;
 
 \#ifdef NTK_GUI
+\#ifdef CARLA_VERSION_STRING
+ Fl_Image *white_up = Fl_Shared_Image::get( gUiPixmapPath + "white_key.png" );
+ Fl_Image *white_down = Fl_Shared_Image::get( gUiPixmapPath + "white_key_pressed.png" );
+ Fl_Image *black_up = Fl_Shared_Image::get( gUiPixmapPath + "black_key.png" );
+ Fl_Image *black_down = Fl_Shared_Image::get( gUiPixmapPath + "black_key_pressed.png" );
+\#else
  Fl_Image *white_up = Fl_Shared_Image::get( PIXMAP_PATH "white_key.png" );
  Fl_Image *white_down = Fl_Shared_Image::get( PIXMAP_PATH "white_key_pressed.png" );
  Fl_Image *black_up = Fl_Shared_Image::get( PIXMAP_PATH "black_key.png" );
  Fl_Image *black_down = Fl_Shared_Image::get( PIXMAP_PATH "black_key_pressed.png" );
+ \#endif
 
 
 //On error fetch everything from source directory
-- 
2.47.0

