From 8a900ef2a4e804d7df3e220113bbb3ff2582c206 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Tue, 6 May 2014 16:27:59 -0400
Subject: [PATCH 0175/1936] Add examples to list of installed files

---
 CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e7963b0b..169877a3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -17,3 +17,5 @@ install(FILES zynaddsubfx.svg
     DESTINATION share/pixmaps)
 install(DIRECTORY instruments/banks
     DESTINATION share/zynaddsubfx)
+install(DIRECTORY instruments/examples
+    DESTINATION share/zynaddsubfx)
-- 
2.47.0

