From 47b9935aeea27aac135d7083913aeb96d79530c5 Mon Sep 17 00:00:00 2001
From: Robin Gareus <robin@gareus.org>
Date: Mon, 19 May 2014 06:34:13 +0200
Subject: [PATCH 0179/1936] CFG: add default paths for OSX bundle

---
 src/Misc/Config.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/Misc/Config.cpp b/src/Misc/Config.cpp
index ea83d458..8c5b7b20 100644
--- a/src/Misc/Config.cpp
+++ b/src/Misc/Config.cpp
@@ -87,14 +87,22 @@ void Config::init()
         cfg.bankRootDirList[1] = "./";
         cfg.bankRootDirList[2] = "/usr/share/zynaddsubfx/banks";
         cfg.bankRootDirList[3] = "/usr/local/share/zynaddsubfx/banks";
+#ifdef __APPLE__
+        cfg.bankRootDirList[4] = "../Resources/banks";
+#else
         cfg.bankRootDirList[4] = "../banks";
+#endif
         cfg.bankRootDirList[5] = "banks";
     }
 
     if(cfg.presetsDirList[0].empty()) {
         //presets
         cfg.presetsDirList[0] = "./";
+#ifdef __APPLE__
+        cfg.presetsDirList[1] = "../Resources/presets";
+#else
         cfg.presetsDirList[1] = "../presets";
+#endif
         cfg.presetsDirList[2] = "presets";
         cfg.presetsDirList[3] = "/usr/share/zynaddsubfx/presets";
         cfg.presetsDirList[4] = "/usr/local/share/zynaddsubfx/presets";
-- 
2.47.0

