From 29dad78c13fff1f21caca0c8476a60d9e2189d6a Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sat, 28 Jun 2014 22:05:15 -0400
Subject: [PATCH 0200/1936] Update XML Version Number

---
 src/Misc/XMLwrapper.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Misc/XMLwrapper.cpp b/src/Misc/XMLwrapper.cpp
index 0fd40f35..6148d68d 100644
--- a/src/Misc/XMLwrapper.cpp
+++ b/src/Misc/XMLwrapper.cpp
@@ -98,7 +98,7 @@ XMLwrapper::XMLwrapper()
 {
     version.Major    = 2;
     version.Minor    = 4;
-    version.Revision = 3;
+    version.Revision = 4;
 
     minimal = true;
 
-- 
2.47.0

