From 97dfb820581b8408cfb34ddae73d6722692d445c Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sat, 28 Jun 2014 23:30:39 -0400
Subject: [PATCH 0206/1936] Fix Unportable Envelope Code

---
 src/Synth/Envelope.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Synth/Envelope.cpp b/src/Synth/Envelope.cpp
index fc366e8d..41bb8047 100644
--- a/src/Synth/Envelope.cpp
+++ b/src/Synth/Envelope.cpp
@@ -160,7 +160,7 @@ float Envelope::envout()
 }
 
 inline float Envelope::env_dB2rap(float db) {
-    return (exp10f((db) / 20.0f) - 0.01)/.99f;
+    return (powf(10.0f, db / 20.0f) - 0.01)/.99f;
 }
 
 inline float Envelope::env_rap2dB(float rap) {
-- 
2.47.0

