From 68354f36682c654f1e9f020bee7eda3d33cea4e7 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sat, 12 Jul 2014 16:02:32 -0400
Subject: [PATCH 0208/1936] Fix Repeated #define on FreeBSD

---
 src/Misc/Util.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/Misc/Util.h b/src/Misc/Util.h
index c2574f4b..b105e4c2 100644
--- a/src/Misc/Util.h
+++ b/src/Misc/Util.h
@@ -123,7 +123,9 @@ inline void sprng(prng_t p)
 /*
  * The random generator (0.0f..1.0f)
  */
-# define INT32_MAX      (2147483647)
+#ifndef INT32_MAX
+#define INT32_MAX      (2147483647)
+#endif
 #define RND (prng() / (INT32_MAX * 1.0f))
 
 //Linear Interpolation
-- 
2.47.0

