From a4ae50bd7c910377935362d31ae8507bd8460832 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Wed, 16 Jul 2014 13:09:06 -0400
Subject: [PATCH 0209/1936] Fix /part*/Pvolume Routeing

---
 src/Misc/Part.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/Misc/Part.cpp b/src/Misc/Part.cpp
index c2513c5b..afa2941e 100644
--- a/src/Misc/Part.cpp
+++ b/src/Misc/Part.cpp
@@ -49,7 +49,11 @@ static Ports partPorts = {
     rRecursp(partefx, 3, "Part Effect"),
     rRecur(ctl,       "Controller"),
     rToggle(Penabled, "Part enable"),
+#undef rChangeCb
+#define rChangeCb obj->setPvolume(obj->Pvolume);
     rParam(Pvolume, "Part Volume"),
+#undef rChangeCb
+#define rChangeCb
     rParam(Pminkey, "Min Used Key"),
     rParam(Pmaxkey, "Max Used Key"),
     rParam(Pkeyshift, "Part keyshift"),
-- 
2.47.0

