From 8d4b39d5b95c5ef16ac42a27bfdbc8b1fe42f5e8 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Wed, 16 Jul 2014 16:08:19 -0400
Subject: [PATCH 0210/1936] Update rtosc submodule

---
 rtosc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rtosc b/rtosc
index e40ae9fb..4d5ea21c 160000
--- a/rtosc
+++ b/rtosc
@@ -1 +1 @@
-Subproject commit e40ae9fb3730b8fa0d759a40cef6968a48e70696
+Subproject commit 4d5ea21cabe5242905f3fad165686f03f499e1ff
-- 
2.47.0

