From e1e09185c60e1a0319a5337561f0ed3320b433a2 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Wed, 16 Jul 2014 16:10:14 -0400
Subject: [PATCH 0211/1936] Cmake: Fix Mixed PA 1/2 Include Path Search

---
 src/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 2146ca6b..198ea5b5 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -165,7 +165,7 @@ if(OssEnable)
 endif(OssEnable)
 
 if(PaEnable)
-	include_directories(${PORTAUDIO_INCLUDE_DIR})
+	include_directories(${PORTAUDIO_INCLUDEDIR} /usr/lib/blam/bax)
 	add_definitions(-DPORTAUDIO=1)
 	list(APPEND AUDIO_LIBRARIES ${PORTAUDIO_LIBRARIES})
 	list(APPEND AUDIO_LIBRARY_DIRS ${PORTAUDIO_LIBRARY_DIRS})
-- 
2.47.0

