From 6ecda2b036071367865180d496e2277a798f6eaf Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Mon, 1 Sep 2014 12:02:19 -0400
Subject: [PATCH 0225/1936] Update rtosc submodule

---
 rtosc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rtosc b/rtosc
index a63e2b97..44ba4f94 160000
--- a/rtosc
+++ b/rtosc
@@ -1 +1 @@
-Subproject commit a63e2b97b8276658f1709769592eb2896eabfe8b
+Subproject commit 44ba4f9455c925958f6de38171c0d9b317e58c19
-- 
2.47.0

