From 844b9f5ad90d9dfaa63784607434cfbbb4b14714 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Mon, 1 Sep 2014 12:45:12 -0400
Subject: [PATCH 0227/1936] Fix BSD Build [Missing std::to_string()]

---
 src/Misc/MiddleWare.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Misc/MiddleWare.cpp b/src/Misc/MiddleWare.cpp
index c22a7a8b..9f636269 100644
--- a/src/Misc/MiddleWare.cpp
+++ b/src/Misc/MiddleWare.cpp
@@ -323,7 +323,7 @@ class MiddleWareImpl
     //! returns file name to where UDP port is saved
     std::string get_tmp_nam() const
     {
-         return "/tmp/zynaddsubfx_" + std::to_string(getpid());
+         return "/tmp/zynaddsubfx_" + to_s(getpid());
     }
 public:
     MiddleWareImpl(void)
-- 
2.47.0

