From 16d9dce80cc793472e5e11ce1894ad6e2b5caebb Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Tue, 2 Sep 2014 14:35:55 -0400
Subject: [PATCH 0229/1936] UI: Fix Segfault with NTK

---
 src/UI/Connection.cpp | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/UI/Connection.cpp b/src/UI/Connection.cpp
index 1b1ed4ec..8ef66219 100644
--- a/src/UI/Connection.cpp
+++ b/src/UI/Connection.cpp
@@ -34,8 +34,10 @@ set_module_parameters ( Fl_Widget *o )
     o->color( FL_BLACK );
     o->image( module_backdrop );
     o->labeltype( FL_SHADOW_LABEL );
-    o->parent()->labeltype(FL_NO_LABEL);
-    o->parent()->box(FL_NO_BOX);
+    if(o->parent()) {
+        o->parent()->labeltype(FL_NO_LABEL);
+        o->parent()->box(FL_NO_BOX);
+    }
 #else
     o->box( FL_PLASTIC_UP_BOX );
     o->color( FL_CYAN );
-- 
2.47.0

