From c704c4f235b6bfc9cd971a5c5592c8370d66ffb9 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Fri, 17 Oct 2014 11:34:51 -0400
Subject: [PATCH 0249/1936] Fix Broken Bankselect Fl_Choice Widget

---
 src/UI/BankView.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/UI/BankView.cpp b/src/UI/BankView.cpp
index 5bd8c3e9..c568cb16 100644
--- a/src/UI/BankView.cpp
+++ b/src/UI/BankView.cpp
@@ -27,6 +27,7 @@ void BankList::OSC_raw(const char *msg)
     const int   pos  = rtosc_argument(msg, 0).i;
     const char *path = rtosc_argument(msg, 1).s;
 
+    value(0);
     if(pos == 0)
         this->clear();
 
-- 
2.47.0

