From f6d8b8faad51b121b362ee6135bc1b48ea21604e Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sun, 19 Oct 2014 14:50:48 -0400
Subject: [PATCH 0255/1936] Fix FFT Mutex Cleanup

---
 src/DSP/FFTwrapper.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/DSP/FFTwrapper.cpp b/src/DSP/FFTwrapper.cpp
index 84f74aa3..cdee611f 100644
--- a/src/DSP/FFTwrapper.cpp
+++ b/src/DSP/FFTwrapper.cpp
@@ -98,4 +98,5 @@ void FFT_cleanup()
     fftw_cleanup();
     pthread_mutex_destroy(mutex);
     delete mutex;
+    mutex = NULL;
 }
-- 
2.47.0

