From 98930de7f2d31a405055d306dafc5ee96efe3bb6 Mon Sep 17 00:00:00 2001
From: Hans Petter Selasky <hps@selasky.org>
Date: Fri, 24 Oct 2014 11:06:23 +0200
Subject: [PATCH 0263/1936] Remove useless assignment.

Signed-off-by: Hans Petter Selasky <hps@selasky.org>
---
 src/UI/PartUI.fl | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/UI/PartUI.fl b/src/UI/PartUI.fl
index e69fec1b..cc2a37fd 100644
--- a/src/UI/PartUI.fl
+++ b/src/UI/PartUI.fl
@@ -996,7 +996,6 @@ assert(!loc_.empty());
 bankui=bankui_;
 part_path = part_path_;
 base = part_path;
-osc  = osc;
 npart=npart_;
 ninseff=0;
 osc=osc_;
-- 
2.47.0

