From bf3140e92de0df4a1db61d3da5a0ee87e64349a6 Mon Sep 17 00:00:00 2001
From: Hans Petter Selasky <hps@selasky.org>
Date: Fri, 24 Oct 2014 11:21:50 +0200
Subject: [PATCH 0265/1936] Add missing function argument reference.

Signed-off-by: Hans Petter Selasky <hps@selasky.org>
---
 src/Params/PADnoteParameters.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Params/PADnoteParameters.cpp b/src/Params/PADnoteParameters.cpp
index 11a5f1de..0301caac 100644
--- a/src/Params/PADnoteParameters.cpp
+++ b/src/Params/PADnoteParameters.cpp
@@ -137,7 +137,7 @@ static rtosc::Ports localPorts =
             d.reply(d.loc, "i", realbw);
             delete[] tmp;}},
     {"sample#64:ifb", rDoc("Nothing to see here"), 0,
-        [](const char *m, rtosc::RtData d)
+        [](const char *m, rtosc::RtData &d)
         {
             PADnoteParameters *p = (PADnoteParameters*)d.obj;
             const char *mm = m;
-- 
2.47.0

