From 12bbb35c2f866f0b2e2eb8f9e4ae82c2f61428c1 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sat, 25 Oct 2014 21:57:23 -0400
Subject: [PATCH 0269/1936] Fix OscilGen Segfault

---
 src/UI/OscilGenUI.fl | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/UI/OscilGenUI.fl b/src/UI/OscilGenUI.fl
index a660b4a9..1b42f11c 100644
--- a/src/UI/OscilGenUI.fl
+++ b/src/UI/OscilGenUI.fl
@@ -355,7 +355,6 @@ return;
 if(o->value()==0||o->value()==127) basefuncmodulation->deactivate();
 else basefuncmodulation->activate();}
             xywh {375 290 90 15} down_box BORDER_BOX labelsize 10 align 5 when 1 textsize 11
-            code0 {o->init("Pcurrentbasefunc");}
             class Fl_Osc_Choice
           } {
             MenuItem {} {
@@ -937,6 +936,7 @@ basefuncdisplaygroup = NULL;
 bfslider = NULL;
 bftype = NULL;
 make_window();
+bftype->init("Pcurrentbasefunc");
 initialized = true;
 
 refresh();
-- 
2.47.0

