From 65cf27fcc10594d85029537581588f0cb06e921b Mon Sep 17 00:00:00 2001
From: Hans Petter Selasky <hps@selasky.org>
Date: Tue, 28 Oct 2014 08:52:01 +0100
Subject: [PATCH 0273/1936] Add missing include file for definition of error
 codes.

Signed-off-by: Hans Petter Selasky <hps@selasky.org>
---
 src/Nio/OssEngine.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/Nio/OssEngine.cpp b/src/Nio/OssEngine.cpp
index 6617ba96..1964e4c8 100644
--- a/src/Nio/OssEngine.cpp
+++ b/src/Nio/OssEngine.cpp
@@ -28,6 +28,7 @@
 #include <stdlib.h>
 #include <stdio.h>
 #include <fcntl.h>
+#include <errno.h>
 #include <sys/soundcard.h>
 #include <sys/stat.h>
 #include <sys/ioctl.h>
-- 
2.47.0

