From ec21f8d84a7eef29471e2a6b9c3d493e1a8406c2 Mon Sep 17 00:00:00 2001
From: Hans Petter Selasky <hps@selasky.org>
Date: Sat, 1 Nov 2014 18:00:18 +0100
Subject: [PATCH 0283/1936] Compile fix. Variable name shadows std::micro.

Signed-off-by: Hans Petter Selasky <hps@selasky.org>
---
 src/Tests/InstrumentStats.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/Tests/InstrumentStats.cpp b/src/Tests/InstrumentStats.cpp
index ac1cd964..87d4b2c7 100644
--- a/src/Tests/InstrumentStats.cpp
+++ b/src/Tests/InstrumentStats.cpp
@@ -18,7 +18,7 @@ MiddleWare *middleware;
 
 float *outL, *outR;
 Allocator  alloc;
-Microtonal micro;
+Microtonal microtonal;
 FFTwrapper fft(1024);
 Part *p;
 
@@ -55,7 +55,7 @@ void setup() {
     denormalkillbuf = new float[synth->buffersize];
     for(int i = 0; i < synth->buffersize; ++i)
         denormalkillbuf[i] = 0;
-    p = new Part(alloc, &micro, &fft);
+    p = new Part(alloc, &microtonal, &fft);
 }
 
 void load(string s)
-- 
2.47.0

