From 3b2bf2b32163a13604052ca1c4be67f4a6f6c60e Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Wed, 5 Nov 2014 11:19:20 -0500
Subject: [PATCH 0286/1936] Add Panic OSC Callback

---
 src/Misc/Master.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/Misc/Master.cpp b/src/Misc/Master.cpp
index 3ebae05e..7e63a6d6 100644
--- a/src/Misc/Master.cpp
+++ b/src/Misc/Master.cpp
@@ -140,6 +140,11 @@ static Ports localports = {
         [](const char *m,RtData &d){
             Master *M =  (Master*)d.obj;
             M->setController(rtosc_argument(m,0).i,rtosc_argument(m,1).i,rtosc_argument(m,2).i);}},
+    {"Panic:", rDoc("Stop All Sound"), 0,
+        [](const char *, RtData &d) {
+            Master &M =  *(Master*)d.obj;
+            M.ShutUp();
+        }},
     {"freeze_state:", rDoc("Internal Read-only Mode"), 0,
         [](const char *,RtData &d) {
             Master *M =  (Master*)d.obj;
-- 
2.47.0

