From ed0d3d01e15f02b89d8f2c99ab6e49fea929ddaf Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Wed, 17 Dec 2014 12:09:34 -0500
Subject: [PATCH 0296/1936] rtosc: Fixing Some Dead Ports

:c -> :i
---
 src/Misc/Util.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Misc/Util.h b/src/Misc/Util.h
index c0f08961..7ca67222 100644
--- a/src/Misc/Util.h
+++ b/src/Misc/Util.h
@@ -149,7 +149,7 @@ float cinterpolate(const float *data, size_t len, float pos);
 const char *message_snip(const char *m);
 
 #define rParamZyn(name, ...) \
-  {STRINGIFY(name) "::i",  rProp(parameter) rMap(min, 0) rMap(max, 127) DOC(__VA_ARGS__), NULL, rParamCb(name)}
+  {STRINGIFY(name) "::i",  rProp(parameter) rMap(min, 0) rMap(max, 127) DOC(__VA_ARGS__), NULL, rParamICb(name)}
 
 ///floating point parameter - with lookup code
 #define PARAMF(type, var, name, scale, _min, _max, desc) \
-- 
2.47.0

