From c9d40150022116a1ab9d11d60ca07ceda7f7b67e Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sun, 21 Dec 2014 11:03:45 -0500
Subject: [PATCH 0300/1936] Fix Resonance Application

---
 src/Synth/Resonance.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Synth/Resonance.cpp b/src/Synth/Resonance.cpp
index af09413c..2f0cfa8f 100644
--- a/src/Synth/Resonance.cpp
+++ b/src/Synth/Resonance.cpp
@@ -100,7 +100,7 @@ void Resonance::applyres(int n, fft_t *fftdata, float freq) const
 
     for(int i = 1; i < n; ++i) {
         //compute where the n-th hamonics fits to the graph
-        const float x  = limit((logf(freq) - l1) / l2, 0.0f, INFINITY) * N_RES_POINTS;
+        const float x  = limit((logf(freq*i) - l1) / l2, 0.0f, INFINITY) * N_RES_POINTS;
         const float dx = x - floor(x);
         const int kx1  = limit<int>(floor(x), 0, N_RES_POINTS - 1);
         const int kx2  = limit<int>(kx1 + 1,  0, N_RES_POINTS - 1);
-- 
2.47.0

