From 79230e42f40a6327856f362708ddc519672b8f97 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Wed, 18 Feb 2015 19:19:52 -0500
Subject: [PATCH 0322/1936] UI: Fix Virtual Keyboard OSC

---
 src/UI/VirKeyboard.fl | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/UI/VirKeyboard.fl b/src/UI/VirKeyboard.fl
index aaeff8bc..07f4dfb4 100644
--- a/src/UI/VirKeyboard.fl
+++ b/src/UI/VirKeyboard.fl
@@ -370,7 +370,7 @@ virkeyboardwindow->hide();}
         label Cval
         callback {int ctl=midictl;
 
-osc->write("/setController", "ccc", virkeys->midich,ctl,(int) o->value());
+osc->write("/setController", "iii", virkeys->midich,ctl,(int) o->value());
 virkeys->take_focus();}
         tooltip {Controller value} xywh {605 10 15 115} type {Vert Fill} box ENGRAVED_BOX selection_color 229 labelsize 8 align 5 minimum 127 maximum 0 step 1 value 64 textsize 7
       }
@@ -450,7 +450,7 @@ virkeys->take_focus();}
       }
       Fl_Roller pitchwheelroller {
         label Pwh
-        callback {osc->write("/setController", "ccc", virkeys->midich,C_pitchwheel,-(int) o->value());
+        callback {osc->write("/setController", "iii", virkeys->midich,C_pitchwheel,-(int) o->value());
 virkeys->take_focus();}
         tooltip {Pitch Wheel} xywh {625 10 20 95} labelsize 8 align 1 when 3 minimum -8192 maximum 8192 step 64
       }
-- 
2.47.0

