From 2c4e122b82b00dcba7b21a31b41bdb67f206fbe2 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Wed, 18 Feb 2015 22:21:16 -0500
Subject: [PATCH 0326/1936] UI: Fix Insertion Effect Channel Selection

---
 src/UI/MasterUI.fl | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/UI/MasterUI.fl b/src/UI/MasterUI.fl
index c4ef73e3..e518eb89 100644
--- a/src/UI/MasterUI.fl
+++ b/src/UI/MasterUI.fl
@@ -676,7 +676,7 @@ if ((int) o->value()==1){
               xywh {100 188 80 22} down_box BORDER_BOX labelfont 1 labelsize 10 align 5 textsize 10
               code0 {o->add("Master Out");o->add("Off");}
               code1 {char tmp[50]; for (int i=0;i<NUM_MIDI_PARTS;i++) {sprintf(tmp,"Part %2d",i+1);o->add(tmp);};}
-              code3 {o->init("Pinsparts"+to_s(ninseff),2);}
+              code3 {o->init("Pinsparts"+to_s(ninseff),-2);}
               class Fl_Osc_Choice
             } {}
             Fl_Button {} {
-- 
2.47.0

