From ef5a57606fd5671601f57f0a1e7f8ab5b76c1317 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sun, 22 Feb 2015 14:15:07 -0500
Subject: [PATCH 0349/1936] UI: Fix Bad Return in VuMasterMeter::handle

---
 src/UI/VuMasterMeter.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/UI/VuMasterMeter.h b/src/UI/VuMasterMeter.h
index f783ccdb..bb32b05b 100644
--- a/src/UI/VuMasterMeter.h
+++ b/src/UI/VuMasterMeter.h
@@ -29,10 +29,10 @@ class VuMasterMeter: public VuMeter
                     break;
                 case FL_PUSH:
                     osc->requestValue("/reset-vu");
-                    break;
+                    return 1;
             }
 
-            return 1;
+            return 0;
         }
 
         static void tick(void *v)
-- 
2.47.0

