From 78222b8a697f626aed6f075d8c2913cc8ea02245 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Mon, 23 Feb 2015 12:23:26 -0500
Subject: [PATCH 0352/1936] UI: Fix Pause || in MasterUI

---
 src/UI/MasterUI.fl | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/UI/MasterUI.fl b/src/UI/MasterUI.fl
index df2a98b5..f2a8258c 100644
--- a/src/UI/MasterUI.fl
+++ b/src/UI/MasterUI.fl
@@ -727,7 +727,7 @@ mastermenu->redraw();}
             class Fl_Osc_Button
           }
           Fl_Button pausebutton {
-            label {@||}
+            label {| |}
             callback {o->deactivate();
 o->oscWrite("HDDRecorder/pause");
 recordbutton->activate();
-- 
2.47.0

