From c8ba32cc755f305aabc155b52b3ca3a80d6f4a0e Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Mon, 23 Feb 2015 12:48:01 -0500
Subject: [PATCH 0354/1936] Update rtosc

---
 rtosc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rtosc b/rtosc
index baaeeb5e..6df6d35e 160000
--- a/rtosc
+++ b/rtosc
@@ -1 +1 @@
-Subproject commit baaeeb5e06c3b31d1f6a5fd44cdb49354a399cdc
+Subproject commit 6df6d35efbc2001ea310a4a0e918c622501479b2
-- 
2.47.0

