From 87ce086771360fabaa12dc073f7ded89cff3bbec Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Mon, 23 Feb 2015 18:16:16 -0500
Subject: [PATCH 0357/1936] MasterUI: Conditionally Compile Window Icon

---
 src/UI/MasterUI.fl | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/UI/MasterUI.fl b/src/UI/MasterUI.fl
index f2a8258c..847b0e72 100644
--- a/src/UI/MasterUI.fl
+++ b/src/UI/MasterUI.fl
@@ -1485,9 +1485,11 @@ configui=new ConfigUI();
 make_window();
 fl_open_display();
 
+#ifndef __APPLE__
 Pixmap p = XCreateBitmapFromData(fl_display, DefaultRootWindow(fl_display),
                                  (char*)zynaddsubfx_bits, zynaddsubfx_width, zynaddsubfx_height);
 masterwindow->icon((char *)p);
+#endif
 presetsui=new PresetsUI();
 setfilelabel(NULL);
 swapefftype=0;
-- 
2.47.0

