From ce4ef7ba3ebc7f2bc715fd7fdbefaa829c258487 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Mon, 23 Feb 2015 18:34:04 -0500
Subject: [PATCH 0358/1936] Alienwah: Fix Include Order

---
 src/Effects/Alienwah.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Effects/Alienwah.h b/src/Effects/Alienwah.h
index f0022f69..7f8358df 100644
--- a/src/Effects/Alienwah.h
+++ b/src/Effects/Alienwah.h
@@ -23,9 +23,9 @@
 #ifndef ALIENWAH_H
 #define ALIENWAH_H
 
-#include <complex>
 #include "Effect.h"
 #include "EffectLFO.h"
+#include <complex>
 
 using namespace std;
 
-- 
2.47.0

