From 59b10bf2b065faba89ab6cf3804321dd1fcb562d Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Mon, 23 Feb 2015 19:40:50 -0500
Subject: [PATCH 0361/1936] CMake: Install README.txt -> README.adoc

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 81890093..c8432178 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -26,7 +26,7 @@ include(CTestConfig.cmake)
 #Currently the only directory that uses cmake
 add_subdirectory(src)
 
-install(FILES AUTHORS.txt COPYING FAQ.txt HISTORY.txt README.txt
+install(FILES AUTHORS.txt COPYING FAQ.txt HISTORY.txt README.adoc
 	DESTINATION share/doc/zynaddsubfx
 	)
 install(FILES zynaddsubfx-jack.desktop zynaddsubfx-alsa.desktop
-- 
2.47.0

