From d5db633c29c69a64695dccd9b617149903b05b86 Mon Sep 17 00:00:00 2001
From: Hans Petter Selasky <hps@selasky.org>
Date: Tue, 24 Feb 2015 22:43:07 +0100
Subject: [PATCH 0362/1936] Fix compilation under FreeBSD when using clang.

Signed-off-by: Hans Petter Selasky <hps@selasky.org>
---
 src/globals.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/globals.h b/src/globals.h
index fb56a0a6..1d686574 100644
--- a/src/globals.h
+++ b/src/globals.h
@@ -64,7 +64,7 @@ class  Controller;
 class  Master;
 class  Part;
 
-#ifdef __APPLE__
+#if defined(__APPLE__) || defined(__FreeBSD__)
 #include <complex>
 #else
 namespace std {
-- 
2.47.0

