From e12962c852faa495d2f7011b8f8d819c6c613553 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Wed, 25 Feb 2015 17:54:42 -0500
Subject: [PATCH 0370/1936] UI: Make Panel Window Update On New XMZ Load

---
 src/UI/MasterUI.fl | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/src/UI/MasterUI.fl b/src/UI/MasterUI.fl
index 756bf3eb..3a8d1ae0 100644
--- a/src/UI/MasterUI.fl
+++ b/src/UI/MasterUI.fl
@@ -193,7 +193,14 @@ panellistitem->show();
 end();} {}
   }
   Function {refresh()} {} {
-    code {if ((int)bankui->cbwig->value()!=(npart+1))
+    code {
+    partenabled->update();
+    partrcv->update();
+    partpanning->update();
+    partvolume->update();
+    partname->update();
+    
+    if ((int)bankui->cbwig->value()!=(npart+1))
    panellistitemgroup->color(fl_rgb_color(160,160,160));
 else 
    panellistitemgroup->color(fl_rgb_color(50,190,240));
-- 
2.47.0

