From 8114add08d1d337f75dab385a8043da66e07c28b Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Wed, 25 Feb 2015 18:24:49 -0500
Subject: [PATCH 0373/1936] UI: Remove Redundant MIDI Learn Window

---
 src/UI/Connection.cpp | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/src/UI/Connection.cpp b/src/UI/Connection.cpp
index 9947a653..24b5f96e 100644
--- a/src/UI/Connection.cpp
+++ b/src/UI/Connection.cpp
@@ -89,12 +89,12 @@ ui_handle_t GUI::createUi(Fl_Osc_Interface *osc, void *exit)
     Fl::foreground(255, 255, 255);
 #endif
 
-    Fl_Window *midi_win = new Fl_Double_Window(400, 400, "Midi connections");
-    Fl_Osc_Tree *tree   = new Fl_Osc_Tree(0,0,400,400);
-    midi_win->resizable(tree);
-    tree->root_ports    = &Master::ports;
-    tree->osc           = osc;
-    midi_win->show();
+    //Fl_Window *midi_win = new Fl_Double_Window(400, 400, "Midi connections");
+    //Fl_Osc_Tree *tree   = new Fl_Osc_Tree(0,0,400,400);
+    //midi_win->resizable(tree);
+    //tree->root_ports    = &Master::ports;
+    //tree->osc           = osc;
+    //midi_win->show();
 
     Fl::add_handler(undo_redo_handler);
     return (void*) (ui = new MasterUI((int*)exit, osc));
-- 
2.47.0

