From b165f7089dc40ebc2c56180386865c26bb5760bb Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Fri, 27 Feb 2015 18:57:34 -0500
Subject: [PATCH 0376/1936] MiddleWare: Fix Bad Oscilgen Pointer

---
 src/Misc/MiddleWare.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/Misc/MiddleWare.cpp b/src/Misc/MiddleWare.cpp
index e4084655..01c17937 100644
--- a/src/Misc/MiddleWare.cpp
+++ b/src/Misc/MiddleWare.cpp
@@ -342,7 +342,7 @@ struct NonRtObjStore
     void extractPart(Part *part, int i)
     {
         for(int j=0; j < NUM_KIT_ITEMS; ++j) {
-            auto &obj = part->kit[i];
+            auto &obj = part->kit[j];
             extractAD(obj.adpars, i, j);
             extractPAD(obj.padpars, i, j);
         }
@@ -937,6 +937,8 @@ bool MiddleWareImpl::handleOscil(string path, const char *msg, void *v)
     memset(buffer, 0, sizeof(buffer));
     DummyDataObj d(buffer, 1024, v, cb, ui, osc);
     strcpy(buffer, path.c_str());
+    if(!v)
+        return true;
 
     for(auto &p:OscilGen::ports.ports) {
         if(strstr(p.name,msg) && strstr(p.metadata, "realtime") &&
-- 
2.47.0

