From 8b6ed110e6f9af96deabdb6f174d488cbd089c12 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Fri, 27 Feb 2015 20:20:12 -0500
Subject: [PATCH 0377/1936] UI: Adjust OscilSpectra Color Palette

---
 src/UI/Fl_OscilSpectrum.h | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/UI/Fl_OscilSpectrum.h b/src/UI/Fl_OscilSpectrum.h
index 268b2230..fa53beff 100644
--- a/src/UI/Fl_OscilSpectrum.h
+++ b/src/UI/Fl_OscilSpectrum.h
@@ -70,10 +70,12 @@ class Fl_OscilSpectrum : public Fl_Box, Fl_Osc_Widget
             int n=lx/GX-1;
             if (n>synth->oscilsize/2)
                 n=synth->oscilsize/2;
+                
+            fl_rectf(ox,oy,lx,ly,0,0,0);
 
             //draw
             if(this->active_r()) 
-                fl_color(this->parent()->labelcolor());
+                fl_color(0,0,255);
             else 
                 fl_color(this->parent()->color());
             fl_line_style(FL_DOT);
@@ -93,7 +95,7 @@ class Fl_OscilSpectrum : public Fl_Box, Fl_Osc_Widget
             }
 
             if (this->active_r())
-                fl_color(this->parent()->selection_color());
+                fl_color(0,255,0);
             else
                 fl_color(this->parent()->color());
             fl_line_style(0);
-- 
2.47.0

