From a638d75b586447ae16f7006041028631fdc2212e Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sun, 15 Mar 2015 10:12:21 -0400
Subject: [PATCH 0395/1936] Middleware: Add Missing Include

---
 src/Misc/MiddleWare.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/Misc/MiddleWare.cpp b/src/Misc/MiddleWare.cpp
index ab078335..5a573c01 100644
--- a/src/Misc/MiddleWare.cpp
+++ b/src/Misc/MiddleWare.cpp
@@ -3,6 +3,7 @@
 #include <cstring>
 #include <cstdio>
 #include <cstdlib>
+#include <cstdarg>
 #include <fstream>
 
 #include <rtosc/undo-history.h>
-- 
2.47.0

