From e64ba962bdab22ca68e111da4626d60de52517c3 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Tue, 14 Apr 2015 18:12:02 -0400
Subject: [PATCH 0409/1936] Middleware: Remove Memory Leak of Ringbuffers

---
 src/Misc/MiddleWare.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/Misc/MiddleWare.cpp b/src/Misc/MiddleWare.cpp
index 7ad93908..7547d034 100644
--- a/src/Misc/MiddleWare.cpp
+++ b/src/Misc/MiddleWare.cpp
@@ -812,6 +812,9 @@ MiddleWareImpl::~MiddleWareImpl(void)
 
     delete master;
     delete osc;
+    delete bToU;
+    delete uToB;
+
 }
 
 /** Threading When Saving
-- 
2.47.0

