From 0edfe5f27afee40d721dd8a28ab85f94fea0bd27 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Fri, 17 Apr 2015 11:07:18 -0400
Subject: [PATCH 0411/1936] Master: Fix Transposed memset args

---
 src/Misc/Master.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/Misc/Master.cpp b/src/Misc/Master.cpp
index 7d6d7e8f..46921268 100644
--- a/src/Misc/Master.cpp
+++ b/src/Misc/Master.cpp
@@ -304,7 +304,7 @@ Master::Master()
     {
         char loc_buf[1024];
         DataObj d{loc_buf, 1024, the_master, the_bToU};
-        memset(loc_buf, sizeof(loc_buf), 0);
+        memset(loc_buf, 0, sizeof(loc_buf));
         //printf("sending an event to the owner of '%s'\n", m);
         Master::ports.dispatch(m+1, d);
     };
@@ -319,7 +319,7 @@ void Master::applyOscEvent(const char *msg)
 {
     char loc_buf[1024];
     DataObj d{loc_buf, 1024, this, bToU};
-    memset(loc_buf, sizeof(loc_buf), 0);
+    memset(loc_buf, 0, sizeof(loc_buf));
     d.matches = 0;
     ports.dispatch(msg+1, d);
     if(d.matches == 0)
@@ -588,7 +588,7 @@ void Master::AudioOut(float *outl, float *outr)
     //Handle user events TODO move me to a proper location
     char loc_buf[1024];
     DataObj d{loc_buf, 1024, this, bToU};
-    memset(loc_buf, sizeof(loc_buf), 0);
+    memset(loc_buf, 0, sizeof(loc_buf));
     int events = 0;
     while(uToB->hasNext() && events < 10) {
         const char *msg = uToB->read();
-- 
2.47.0

