From 8454adc89d12be4b808939dc578d8bc0aa468837 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Fri, 17 Apr 2015 12:40:18 -0400
Subject: [PATCH 0418/1936] DSSI: Add Liblo Libraries

---
 src/CMakeLists.txt | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 6cfcd4fc..9e94e8a9 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -194,7 +194,7 @@ endif()
 if(LibloEnable)
 	include_directories(${LIBLO_INCLUDE_DIRS})
 	add_definitions(-DUSE_NSM=1)
-        list(APPEND AUDIO_LIBRARIES ${LIBLO_LDFLAGS})
+    list(APPEND AUDIO_LIBRARIES ${LIBLO_LIBRARIES})
 	list(APPEND AUDIO_LIBRARY_DIRS ${LIBLO_LIBRARY_DIRS})
 	message(STATUS "Compiling with liblo")
 endif()
@@ -389,6 +389,7 @@ if (DssiEnable)
 	target_link_libraries(zynaddsubfx_dssi
         zynaddsubfx_core
 		${OS_LIBRARIES}
+        ${LIBLO_LIBRARIES}
 		)
 	if (${CMAKE_SIZEOF_VOID_P} EQUAL "8")
 		install(TARGETS zynaddsubfx_dssi LIBRARY DESTINATION lib64/dssi/)
-- 
2.47.0

