From ca158d3535d383309ad733534a91607ed6f34b30 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Wed, 29 Apr 2015 11:58:06 -0400
Subject: [PATCH 0429/1936] Middleware: Add Missing Include

---
 src/Misc/MiddleWare.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/Misc/MiddleWare.h b/src/Misc/MiddleWare.h
index 2393a202..0c69404e 100644
--- a/src/Misc/MiddleWare.h
+++ b/src/Misc/MiddleWare.h
@@ -1,6 +1,7 @@
 #pragma once
 #include <functional>
 #include <cstdarg>
+#include <string>
 
 //Link between realtime and non-realtime layers
 class MiddleWare
-- 
2.47.0

