From d1e197ae055475af496e640215c6b989e91341f7 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Wed, 29 Apr 2015 16:50:14 -0400
Subject: [PATCH 0431/1936] CMake: Add Dependency Between UI Build Targets

---
 src/UI/CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/UI/CMakeLists.txt b/src/UI/CMakeLists.txt
index 620b9fd4..730ca0b1 100644
--- a/src/UI/CMakeLists.txt
+++ b/src/UI/CMakeLists.txt
@@ -64,6 +64,8 @@ add_library(zynaddsubfx_gui STATIC
 	)
 add_library(zynaddsubfx_gui_bridge STATIC
     Connection.cpp)
+add_dependencies(zynaddsubfx_gui_bridge zynaddsubfx_gui)
+
 
 if(NtkGui)
   target_link_libraries(zynaddsubfx_gui ${NTK_LIBRARIES} ${NTK_IMAGES_LIBRARIES})
-- 
2.47.0

