From 919185989b23c55ed3531d53726ce5eaf6d2de16 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Thu, 7 May 2015 17:09:42 -0400
Subject: [PATCH 0444/1936] EffUI: Fix Bad Filter Subpath

---
 src/UI/EffUI.fl | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/UI/EffUI.fl b/src/UI/EffUI.fl
index 2ae731f7..f444e997 100644
--- a/src/UI/EffUI.fl
+++ b/src/UI/EffUI.fl
@@ -1130,7 +1130,7 @@ eqgraph->update();}
       Fl_Group {} {
         label {DynFilter effect  - Filter}
         xywh {5 5 275 75} box FLAT_BOX color 50 align 144
-        code0 {o->init("",osc, loc(), "Filter/");}
+        code0 {o->init("",osc, loc(), "filterpars/");}
         code1 {o->use_for_dynamic_filter();}
         class FilterUI
       } {}
-- 
2.47.0

