From e27ce7833314b7d8756ff72d0fbd51c050aa6f89 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Fri, 8 May 2015 23:47:53 -0400
Subject: [PATCH 0453/1936] Fix Jack Detector

---
 src/Nio/Nio.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Nio/Nio.cpp b/src/Nio/Nio.cpp
index 21a4f236..83cd6a11 100644
--- a/src/Nio/Nio.cpp
+++ b/src/Nio/Nio.cpp
@@ -119,7 +119,7 @@ string Nio::getSink()
 void Nio::preferedSampleRate(unsigned &rate)
 {
     //XXX hello non portable code
-    if(system("ps | grep jack"))
+    if(system("ps aux | grep jack"))
         return;
 
     jack_client_t *client = jack_client_open("temp-client",
-- 
2.47.0

