From f91471407929047561ac510b99fc8d9445732e35 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sat, 9 May 2015 10:40:08 -0400
Subject: [PATCH 0454/1936] Fix xmz Save/Reload Test

---
 src/Tests/guitar-adnote.xmz | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/Tests/guitar-adnote.xmz b/src/Tests/guitar-adnote.xmz
index 97e8f951..acd391bf 100644
--- a/src/Tests/guitar-adnote.xmz
+++ b/src/Tests/guitar-adnote.xmz
@@ -874,6 +874,8 @@ version-revision="0" ZynAddSubFX-author="Nasca Octavian Paul">
 </INSTRUMENT>
 <CONTROLLER>
 <par name="pitchwheel_bendrange" value="200" />
+<par name="pitchwheel_bendrange_down" value="0" />
+<par_bool name="pitchwheel_split" value="no" />
 <par_bool name="expression_receive" value="yes" />
 <par name="panning_depth" value="64" />
 <par name="filter_cutoff_depth" value="64" />
@@ -1138,6 +1140,8 @@ version-revision="0" ZynAddSubFX-author="Nasca Octavian Paul">
 </INSTRUMENT>
 <CONTROLLER>
 <par name="pitchwheel_bendrange" value="200" />
+<par name="pitchwheel_bendrange_down" value="0" />
+<par_bool name="pitchwheel_split" value="no" />
 <par_bool name="expression_receive" value="yes" />
 <par name="panning_depth" value="64" />
 <par name="filter_cutoff_depth" value="64" />
@@ -3491,6 +3495,8 @@ version-revision="0" ZynAddSubFX-author="Nasca Octavian Paul">
 </INSTRUMENT>
 <CONTROLLER>
 <par name="pitchwheel_bendrange" value="200" />
+<par name="pitchwheel_bendrange_down" value="0" />
+<par_bool name="pitchwheel_split" value="no" />
 <par_bool name="expression_receive" value="yes" />
 <par name="panning_depth" value="64" />
 <par name="filter_cutoff_depth" value="64" />
-- 
2.47.0

