From 801b05b290eafb104ac5f6a85ae9cbacf5f36c36 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sat, 9 May 2015 12:17:59 -0400
Subject: [PATCH 0456/1936] Fix Legato/Kit Initialization

---
 src/Misc/Part.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Misc/Part.cpp b/src/Misc/Part.cpp
index e26fa45b..7bcf1e5c 100644
--- a/src/Misc/Part.cpp
+++ b/src/Misc/Part.cpp
@@ -629,7 +629,7 @@ void Part::NoteOn(unsigned char note,
 
                 // Spawn another note (but silent) if legatomodevalid==true
                 if(legatomodevalid) {
-                    auto &note2 = partnote[pos].kititem[ci];
+                    auto &note2 = partnote[posb].kititem[ci];
                     note2.sendtoparteffect = limit((int)kit[item].Psendtoparteffect, 0, NUM_PART_EFX);
 
                     pars.quiet = true;
-- 
2.47.0

