From beb30b873f49350c142217b95f5a233f160c2396 Mon Sep 17 00:00:00 2001
From: "Christopher A. Oliver" <oliver@onion.private>
Date: Sat, 9 May 2015 14:34:43 -0400
Subject: [PATCH 0459/1936] Update part's displayed instrument name when anyone
 sets it.

---
 src/UI/PartNameButton.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/UI/PartNameButton.cpp b/src/UI/PartNameButton.cpp
index 3db9db15..ecc09477 100644
--- a/src/UI/PartNameButton.cpp
+++ b/src/UI/PartNameButton.cpp
@@ -9,4 +9,5 @@ void PartNameButton::OSC_value(const char *label_)
 {
     the_string = label_;
     label(the_string.c_str());
+    redraw();
 }
-- 
2.47.0

