From 30eb0ca64c7f277899fcc663571ea8288548687a Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sat, 9 May 2015 19:23:54 -0400
Subject: [PATCH 0460/1936] Fix Part Panning rtosc Port

---
 src/Misc/Part.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/Misc/Part.cpp b/src/Misc/Part.cpp
index 7bcf1e5c..d22bad3b 100644
--- a/src/Misc/Part.cpp
+++ b/src/Misc/Part.cpp
@@ -57,6 +57,9 @@ static const Ports partPorts = {
 #define rChangeCb obj->setPvolume(obj->Pvolume);
     rParamZyn(Pvolume, "Part Volume"),
 #undef rChangeCb
+#define rChangeCb obj->setPpanning(obj->Ppanning);
+    rParamZyn(Ppanning, "Set Panning"),
+#undef rChangeCb
 #define rChangeCb obj->setkeylimit(obj->Pkeylimit);
     rParamI(Pkeylimit, rProp(parameter), rMap(min,0), rMap(max, POLYPHONY), "Key limit per part"),
 #undef rChangeCb
@@ -65,7 +68,6 @@ static const Ports partPorts = {
     rParamZyn(Pmaxkey, "Max Used Key"),
     rParamZyn(Pkeyshift, "Part keyshift"),
     rParamZyn(Prcvchn,  "Active MIDI channel"),
-    rParamZyn(Ppanning, "Set Panning"),
     rParamZyn(Pvelsns,   "Velocity sensing"),
     rParamZyn(Pveloffs,  "Velocity offset"),
     rToggle(Pnoteon,  "If the channel accepts note on events"),
-- 
2.47.0

