From 88bdd3ab509e74ed42699c4ad6466ab01fc91ceb Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sun, 17 May 2015 15:22:18 -0400
Subject: [PATCH 0470/1936] NTK External UI Fix

---
 src/UI/CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/UI/CMakeLists.txt b/src/UI/CMakeLists.txt
index e7a424d8..a42eaab5 100644
--- a/src/UI/CMakeLists.txt
+++ b/src/UI/CMakeLists.txt
@@ -34,6 +34,10 @@ if(FltkGui)
     add_executable(zynaddsubfx_ext_gui guimain.cpp)
     target_link_libraries(zynaddsubfx_ext_gui zynaddsubfx_gui ${FLTK_LIBRARIES}
         ${FLTK_LIBRARIES} ${OPENGL_LIBRARIES} ${X11_LIBRARIES} ${LIBLO_LIBRARIES} rtosc rtosc-cpp)
+elseif(NtkGui)
+    add_executable(zynaddsubfx_ext_gui guimain.cpp)
+    target_link_libraries(zynaddsubfx_ext_gui zynaddsubfx_gui ${NTK_LDFLAGS}
+        ${OPENGL_LIBRARIES} ${X11_LIBRARIES} ${LIBLO_LIBRARIES} rtosc rtosc-cpp)
 endif()
 
 add_library(zynaddsubfx_gui STATIC
-- 
2.47.0

