From d92788ac8f3a41f4259b41399d176e9fe9be5549 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Fri, 29 May 2015 18:30:42 -0400
Subject: [PATCH 0478/1936] CMake: Add zynaddsubfx-ext-gui Install Target

---
 src/UI/CMakeLists.txt | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/src/UI/CMakeLists.txt b/src/UI/CMakeLists.txt
index a42eaab5..63f93230 100644
--- a/src/UI/CMakeLists.txt
+++ b/src/UI/CMakeLists.txt
@@ -31,13 +31,15 @@ if(NtkGui)
 endif()
 
 if(FltkGui)
-    add_executable(zynaddsubfx_ext_gui guimain.cpp)
-    target_link_libraries(zynaddsubfx_ext_gui zynaddsubfx_gui ${FLTK_LIBRARIES}
+    add_executable(zynaddsubfx-ext-gui guimain.cpp)
+    target_link_libraries(zynaddsubfx-ext-gui zynaddsubfx_gui ${FLTK_LIBRARIES}
         ${FLTK_LIBRARIES} ${OPENGL_LIBRARIES} ${X11_LIBRARIES} ${LIBLO_LIBRARIES} rtosc rtosc-cpp)
+    install(TARGETS zynaddsubfx-ext-gui RUNTIME DESTINATION bin)
 elseif(NtkGui)
-    add_executable(zynaddsubfx_ext_gui guimain.cpp)
-    target_link_libraries(zynaddsubfx_ext_gui zynaddsubfx_gui ${NTK_LDFLAGS}
+    add_executable(zynaddsubfx-ext-gui guimain.cpp)
+    target_link_libraries(zynaddsubfx-ext-gui zynaddsubfx_gui ${NTK_LDFLAGS}
         ${OPENGL_LIBRARIES} ${X11_LIBRARIES} ${LIBLO_LIBRARIES} rtosc rtosc-cpp)
+    install(TARGETS zynaddsubfx-ext-gui RUNTIME DESTINATION bin)
 endif()
 
 add_library(zynaddsubfx_gui STATIC
-- 
2.47.0

