From 33efee9311454c51362314fee5bee20be3d52b81 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Mon, 1 Jun 2015 18:39:07 -0400
Subject: [PATCH 0485/1936] ADVoiceUI: Fix White Noise Label Display Behavior

---
 src/UI/ADnoteUI.fl | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/UI/ADnoteUI.fl b/src/UI/ADnoteUI.fl
index 7b6866cb..2a5343e5 100644
--- a/src/UI/ADnoteUI.fl
+++ b/src/UI/ADnoteUI.fl
@@ -761,7 +761,7 @@ bypassfiltercheckbutton->redraw();}
         }
         Fl_Box noiselabel {
           label {White Noise}
-          xywh {150 430 300 65} labelfont 1 labelsize 50 labelcolor 53
+          xywh {150 430 300 65} labelfont 1 labelsize 50 labelcolor 53 hide
         }
       }
       Fl_Check_Button voiceonbutton {
-- 
2.47.0

