From ddd10537c5e8cc32d1e239ad8861b0f75e9653ad Mon Sep 17 00:00:00 2001
From: "Christopher A. Oliver" <oliver@onion.private>
Date: Mon, 1 Jun 2015 19:55:56 -0400
Subject: [PATCH 0486/1936] Fix beginner menubar width.

---
 src/UI/MasterUI.fl | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/UI/MasterUI.fl b/src/UI/MasterUI.fl
index d3ead20f..b31341f2 100644
--- a/src/UI/MasterUI.fl
+++ b/src/UI/MasterUI.fl
@@ -926,7 +926,7 @@ if (fl_choice("Exit and leave the unsaved data?","No","Yes",NULL))
         code1 {simplemasterwindow->base = "/";}
       }
       Fl_Menu_Bar simplemastermenu {
-        xywh {0 0 690 25}
+        xywh {0 0 600 25}
       } {
         Submenu {} {
           label {&File}
-- 
2.47.0

