From 314ea4390cd6fcb374d025f720241b49612bfd40 Mon Sep 17 00:00:00 2001
From: falkTX <falktx@gmail.com>
Date: Mon, 8 Jun 2015 23:51:17 +0200
Subject: [PATCH 0492/1936] Don't set MasterUI icon for plugin version, not
 needed

---
 src/UI/MasterUI.fl | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/src/UI/MasterUI.fl b/src/UI/MasterUI.fl
index def3fa24..2fee90ac 100644
--- a/src/UI/MasterUI.fl
+++ b/src/UI/MasterUI.fl
@@ -17,7 +17,9 @@ decl {\#include <stdio.h>} {public local
 decl {\#include <string.h>} {public local
 } 
 
-decl {\#include "zynaddsubfx.xpm"} {private local
+decl {\#if defined(__linux__) && ! defined(PLUGINVERSION)
+\#include "zynaddsubfx.xpm"
+\#endif} {public local
 } 
 
 decl {\#include "WidgetPDial.h"} {public local
@@ -74,7 +76,7 @@ extern NSM_Client *nsm;
 \#endif} {public local
 } 
 
-decl {\#if __linux__
+decl {\#if defined(__linux__) && ! defined(PLUGINVERSION)
 \#include <X11/xpm.h>
 \#endif} {public local
 } 
@@ -1500,7 +1502,7 @@ configui=new ConfigUI(osc);
 make_window();
 fl_open_display();
 
-\#ifdef __linux__
+\#if defined(__linux__) && ! defined(PLUGINVERSION)
 Pixmap p, mask;
 XCreatePixmapFromData(fl_display, DefaultRootWindow(fl_display),
                                  (char**)zynaddsubfx_xpm, &p, &mask, NULL);
-- 
2.47.0

