From c6e7602e668eeed3571aefa143b8141851d823ca Mon Sep 17 00:00:00 2001
From: falkTX <falktx@gmail.com>
Date: Mon, 8 Jun 2015 23:52:31 +0200
Subject: [PATCH 0493/1936] Fix plugin build

---
 src/UI/NioUI.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/UI/NioUI.cpp b/src/UI/NioUI.cpp
index 4bd34592..b0879370 100644
--- a/src/UI/NioUI.cpp
+++ b/src/UI/NioUI.cpp
@@ -14,14 +14,14 @@
 
 using namespace std;
 
-static void callback_fn_choice(Fl_Widget *w, void *);
+static void callback_fn_choice_nio(Fl_Widget *w, void *);
 class Fl_Osc_StrChoice:public Fl_Choice, public Fl_Osc_Widget
 {
     public:
         Fl_Osc_StrChoice(int X, int Y, int W, int H, const char *label = NULL)
         :Fl_Choice(X,Y,W,H, label), Fl_Osc_Widget(this), cb_data(NULL, NULL)
         {
-            Fl_Choice::callback(callback_fn_choice, NULL);
+            Fl_Choice::callback(callback_fn_choice_nio, NULL);
         }
 
         virtual ~Fl_Osc_StrChoice(void) {};
@@ -69,7 +69,7 @@ class Fl_Osc_StrChoice:public Fl_Choice, public Fl_Osc_Widget
         int min;
         std::pair<Fl_Callback*, void*> cb_data;
 };
-static void callback_fn_choice(Fl_Widget *w, void *)
+static void callback_fn_choice_nio(Fl_Widget *w, void *)
 {
     ((Fl_Osc_StrChoice*)w)->cb();
 }
-- 
2.47.0

