From 51b78a8500059c1f6e324b59dd76aa1ccfb373d6 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Mon, 8 Jun 2015 18:50:14 -0400
Subject: [PATCH 0495/1936] Fix Ext GUI Induced Crashing

---
 src/UI/guimain.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/UI/guimain.cpp b/src/UI/guimain.cpp
index df9cc1cf..a22eb33d 100644
--- a/src/UI/guimain.cpp
+++ b/src/UI/guimain.cpp
@@ -328,13 +328,15 @@ class UI_Interface:public Fl_Osc_Interface
 
         void write(string s, const char *args, ...) override
         {
+            char buffer[4096];
             va_list va;
             va_start(va, args);
+            rtosc_vmessage(buffer, sizeof(buffer), s.c_str(), args, va);
             //fprintf(stderr, "%c[%d;%d;%dm", 0x1B, 0, 4 + 30, 0 + 40);
             ////fprintf(stderr, ".");
             //fprintf(stderr, "write(%s:%s)\n", s.c_str(), args);
             //fprintf(stderr, "%c[%d;%d;%dm", 0x1B, 0, 7 + 30, 0 + 40);
-            transmitMsg(s.c_str(), args, va);
+            transmitMsg(buffer);
             va_end(va);
         }
 
-- 
2.47.0

