From f748fd350c6cbd9d9ce6733e16911de6a6074c63 Mon Sep 17 00:00:00 2001
From: Johannes Lorenz <johannes89@ist-einmalig.de>
Date: Sat, 13 Jun 2015 07:52:25 +0200
Subject: [PATCH 0497/1936] Fixed alias behaviour for denormalkillbuf.

---
 TODO.txt        | 2 +-
 src/globals.cpp | 3 +++
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/TODO.txt b/TODO.txt
index 9cbd3129..4f2ca472 100644
--- a/TODO.txt
+++ b/TODO.txt
@@ -15,7 +15,7 @@ Status:
 (2) will be fixed with fftw3 (?), until then maybe use separate mutex?
 (3) will be fixed by fundamental
 (4)
-(5) fixed + tested, but alias() behaviour is unclear yet
+(5) fixed + tested
 (6)
 (7) -> will not be fixed (should stay random)
 (8) fixed + tested
diff --git a/src/globals.cpp b/src/globals.cpp
index c44edf96..08cb5447 100644
--- a/src/globals.cpp
+++ b/src/globals.cpp
@@ -32,6 +32,9 @@ void SYNTH_T::alias()
     oscilsize_f      = oscilsize;
 
     //produce denormal buf
+    // note: once there will be more buffers, use a cleanup function
+    // for deleting the buffers and also call it in the dtor
+    delete[] denormalkillbuf;
     denormalkillbuf = new float [buffersize];
     for(int i = 0; i < buffersize; ++i)
         denormalkillbuf[i] = (RND - 0.5f) * 1e-16;
-- 
2.47.0

