From b3c81bbfae24cf5b798e4e4c4c01d6c30e4a98ae Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Mon, 3 Aug 2015 18:28:20 -0400
Subject: [PATCH 0529/1936] OscilGen: Fix Possible Crash On Copy/Paste

---
 src/Synth/OscilGen.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/Synth/OscilGen.cpp b/src/Synth/OscilGen.cpp
index 2e7637d0..4022fa12 100644
--- a/src/Synth/OscilGen.cpp
+++ b/src/Synth/OscilGen.cpp
@@ -482,7 +482,8 @@ void OscilGen::changebasefunction(void)
 {
     if(Pcurrentbasefunc != 0) {
         getbasefunction(tmpsmps);
-        fft->smps2freqs(tmpsmps, basefuncFFTfreqs);
+        if(fft)
+            fft->smps2freqs(tmpsmps, basefuncFFTfreqs);
         clearDC(basefuncFFTfreqs);
     }
     else //in this case basefuncFFTfreqs are not used
@@ -1139,6 +1140,8 @@ void OscilGen::paste(OscilGen &o)
     RESTORE(fft);
     RESTORE(basefuncFFTfreqs);
     RESTORE(res);
+    if(this->Pcurrentbasefunc)
+        changebasefunction();
     this->prepare();
 }
 
-- 
2.47.0

