From 899dbaf5c661cca6b46f801e434c8ef8c860c865 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Tue, 29 Sep 2015 10:45:20 -0400
Subject: [PATCH 0540/1936] CMake: Remove FAQ From Install Directive

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 95793b74..06044763 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -28,7 +28,7 @@ include(CTestConfig.cmake)
 #Currently the only directory that uses cmake
 add_subdirectory(src)
 
-install(FILES AUTHORS.txt COPYING FAQ.txt HISTORY.txt README.adoc
+install(FILES AUTHORS.txt COPYING HISTORY.txt README.adoc
 	DESTINATION share/doc/zynaddsubfx
 	)
 install(FILES zynaddsubfx-jack.desktop zynaddsubfx-alsa.desktop
-- 
2.47.0

