From 158525bf09be375da84d284338f50a5656608334 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Wed, 30 Sep 2015 21:43:14 -0400
Subject: [PATCH 0544/1936] NotePool: Fix Legato Transition

---
 src/Containers/NotePool.cpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/Containers/NotePool.cpp b/src/Containers/NotePool.cpp
index 8b1f20af..7d1b1f40 100644
--- a/src/Containers/NotePool.cpp
+++ b/src/Containers/NotePool.cpp
@@ -117,8 +117,9 @@ int NotePool::getRunningNotes(void) const
 {
     bool running[256] = {0};
     for(auto &desc:activeDesc()) {
-        printf("note!(%d)\n", desc.note);
-        running[desc.note] = true;
+        //printf("note!(%d)\n", desc.note);
+        if(desc.status == Part::KEY_PLAYING)
+            running[desc.note] = true;
     }
 
     int running_count = 0;
-- 
2.47.0

