From 0948c5a9aba95bc98b22f4326f6696cd4d1ed3b7 Mon Sep 17 00:00:00 2001
From: "Christopher A. Oliver" <oliver@onion.private>
Date: Thu, 1 Oct 2015 10:54:53 -0400
Subject: [PATCH 0546/1936] Fix menu offset for voice unison

---
 src/UI/ADnoteUI.fl | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/UI/ADnoteUI.fl b/src/UI/ADnoteUI.fl
index cab22ee3..0cba412f 100644
--- a/src/UI/ADnoteUI.fl
+++ b/src/UI/ADnoteUI.fl
@@ -567,7 +567,7 @@ voiceonbutton->redraw();} open
               label Unison
               tooltip {Unison size} xywh {10 560 75 20} down_box BORDER_BOX labelfont 1 align 5 textfont 1 textsize 10
               code0 {o->add("OFF");char tmp[100];for (int i=1;ADnote_unison_sizes[i];i++){snprintf(tmp,100,"size %d",ADnote_unison_sizes[i]);o->add(tmp);};}
-              code1 {o->init("Unison_size");}
+              code1 {o->init("Unison_size", 1);}
               class Fl_Osc_Choice
             } {}
             Fl_Dial {} {
-- 
2.47.0

