From 03417924f0f74a9a3aad6330fc5e80b1b6e74841 Mon Sep 17 00:00:00 2001
From: "Christopher A. Oliver" <caowasteland@gmail.com>
Date: Sun, 4 Oct 2015 23:19:31 -0400
Subject: [PATCH 0560/1936] Try to position free mode edit window more
 reasonably.

---
 src/UI/EnvelopeUI.fl | 32 ++++++++++++++++++++++++--------
 1 file changed, 24 insertions(+), 8 deletions(-)

diff --git a/src/UI/EnvelopeUI.fl b/src/UI/EnvelopeUI.fl
index 5008d32e..8713333c 100644
--- a/src/UI/EnvelopeUI.fl
+++ b/src/UI/EnvelopeUI.fl
@@ -237,7 +237,10 @@ envfree->redraw();}
         }
         Fl_Button {} {
           label E
-          callback {freemodeeditwindow->show();}
+          callback {freemodeeditwindow->show();
+                    freemodeeditwindow->position(Fl::event_x_root()-20,
+                        Fl::event_y_root()+20);
+                   }
           tooltip {Envelope window} xywh {185 5 15 15} labelfont 1 labelsize 10
         }
         Fl_Check_Button e1linearenvelope {
@@ -312,7 +315,10 @@ envfree->redraw();}
       }
       Fl_Button {} {
         label E
-        callback {freemodeeditwindow->show();}
+        callback {freemodeeditwindow->show();
+                  freemodeeditwindow->position(Fl::event_x_root()-20,
+                      Fl::event_y_root()+20);
+                 }
         tooltip {Envelope window} xywh {190 5 15 15} labelfont 1 labelsize 10
       }
     }
@@ -393,7 +399,10 @@ envfree->redraw();}
         }
         Fl_Button {} {
           label E
-          callback {freemodeeditwindow->show();}
+          callback {freemodeeditwindow->show();
+                    freemodeeditwindow->position(Fl::event_x_root()-20,
+                        Fl::event_y_root()+20);
+                   }
           xywh {255 5 15 15} labelfont 1 labelsize 10
         }
       }
@@ -463,7 +472,10 @@ envfree->redraw();}
       }
       Fl_Button {} {
         label E
-        callback {freemodeeditwindow->show();}
+        callback {freemodeeditwindow->show();
+                  freemodeeditwindow->position(Fl::event_x_root()-20,
+                      Fl::event_y_root()+20);
+                 }
         xywh {190 5 15 15} labelfont 1 labelsize 10
       }
     }
@@ -481,7 +493,10 @@ envfree->redraw();}
       } {
         Fl_Button {} {
           label E
-          callback {freemodeeditwindow->show();}
+          callback {freemodeeditwindow->show();
+                    freemodeeditwindow->position(Fl::event_x_root()-20,
+                        Fl::event_y_root()+20);
+                   }
           xywh {185 5 15 15} labelfont 1 labelsize 10
         }
         Fl_Box freeeditsmall {
@@ -555,9 +570,10 @@ refresh();} {}
         freeedit->update();
 
     hide();
-    const int winx=freemodeeditwindow->x();
-    const int winy=freemodeeditwindow->y();
-
+    int winx = Fl::event_x_root()-10;
+    int winy = Fl::event_y_root()-155;
+    winx = winx < 1 ? 1 : winx;
+    winy = winy < 1 ? 1 : winy;
     bool reshow = freemodeeditwindow->visible();
     freemodeeditwindow->hide();
 
-- 
2.47.0

