From 184c5f62b8ae20ef39de2ea7fc79c25f537cfe03 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Mon, 5 Oct 2015 20:26:30 -0400
Subject: [PATCH 0564/1936] Partial Fix To Checkboxes In Effect UIs

---
 src/Effects/EffectMgr.cpp | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/src/Effects/EffectMgr.cpp b/src/Effects/EffectMgr.cpp
index 72d320fe..a196c7cc 100644
--- a/src/Effects/EffectMgr.cpp
+++ b/src/Effects/EffectMgr.cpp
@@ -45,7 +45,7 @@ static const rtosc::Ports local_ports = {
     rSelf(EffectMgr),
     rPaste,
     rRecurp(filterpars, "Filter Parameter for Dynamic Filter"),
-    {"parameter#128::i", rProp(alias) rDoc("Parameter Accessor"), NULL,
+    {"parameter#128::i:T:F", rProp(alias) rDoc("Parameter Accessor"), NULL,
         [](const char *msg, rtosc::RtData &d)
         {
             EffectMgr *eff = (EffectMgr*)d.obj;
@@ -54,9 +54,15 @@ static const rtosc::Ports local_ports = {
 
             if(!rtosc_narguments(msg))
                 d.reply(d.loc, "i", eff->geteffectparrt(atoi(mm)));
-            else {
+            else if(rtosc_type(msg, 0) == 'i'){
                 eff->seteffectparrt(atoi(mm), rtosc_argument(msg, 0).i);
                 d.broadcast(d.loc, "i", eff->geteffectparrt(atoi(mm)));
+            } else if(rtosc_type(msg, 0) == 'T'){
+                eff->seteffectparrt(atoi(mm), 127);
+                d.broadcast(d.loc, "i", eff->geteffectparrt(atoi(mm)));
+            } else if(rtosc_type(msg, 0) == 'F'){
+                eff->seteffectparrt(atoi(mm), 0);
+                d.broadcast(d.loc, "i", eff->geteffectparrt(atoi(mm)));
             }
         }},
     {"preset::i", rProp(alias) rDoc("Effect Preset Selector"), NULL,
-- 
2.47.0

