From 61cc2daef66cecfc31d1acace1c8d7d41e3efcd7 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Mon, 5 Oct 2015 20:31:22 -0400
Subject: [PATCH 0566/1936] Fix Unison Test

---
 src/Tests/UnisonTest.h | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/Tests/UnisonTest.h b/src/Tests/UnisonTest.h
index e207272f..9be69a35 100644
--- a/src/Tests/UnisonTest.h
+++ b/src/Tests/UnisonTest.h
@@ -93,7 +93,10 @@ class AdNoteTest:public CxxTest::TestSuite
         void run_test(int a, int b, int c, int d, int e, int f, float values[4])
         {
             sprng(0);
-            //params->set_unison_size_index(0,a);
+            
+            const int ADnote_unison_sizes[] =
+            {1, 2, 3, 4, 5, 6, 8, 10, 12, 15, 20, 25, 30, 40, 50, 0};
+            params->VoicePar[0].Unison_size = ADnote_unison_sizes[a];
             params->VoicePar[0].Unison_frequency_spread = b;
             params->VoicePar[0].Unison_stereo_spread    = c;
             params->VoicePar[0].Unison_vibratto         = d;
-- 
2.47.0

