From 4946e31f7a7ee91dc23be10d00bea0bc62f521e9 Mon Sep 17 00:00:00 2001
From: "Christopher A. Oliver" <caowasteland@gmail.com>
Date: Tue, 6 Oct 2015 23:27:04 -0400
Subject: [PATCH 0577/1936] Multiple includsion guard

---
 src/UI/TipWin.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/UI/TipWin.h b/src/UI/TipWin.h
index fe1698cb..eb791753 100644
--- a/src/UI/TipWin.h
+++ b/src/UI/TipWin.h
@@ -1,3 +1,5 @@
+#ifndef TIPWIN_H
+#define TIPWIN_H
 using namespace std;
 
 class TipWin:public Fl_Menu_Window
@@ -15,3 +17,4 @@ class TipWin:public Fl_Menu_Window
         string text;
         bool   textmode;
 };
+#endif
-- 
2.47.0

