From d94160d64c7fa2724047a3a645684f9b11cfc692 Mon Sep 17 00:00:00 2001
From: "Christopher A. Oliver" <caowasteland@gmail.com>
Date: Wed, 7 Oct 2015 03:27:56 -0400
Subject: [PATCH 0578/1936] Oops.  Scale factor should default to 1.0.

---
 src/UI/WidgetPDial.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/UI/WidgetPDial.cpp b/src/UI/WidgetPDial.cpp
index f7304d27..0e3abbeb 100644
--- a/src/UI/WidgetPDial.cpp
+++ b/src/UI/WidgetPDial.cpp
@@ -19,7 +19,7 @@
 
 WidgetPDial::WidgetPDial(int x, int y, int w, int h, const char *label)
   :Fl_Dial(x, y, w, h, label), oldvalue(0.0f), pos(false), textset(false),
-   value_offset(0.0), value_scale(0.0)
+   value_offset(0.0), value_scale(1.0)
 {
     //cout << "[" << label << "] There are now " << ++numobj << endl;
     Fl_Group *save = Fl_Group::current();
-- 
2.47.0

