From 41559b4f8e7dd472ddd6f065d1133c02f1622ef3 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Wed, 7 Oct 2015 14:31:28 -0400
Subject: [PATCH 0580/1936] Update Rtosc Again

Fixes Small Linker Issue
---
 rtosc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rtosc b/rtosc
index 09b44c59..bcc4f039 160000
--- a/rtosc
+++ b/rtosc
@@ -1 +1 @@
-Subproject commit 09b44c5998657c1de5276f659bd7f1a3b5ae8042
+Subproject commit bcc4f0398e1fc63ce7c5bd0ec2e8a48d70307d7f
-- 
2.47.0

