From aa3f00093471e8d15a1db67b201cde2785f11881 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Wed, 7 Oct 2015 15:56:24 -0400
Subject: [PATCH 0581/1936] Rtosc: Fix Duplicate Port In PADnoteParameters

---
 src/Params/PADnoteParameters.cpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/Params/PADnoteParameters.cpp b/src/Params/PADnoteParameters.cpp
index 8b5367cc..fe92307d 100644
--- a/src/Params/PADnoteParameters.cpp
+++ b/src/Params/PADnoteParameters.cpp
@@ -90,8 +90,6 @@ static const rtosc::Ports PADnotePorts =
 
     PC(fixedfreq),
     PC(fixedfreqET),
-    //TODO detune, coarse detune
-    PC(DetuneType),
     PC(Stereo),
     PC(Panning),
     PC(AmpVelocityScaleFunction),
-- 
2.47.0

