From ebbb2009ec944a6c8772977df402424cd59a6076 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sat, 10 Oct 2015 12:21:36 -0400
Subject: [PATCH 0585/1936] Fix Effect Copy/Paste

---
 src/Effects/EffectMgr.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Effects/EffectMgr.cpp b/src/Effects/EffectMgr.cpp
index a196c7cc..2f428454 100644
--- a/src/Effects/EffectMgr.cpp
+++ b/src/Effects/EffectMgr.cpp
@@ -402,7 +402,7 @@ void EffectMgr::paste(EffectMgr &e)
     changeeffectrt(e.nefx, true);
     changepresetrt(e.preset, true);
     for(int i=0;i<128;++i)
-        seteffectparrt(e.settings[i], i);
+        seteffectparrt(i, e.settings[i]);
 }
 
 void EffectMgr::add2XML(XMLwrapper *xml)
-- 
2.47.0

