From a8ab63089ac7f66a0e750df97ecd5c9211d6fb17 Mon Sep 17 00:00:00 2001
From: "Christopher A. Oliver" <caowasteland@gmail.com>
Date: Sat, 10 Oct 2015 15:04:12 -0400
Subject: [PATCH 0587/1936] Remove extraneous stateful widget stuff from
 PountButton.

---
 src/UI/EnvelopeUI.fl | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/UI/EnvelopeUI.fl b/src/UI/EnvelopeUI.fl
index e519d41b..afe2317c 100644
--- a/src/UI/EnvelopeUI.fl
+++ b/src/UI/EnvelopeUI.fl
@@ -62,13 +62,13 @@ class PointButton {open : {public Fl_Button, public Fl_Osc_Widget}}
   } {
     code {} {}
   }
-  Function {Osc_value(bool v)} {open
+  Function {rebase(std::string new_base)} {open
   } {
-    code {Fl_Button::value(v);} {}
+    code {loc = new_base;} {}
   }
   Function {init(std::string path_)} {open
   } {
-    code {ext = path_; oscRegister(ext.c_str());} {}
+    code {ext = path_;} {}
   }
 }
 class EnvelopeUI {open : {public Fl_Osc_Group,PresetsUI_}
-- 
2.47.0

