From ce43dc70254924e55e8db9744ce2db50371d78e1 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sat, 10 Oct 2015 15:16:18 -0400
Subject: [PATCH 0589/1936] Part Effect UI: Fix Copy/Paste

---
 src/UI/PartUI.fl | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/UI/PartUI.fl b/src/UI/PartUI.fl
index 46b2edce..119dba12 100644
--- a/src/UI/PartUI.fl
+++ b/src/UI/PartUI.fl
@@ -724,12 +724,12 @@ if (x==2) part->partefx[ninseff]->setdryonly(true);
       }
       Fl_Button {} {
         label C
-        callback {presetsui->copy(partfx->loc());}
+        callback {presetsui->copy(partfx->loc()+"partefx"+to_s(ninseff)+"/");}
         xywh {90 127 25 15} box THIN_UP_BOX color 179 labelfont 1 labelsize 11 labelcolor 55
       }
       Fl_Button {} {
         label P
-        callback {presetsui->paste(partfx->loc(),inseffectui);}
+        callback {presetsui->paste(partfx->loc()+"partefx"+to_s(ninseff)+"/",inseffectui);}
         xywh {120 127 25 15} box THIN_UP_BOX color 179 labelfont 1 labelsize 11 labelcolor 55
       }
     }
-- 
2.47.0

