From d30e155b0f8135f0556875c9307c9ad11834e9eb Mon Sep 17 00:00:00 2001
From: "Christopher A. Oliver" <caowasteland@gmail.com>
Date: Mon, 12 Oct 2015 11:52:35 -0400
Subject: [PATCH 0594/1936] Make sure legato stragglers get nailed.

---
 src/Misc/Part.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/Misc/Part.cpp b/src/Misc/Part.cpp
index 95b5d4e1..559defa8 100644
--- a/src/Misc/Part.cpp
+++ b/src/Misc/Part.cpp
@@ -754,6 +754,7 @@ void Part::ComputePartSmps()
             partoutr[i] *= tmp;
         }
         notePool.killAllNotes();
+        monomemClear();
         killallnotes = false;
         for(int nefx = 0; nefx < NUM_PART_EFX; ++nefx)
             partefx[nefx]->cleanup();
-- 
2.47.0

