From c698d2cd47df5597dbaea091882839e70b6e849f Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Tue, 13 Oct 2015 20:58:29 -0400
Subject: [PATCH 0599/1936] Update Rtosc - Fixes Another Off By One In Oscdoc

---
 rtosc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rtosc b/rtosc
index c070c1aa..f509907e 160000
--- a/rtosc
+++ b/rtosc
@@ -1 +1 @@
-Subproject commit c070c1aaa4d4d77a4cb1556b2705e633f7a77569
+Subproject commit f509907e7ffad402a569e940dfc08122eaf8f871
-- 
2.47.0

