From 386d97862b301d5ffaa6df1708788540012e8e10 Mon Sep 17 00:00:00 2001
From: "Christopher A. Oliver" <caowasteland@gmail.com>
Date: Sat, 17 Oct 2015 12:40:55 -0400
Subject: [PATCH 0607/1936] Oops.  We won't need DSliders any more.

---
 src/UI/CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/UI/CMakeLists.txt b/src/UI/CMakeLists.txt
index f5c9152e..5ab0cd64 100644
--- a/src/UI/CMakeLists.txt
+++ b/src/UI/CMakeLists.txt
@@ -67,7 +67,6 @@ add_library(zynaddsubfx_gui STATIC
         Fl_Osc_DialF.cpp
         Fl_Osc_Slider.cpp
         Fl_Osc_TSlider.cpp
-        Fl_Osc_DSlider.cpp
         Fl_Osc_VSlider.cpp
         Fl_Osc_Button.cpp
         Fl_Osc_Check.cpp
-- 
2.47.0

