From f320fc1f871513ca5662cfe35c9a9a9520fc9ec5 Mon Sep 17 00:00:00 2001
From: "Christopher A. Oliver" <caowasteland@gmail.com>
Date: Sat, 17 Oct 2015 14:03:04 -0400
Subject: [PATCH 0610/1936] Fix transition between freemode and knobs on paste

---
 src/UI/EnvelopeUI.fl | 20 ++++++++++++++------
 1 file changed, 14 insertions(+), 6 deletions(-)

diff --git a/src/UI/EnvelopeUI.fl b/src/UI/EnvelopeUI.fl
index afe2317c..1e990597 100644
--- a/src/UI/EnvelopeUI.fl
+++ b/src/UI/EnvelopeUI.fl
@@ -178,7 +178,7 @@ envfree->redraw();}
       }
       Fl_Check_Button freemodehack {
           xywh {0 0 0 0} down_box DOWN_BOX
-          callback{refresh();}
+          callback{refresh_display();}
           code0 {o->init("Pfreemode");o->hide();}
           class Fl_Osc_Check
       }
@@ -568,7 +568,7 @@ freeeditsmall->setpair(freeedit);
 freeedit->setpair(freeeditsmall);
 
 
-refresh();} {}
+refresh_display();} {}
   }
   Function {rebase(std::string new_base)} {open
   } {
@@ -600,9 +600,16 @@ refresh();} {}
     show();
     freemodeeditwindow->position(winx,winy);} {}
   }
-  Function {refresh()} {open
+  Function {refresh()} { open }
+  { code {
+    freemodehack->oscWrite("Pfreemode");
+  } {}
+  }
+  Function {refresh_display()} {open
   } {
-    code {sustaincounter->value(Penvsustain);
+    code {
+
+    sustaincounter->value(Penvsustain);
     sustaincounter->maximum(Penvpoints-2);
 
     envstretchdial->value(Penvstretch);
@@ -618,9 +625,10 @@ envADSRfilter->hide();
 envASRbw->hide();
 envfree->hide();
 
-if (freemodehack->value())
+if (freemodehack->value()) {
    envwindow=envfree;
-else {
+   freeedit->update();
+} else {
    freemodeeditwindow->hide();
    switch(Envmode){
          case 1:
-- 
2.47.0

