From fd6de50ff3542a9b4abc9eb48d6d879dbef79405 Mon Sep 17 00:00:00 2001
From: "Christopher A. Oliver" <caowasteland@gmail.com>
Date: Sun, 18 Oct 2015 11:42:58 -0400
Subject: [PATCH 0613/1936] Fix wrong C/P label colors some more.

---
 src/UI/FilterUI.fl | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/UI/FilterUI.fl b/src/UI/FilterUI.fl
index e2ad38f0..ca6a1b2b 100644
--- a/src/UI/FilterUI.fl
+++ b/src/UI/FilterUI.fl
@@ -89,7 +89,7 @@ delete (formantparswindow);} {}
       Fl_Button {} {
         label P
         callback {presetsui->paste(filterui->loc(),this);}
-        xywh {203 5 15 15} box THIN_UP_BOX color 179 labelfont 1 labelsize 10 labelcolor 7
+        xywh {203 5 15 15} box THIN_UP_BOX color 179 labelfont 1 labelsize 10 labelcolor 55
       }
       Fl_Group filterparamswindow {
         label {Filter Parameters}
@@ -258,7 +258,7 @@ delete (formantparswindow);} {}
       Fl_Button {} {
         label P
         callback {presetsui->paste(filterui->loc(),this,nvowel);}
-        xywh {665 25 25 15} box THIN_UP_BOX color 179 labelfont 1 labelsize 11 labelcolor 7
+        xywh {665 25 25 15} box THIN_UP_BOX color 179 labelfont 1 labelsize 11 labelcolor 55
         code0 {formantparswindow->osc = osc;}
         code1 {formantparswindow->base = loc();}
       }
-- 
2.47.0

