From 5bc589224b5c57d3eda0ff077dd809d4c3f9cbd5 Mon Sep 17 00:00:00 2001
From: "Christopher A. Oliver" <caowasteland@gmail.com>
Date: Sun, 18 Oct 2015 23:34:02 -0400
Subject: [PATCH 0616/1936] Fix sense of phase sliders

---
 src/UI/ADnoteUI.fl | 48 ++++++++++++++++++++++++++++++----------------
 1 file changed, 32 insertions(+), 16 deletions(-)

diff --git a/src/UI/ADnoteUI.fl b/src/UI/ADnoteUI.fl
index 9a4d5777..050767bc 100644
--- a/src/UI/ADnoteUI.fl
+++ b/src/UI/ADnoteUI.fl
@@ -79,6 +79,30 @@ decl {\#include "MasterUI.h"} {private local
 decl {extern class MasterUI *ui;} {private local
 } 
 
+class PhaseSlider {: {public Fl_Osc_TSlider}
+} {
+  Function {PhaseSlider(int x,int y, int w, int h, const char *label=0)
+  :Fl_Osc_TSlider(x,y,w,h,label)} {open
+  } { code {
+  setRounding(1);
+  reset_value=0;
+  setTransform(180.0/64, 0);
+  }}
+   Function {OSC_value(int i)} {open return_type void
+   } { code {
+       value(64-i);
+   } {} }
+  Function {cb(void)} {open return_type void
+  } {
+    code {
+	oscWrite(ext, "i", (int)(64-Fl_Slider::value()));
+	if(cb_data.first)
+	    cb_data.first(this, cb_data.second);
+} {}
+   }
+}
+
+
 class ADvoicelistitem {open : {public Fl_Osc_Group}
 } {
   Function {make_window()} {open private
@@ -366,15 +390,11 @@ oscedit=new OscilEditor(true, loc+"FMSmp/", osc_i);}
               }
               Fl_Slider {} {
                 label Phase
-                callback {oscFM->phase=64-(int) o->value();
+                callback {oscFM->phase=64-(int)o->value();
 fmoscil->redraw();}
-                xywh {645 415 105 15} type {Horz Knob} box NO_BOX labelsize 10 align 5 minimum -64 maximum 63 step 1
-                code0 {
-		    o->init("PFMoscilphase", 'i');
-		    o->setTransform(180.0/64, 0);
-		    o->setRounding(1);
-		}
-                class Fl_Osc_TSlider
+                xywh {645 415 105 15} type {Horz Knob} box NO_BOX labelsize 10 align 5 minimum -63 maximum 64 step 1
+                code0 { o->init("PFMoscilphase", 'i'); }
+                class PhaseSlider
               }
               Fl_Choice extFMoscil {
                 label Use
@@ -552,15 +572,11 @@ o->redraw();}
           }
           Fl_Slider {} {
             label Phase
-            callback {osc->phase=64-(int) o->value();
+            callback {osc->phase=64-(int)o->value();
 voiceoscil->redraw();}
-            xywh {10 435 65 10} type {Horz Knob} box NO_BOX labelsize 10 align 5 minimum -64 maximum 63 step 1
-            code0 {
-                o->init("Poscilphase", 'i');
-                o->setTransform(180.0/64, 0);
-		o->setRounding(1);
-            }
-            class Fl_Osc_TSlider
+            xywh {10 435 65 10} type {Horz Knob} box NO_BOX labelsize 10 align 5 minimum -63 maximum 64 step 1
+            code0 { o->init("Poscilphase", 'i'); }
+            class PhaseSlider
           }
           Fl_Check_Button {} {
             label {R.}
-- 
2.47.0

