From 0863e651fb27806be524fd64009f443ca2c19097 Mon Sep 17 00:00:00 2001
From: falkTX <falktx@gmail.com>
Date: Thu, 22 Oct 2015 10:04:01 +0200
Subject: [PATCH 0622/1936] Add missing '#pragma once' to TmpFileMgr.h

---
 src/Misc/TmpFileMgr.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/Misc/TmpFileMgr.h b/src/Misc/TmpFileMgr.h
index fad0c0b6..8809d4e0 100644
--- a/src/Misc/TmpFileMgr.h
+++ b/src/Misc/TmpFileMgr.h
@@ -1,3 +1,4 @@
+#pragma once
 
 /**
     This file provides routines for using zyn's tmp files.
-- 
2.47.0

