From b677e0748450eb877012f97b952ca494a34cb56a Mon Sep 17 00:00:00 2001
From: "Christopher A. Oliver" <caowasteland@gmail.com>
Date: Thu, 22 Oct 2015 20:45:56 -0400
Subject: [PATCH 0625/1936] Reset the wave type and parameter when sine
 conversion's done.

---
 src/UI/OscilGenUI.fl | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/UI/OscilGenUI.fl b/src/UI/OscilGenUI.fl
index 4504845b..8dbbf71a 100644
--- a/src/UI/OscilGenUI.fl
+++ b/src/UI/OscilGenUI.fl
@@ -880,6 +880,9 @@ redrawoscil();}
           callback {if (!fl_choice("Convert to SINE?","No","Yes",NULL)) return;
 
 osc->requestValue(loc+"convert2sine");
+bftype->update();
+bfslider->value(0);
+bfslider->do_callback();
 
 redrawoscil();
 refresh();}
-- 
2.47.0

