From 127e19d36bc61d5ed4beb3dc8c62cf24fdd65b1c Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Fri, 23 Oct 2015 14:21:58 -0400
Subject: [PATCH 0630/1936] UI: Fix Modulator External Osc Display

---
 src/UI/ADnoteUI.fl | 13 ++++++++-----
 1 file changed, 8 insertions(+), 5 deletions(-)

diff --git a/src/UI/ADnoteUI.fl b/src/UI/ADnoteUI.fl
index 498aa535..3c2a1421 100644
--- a/src/UI/ADnoteUI.fl
+++ b/src/UI/ADnoteUI.fl
@@ -421,11 +421,10 @@ o->redraw();}
                 callback {if (oscedit!=NULL)
     delete(oscedit);
 
-//int nv=nvoice;
-//if (extFMoscil->value()>=0)
-//    nv=extFMoscil->value();
-
-oscedit=new OscilEditor(true, loc+"FMSmp/", osc_i);}
+                if(extFMoscil->value() == 0)
+                    oscedit=new OscilEditor(true,loc+"FMSmp/", osc_i);
+                else
+                    oscedit=new OscilEditor(true,loc+"../VoicePar"+to_s(extFMoscil->value()-1)+"/FMSmp/", osc_i);}
                 xywh {700 380 55 15} box THIN_UP_BOX labelfont 1 labelsize 11
                 code0 {(void)o;/*if (extFMoscil->value()>=0) o->labelcolor(FL_BLUE);*/}
               }
@@ -446,6 +445,10 @@ fmoscil->redraw();}
     oscFM->init(false);
     changeFMoscilbutton->labelcolor(FL_BLACK);
 };
+if(o->value() == 0)
+fmoscil->rebase(loc);
+else
+fmoscil->rebase(loc+"../VoicePar"+to_s(o->value()-1)+"/");
 voiceFMparametersgroup->redraw();} open
                 xywh {560 410 75 20} down_box BORDER_BOX labelsize 10 textfont 1 textsize 10
                 code0 {o->add("Internal");}
-- 
2.47.0

