From fd422cd3c4d7019825917b48f741c9fb7615b64a Mon Sep 17 00:00:00 2001
From: "Christopher A. Oliver" <caowasteland@gmail.com>
Date: Tue, 27 Oct 2015 19:43:14 -0400
Subject: [PATCH 0647/1936] Remember the params just written as last_xmz

---
 src/UI/MasterUI.fl | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/UI/MasterUI.fl b/src/UI/MasterUI.fl
index c2e07f3f..92af3f16 100644
--- a/src/UI/MasterUI.fl
+++ b/src/UI/MasterUI.fl
@@ -1689,6 +1689,7 @@ char *tmp;
 if (result<0) fl_alert("Error: Could not save the file.");
 	else
 {
+    osc->write("/last_xmz", "s", filename);
 \#if USE_NSM
         if ( nsm && nsm->is_active() )
              setfilelabel( nsm->display_name );
-- 
2.47.0

