From a46ad045fc916bf21be4c622c5aec401c9130aad Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sun, 1 Nov 2015 21:35:35 -0500
Subject: [PATCH 0664/1936] Mono-Mode: Fix Unexpected Note Replay On Note Off

---
 src/Misc/Part.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Misc/Part.cpp b/src/Misc/Part.cpp
index 60375138..cd917ae7 100644
--- a/src/Misc/Part.cpp
+++ b/src/Misc/Part.cpp
@@ -492,7 +492,7 @@ void Part::NoteOff(unsigned char note) //release the key
         monomemPop(note);
 
     for(auto &desc:notePool.activeDesc()) {
-        if(desc.note != note)
+        if(desc.note != note || desc.status != KEY_PLAYING)
             continue;
         if(!ctl.sustain.sustain) { //the sustain pedal is not pushed
             if((isMonoMode() || isLegatoMode()) && !monomemEmpty())
-- 
2.47.0

