From 70ae04636f723a5e639f750d5e367841f3c854ef Mon Sep 17 00:00:00 2001
From: "Christopher A. Oliver" <caowasteland@gmail.com>
Date: Mon, 2 Nov 2015 12:59:37 -0500
Subject: [PATCH 0670/1936] Add missing saves for documentation edits

---
 src/UI/PartUI.fl | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/UI/PartUI.fl b/src/UI/PartUI.fl
index 114ab841..25867bfc 100644
--- a/src/UI/PartUI.fl
+++ b/src/UI/PartUI.fl
@@ -165,6 +165,7 @@ maxkcounter->do_callback();}
           label {Bass Drum}
           xywh {90 0 130 15} box THIN_DOWN_BOX labelfont 1 labelsize 10 align 20
           code0 {o->init("Pname");}
+	  callback { o->oscWrite(o->ext, "s", o->value()); }
           class Fl_Osc_Input
         }
         Fl_Check_Button adcheck {
@@ -940,6 +941,7 @@ if (x==2) part->partefx[ninseff]->setdryonly(true);
           xywh {5 60 385 50} type Multiline color 124 labelsize 10 align 5
           code0 {o->maximum_size(MAX_INFO_TEXT_SIZE);}
           code1 {o->init("info.Pauthor");}
+	  callback { o->oscWrite(o->ext, "s", o->value()); }
           class Fl_Osc_Input
         }
         Fl_Input {} {
@@ -947,6 +949,7 @@ if (x==2) part->partefx[ninseff]->setdryonly(true);
           xywh {5 125 385 90} type Multiline color 124 labelsize 11 align 5
           code0 {o->maximum_size(MAX_INFO_TEXT_SIZE);}
           code1 {o->init("info.Pcomments");}
+	  callback { o->oscWrite(o->ext, "s", o->value()); }
           class Fl_Osc_Input
         }
         Fl_Choice {} {
-- 
2.47.0

