From e382f35a8d77e22295697c99cded5e825c25493d Mon Sep 17 00:00:00 2001
From: "Christopher A. Oliver" <caowasteland@gmail.com>
Date: Tue, 3 Nov 2015 22:02:49 -0500
Subject: [PATCH 0675/1936] Fix for missing harmonics in base function save
 (fundamental)

---
 src/Synth/OscilGen.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Synth/OscilGen.cpp b/src/Synth/OscilGen.cpp
index b7b2fa27..3013ecef 100644
--- a/src/Synth/OscilGen.cpp
+++ b/src/Synth/OscilGen.cpp
@@ -1239,7 +1239,7 @@ void OscilGen::add2XML(XMLwrapper *xml)
         for(int i = 1; i < synth.oscilsize / 2; ++i) {
             float xc = basefuncFFTfreqs[i].real();
             float xs = basefuncFFTfreqs[i].imag();
-            if((fabs(xs) > 1e-6f) && (fabs(xc) > 1e-6f)) {
+            if((fabs(xs) > 1e-6f) || (fabs(xc) > 1e-6f)) {
                 xml->beginbranch("BF_HARMONIC", i);
                 xml->addparreal("cos", xc);
                 xml->addparreal("sin", xs);
-- 
2.47.0

