From caafd63e7d57a6e78e2ddb9dfff5b85f309bd2c8 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Fri, 6 Nov 2015 19:16:31 -0500
Subject: [PATCH 0690/1936] Remove Undefined Behavior In LFOParams::Paste

---
 src/Params/LFOParams.cpp | 15 +++++++++++----
 1 file changed, 11 insertions(+), 4 deletions(-)

diff --git a/src/Params/LFOParams.cpp b/src/Params/LFOParams.cpp
index 43dc0838..b35cfb9e 100644
--- a/src/Params/LFOParams.cpp
+++ b/src/Params/LFOParams.cpp
@@ -142,10 +142,17 @@ void LFOParams::getfromXML(XMLwrapper *xml)
     Pcontinous  = xml->getparbool("continous", Pcontinous);
 }
 
+#define COPY(y) this->y=x.y
 void LFOParams::paste(LFOParams &x)
 {
-    //Avoid undefined behavior
-    if(&x == this)
-        return;
-    memcpy((char*)this, (const char*)&x, sizeof(*this));
+    COPY(Pfreq);
+    COPY(Pintensity);
+    COPY(Pstartphase);
+    COPY(PLFOtype);
+    COPY(Prandomness);
+    COPY(Pfreqrand);
+    COPY(Pdelay);
+    COPY(Pcontinous);
+    COPY(Pstretch);
 }
+#undef COPY
-- 
2.47.0

