From 7f32cc009e4b5cfec68b0997f84bbc7954b7906e Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Fri, 6 Nov 2015 22:52:44 -0500
Subject: [PATCH 0691/1936] Remove Undefined Behavior In OscilGen::Paste

---
 src/Synth/OscilGen.cpp | 60 +++++++++++++++++++++++++++++-------------
 1 file changed, 42 insertions(+), 18 deletions(-)

diff --git a/src/Synth/OscilGen.cpp b/src/Synth/OscilGen.cpp
index 293c0ad8..174320e7 100644
--- a/src/Synth/OscilGen.cpp
+++ b/src/Synth/OscilGen.cpp
@@ -1177,31 +1177,55 @@ void OscilGen::getcurrentbasefunction(float *smps)
         getbasefunction(smps);   //the sine case
 }
 
-#define PRESERVE(x) decltype(this->x) x = this->x
-#define RESTORE(x)  this->x = x
+#define COPY(y) this->y = o.y
 void OscilGen::paste(OscilGen &o)
 {
     //XXX Figure out a better implementation of this sensitive to RT issues...
-    //Preserve Pointer Elements
-    PRESERVE(oscilFFTfreqs);
-    PRESERVE(pendingfreqs);
-    PRESERVE(tmpsmps);
-    PRESERVE(outoscilFFTfreqs);
-    PRESERVE(fft);
-    PRESERVE(basefuncFFTfreqs);
-    PRESERVE(res);
-    memcpy((char*)this, (char*)&o, sizeof(*this));
-    RESTORE(oscilFFTfreqs);
-    RESTORE(pendingfreqs);
-    RESTORE(tmpsmps);
-    RESTORE(outoscilFFTfreqs);
-    RESTORE(fft);
-    RESTORE(basefuncFFTfreqs);
-    RESTORE(res);
+    for(int i=0; i<MAX_AD_HARMONICS; ++i) {
+        COPY(Phmag[i]);
+        COPY(Phphase[i]);
+    }
+
+    COPY(Phmagtype);
+    COPY(Pcurrentbasefunc);
+    COPY(Pbasefuncpar);
+
+    COPY(Pbasefuncmodulation);
+    COPY(Pbasefuncmodulationpar1);
+    COPY(Pbasefuncmodulationpar2);
+    COPY(Pbasefuncmodulationpar3);
+
+    COPY(Pwaveshaping);
+    COPY(Pwaveshapingfunction);
+    COPY(Pfiltertype);
+    COPY(Pfilterpar1);
+    COPY(Pfilterpar2);
+    COPY(Pfilterbeforews);
+    COPY(Psatype);
+    COPY(Psapar);
+
+    COPY(Pharmonicshift);
+    COPY(Pharmonicshiftfirst);
+
+    COPY(Pmodulation);
+    COPY(Pmodulationpar1);
+    COPY(Pmodulationpar2);
+    COPY(Pmodulationpar3);
+
+    COPY(Prand);
+    COPY(Pamprandpower);
+    COPY(Pamprandtype);
+    COPY(Padaptiveharmonics);
+    COPY(Padaptiveharmonicsbasefreq);
+    COPY(Padaptiveharmonicspower);
+    COPY(Padaptiveharmonicspar);
+
+
     if(this->Pcurrentbasefunc)
         changebasefunction();
     this->prepare();
 }
+#undef COPY
 
 void OscilGen::add2XML(XMLwrapper *xml)
 {
-- 
2.47.0

