From 5a6134383b11211d38365dbd89fe914996a25f76 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sat, 7 Nov 2015 08:39:13 -0500
Subject: [PATCH 0696/1936] Restore PADnoteUI Default Tab

---
 src/UI/PADnoteUI.fl | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/UI/PADnoteUI.fl b/src/UI/PADnoteUI.fl
index b2d687e1..9472ab94 100644
--- a/src/UI/PADnoteUI.fl
+++ b/src/UI/PADnoteUI.fl
@@ -73,8 +73,7 @@ resui=new ResonanceUI(osc_i, location+"resonance/");
 initialized = false;
 applybutton = NULL;
 make_window();
-initialized = true;} {selected
-    }
+initialized = true;} {}
   }
   Function {make_window()} {open
   } {
@@ -85,6 +84,7 @@ initialized = true;} {selected
     } {
 
       Fl_Box dummy {
+        xywh {25 25 25 25}
         code0 {padnotewindow->init(osc_i, location);}
       }
       Fl_Tabs {} {
-- 
2.47.0

