From 80aa7435f120b8eca0157d725ec98e5e1813c736 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sat, 7 Nov 2015 09:55:09 -0500
Subject: [PATCH 0698/1936] CMake: More Explicitly State Found/Enabled Config

---
 src/CMakeLists.txt | 31 +++++++++++++++++++++++--------
 1 file changed, 23 insertions(+), 8 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 615997d1..720b6010 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -451,17 +451,32 @@ package_status(PKG_CONFIG_FOUND "PkgConfig" "found"   ${Red})
 package_status(ZLIB_FOUND       "zlib     " "found"   ${Red})
 package_status(MXML_FOUND       "mxml     " "found"   ${Red})
 package_status(FFTW_FOUND       "fftw3    " "found"   ${Red})
-package_status(OssEnable        "OSS      " "enabled" ${Yellow})
+package_status(LIBLO_FOUND      "liblo    " "found"   ${Red})
+package_status(X11_FOUND        "x11      " "found"   ${Yellow})
+package_status(HAS_XPM          "xpm      " "found"   ${Yellow})
+package_status(FLTK_FOUND       "fltk     " "found"   ${Yellow})
+package_status(NTK_FOUND        "ntk      " "found"   ${Yellow})
+package_status(X11_FOUND        "x11      " "found"   ${Yellow})
+package_status(OSS_FOUND        "OSS      " "found"   ${Yellow})
+package_status(ALSA_FOUND       "ALSA     " "found"   ${Yellow})
+package_status(JACK_FOUND       "JACK     " "found"   ${Yellow})
+package_status(PORTAUDIO_FOUND  "PA       " "found"   ${Yellow})
+package_status(LASH_FOUND       "Lash     " "found"   ${Yellow})
+package_status(DSSI_FOUND       "DSSI     " "found"   ${Yellow})
+package_status(FLTK_FOUND       "fltk     " "found"   ${Yellow})
+package_status(NTK_FOUND        "ntk      " "found"   ${Yellow})
+package_status(X11_FOUND        "x11      " "found"   ${Yellow})
+package_status(CXXTEST_FOUND    "CxxTest  " "found"   ${Yellow})
+package_status(LashEnable       "Lash     " "enabled" ${Yellow})
+package_status(DssiEnable       "DSSI     " "enabled" ${Yellow})
+package_status(LibloEnable      "liblo    " "enabled" ${Red})
+package_status(CompileTests     "tests    " "enabled" ${Yellow})
 package_status(AlsaEnable       "ALSA     " "enabled" ${Yellow})
 package_status(JackEnable       "JACK     " "enabled" ${Yellow})
+package_status(OssEnable        "OSS      " "enabled" ${Yellow})
 package_status(PaEnable         "PA       " "enabled" ${Yellow})
-package_status(CXXTEST_FOUND    "CxxTest  " "found"   ${Yellow})
 #TODO GUI MODULE
-package_status(LASH_FOUND       "Lash     " "enabled" ${Yellow})
-package_status(DSSI_FOUND       "DSSI     " "enabled" ${Yellow})
-package_status(LIBLO_FOUND      "liblo    " "enabled" ${Red})
-package_status(FLTK_FOUND       "fltk     " "found"   ${Yellow})
-package_status(NTK_FOUND        "ntk      " "found"   ${Yellow})
-package_status(X11_FOUND        "x11      " "found"   ${Yellow})
 package_status(HAVE_ASYNC       "c++ async" "usable"  ${Yellow})
+
+
 message(STATUS "Link libraries: ${ZLIB_LIBRARY} ${FFTW_LIBRARY} ${MXML_LIBRARIES} ${AUDIO_LIBRARIES} ${OS_LIBRARIES}")
-- 
2.47.0

