From 5b4172bc7725d55ff9ab51c748bc4479a048f4a6 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sat, 7 Nov 2015 13:54:15 -0500
Subject: [PATCH 0703/1936] Remove Debug Printf

---
 src/UI/PartUI.fl | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/UI/PartUI.fl b/src/UI/PartUI.fl
index 20ac7795..4bd33c7b 100644
--- a/src/UI/PartUI.fl
+++ b/src/UI/PartUI.fl
@@ -91,7 +91,6 @@ end();} {}
     widget->ext      = next;
     osc->renameLink("/"+old, "/"+next, widget);
     osc->requestValue("/"+next);
-    printf("changing <%s> -> <%s>\\n", old.c_str(), next.c_str());
     npart = part;
     } {}
   }
-- 
2.47.0

