From 46d4187bbdebd10b047d5322e8f59eb1864261a8 Mon Sep 17 00:00:00 2001
From: "Christopher A. Oliver" <caowasteland@gmail.com>
Date: Mon, 9 Nov 2015 01:02:36 -0500
Subject: [PATCH 0712/1936] Add shortcut button from ADvoice to parent ADnote
 globals.

---
 src/UI/ADnoteUI.fl | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/src/UI/ADnoteUI.fl b/src/UI/ADnoteUI.fl
index 650851b9..e2bb7bc5 100644
--- a/src/UI/ADnoteUI.fl
+++ b/src/UI/ADnoteUI.fl
@@ -1183,7 +1183,7 @@ resui->resonancewindow->show();}
       Fl_Button {} {
         label {Close Window}
         callback {ADnoteVoice->hide();}
-        xywh {305 601 195 25} box THIN_UP_BOX labelfont 1
+        xywh {468 601 195 25} box THIN_UP_BOX labelfont 1
       }
       Fl_Counter currentvoicecounter {
         label {Current Voice}
@@ -1202,6 +1202,12 @@ advoice->change_voice(nvoice);}
         callback {presetsui->paste(ADnoteVoice->loc(),this,nvoice);}
         xywh {735 609 25 15} box THIN_UP_BOX color 179 labelfont 1 labelsize 11 labelcolor 55
       }
+      Fl_Button {} {
+        label {Edit ADsynth}
+        callback {ADnoteGlobalParameters->show();}
+        xywh {280 601 105 25} labelfont 1 labelsize 12
+      }
+
     }
     Fl_Window ADnoteVoiceList {
       label {ADsynth Voices list} open
-- 
2.47.0

