From 24a9ed108b5651ceb826b71a8a2030353a06df22 Mon Sep 17 00:00:00 2001
From: "Christopher A. Oliver" <caowasteland@gmail.com>
Date: Mon, 9 Nov 2015 01:25:53 -0500
Subject: [PATCH 0713/1936] Similar shortcut button for advoice list

---
 src/UI/ADnoteUI.fl | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/UI/ADnoteUI.fl b/src/UI/ADnoteUI.fl
index e2bb7bc5..7b414135 100644
--- a/src/UI/ADnoteUI.fl
+++ b/src/UI/ADnoteUI.fl
@@ -1255,6 +1255,11 @@ advoice->change_voice(nvoice);}
           code0 {o->begin();for (int i=0;i<NUM_VOICES;i++){voicelistitem[i]=new ADvoicelistitem(0,0,670,25,"");voicelistitem[i]->init(i,loc+"VoicePar"+to_s(i)+"/",osc);}o->end();}
         } {}
       }
+      Fl_Button {} {
+        label {Edit ADsynth}
+        callback {ui->partui->adnoteui->ADnoteGlobalParameters->show();}
+        xywh {5 237 106 20} labelsize 12
+      }
     }
   }
   Function {ADnoteUI(std::string loc_, Fl_Osc_Interface *osc_)} {open
-- 
2.47.0

