From 1463ff06b9df8cdc8a68e00496fd59231355465a Mon Sep 17 00:00:00 2001
From: "Christopher A. Oliver" <caowasteland@gmail.com>
Date: Sun, 15 Nov 2015 22:50:44 -0500
Subject: [PATCH 0726/1936] A one line fix for Godfrey's kvetch of November 15,
 2015.

---
 src/UI/PartUI.fl | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/UI/PartUI.fl b/src/UI/PartUI.fl
index 4bd33c7b..5980dfc5 100644
--- a/src/UI/PartUI.fl
+++ b/src/UI/PartUI.fl
@@ -531,7 +531,9 @@ else {bendrng->oscMove("pitchwheel.bendrange");}}
       }
       Fl_Button {} {
         label {Reset all controllers}
-        callback {o->oscWrite("defaults");//part->SetController(C_resetallcontrollers,0);}
+        callback {o->oscWrite("defaults");//part->SetController(C_resetallcontrollers,0);
+	ctlwindow->update();
+	}
         xywh {5 110 210 20} box THIN_UP_BOX
         class Fl_Osc_Button
       }
-- 
2.47.0

