From 4961a4500bfe7800c8f37814a84a496edf4c18ba Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Fri, 13 Nov 2015 21:04:56 -0500
Subject: [PATCH 0731/1936] Fix Plugin Test

---
 src/Tests/guitar-adnote.xmz | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/Tests/guitar-adnote.xmz b/src/Tests/guitar-adnote.xmz
index 864b31a3..dc34c334 100644
--- a/src/Tests/guitar-adnote.xmz
+++ b/src/Tests/guitar-adnote.xmz
@@ -2,7 +2,7 @@
 <?xml version="1.0f" encoding="UTF-8"?>
 <!DOCTYPE ZynAddSubFX-data>
 <ZynAddSubFX-data version-major="2" version-minor="5"
-version-revision="1" ZynAddSubFX-author="Nasca Octavian Paul">
+version-revision="2" ZynAddSubFX-author="Nasca Octavian Paul">
 <INFORMATION>
 <par_bool name="PADsynth_used" value="yes" />
 </INFORMATION>
@@ -301,6 +301,8 @@ version-revision="1" ZynAddSubFX-author="Nasca Octavian Paul">
 </FREQUENCY_LFO>
 </FREQUENCY_PARAMETERS>
 <FILTER_PARAMETERS>
+<par name="velocity_sensing_amplitude" value="0" />
+<par name="velocity_sensing" value="64" />
 <FILTER>
 <par name="category" value="0" />
 <par name="type" value="2" />
-- 
2.47.0

