From f59857c8457e57ae919db330073ba2a1ffc5303e Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Tue, 17 Nov 2015 15:35:34 -0500
Subject: [PATCH 0735/1936] NioUI: Fix Dropdown Undefined Behavior

---
 src/Misc/Util.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Misc/Util.cpp b/src/Misc/Util.cpp
index c3f43fdc..269449f1 100644
--- a/src/Misc/Util.cpp
+++ b/src/Misc/Util.cpp
@@ -219,7 +219,7 @@ char *rtosc_splat(const char *path, std::set<std::string> v)
     char argT[v.size()+1];
     rtosc_arg_t arg[v.size()];
     unsigned i=0;
-    for(auto vv : v) {
+    for(auto &vv : v) {
         argT[i]  = 's';
         arg[i].s = vv.c_str();
         i++;
-- 
2.47.0

