From c021b468e807e76fc329b163c953fe20e77e60ec Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sat, 21 Nov 2015 10:21:30 -0500
Subject: [PATCH 0743/1936] Ext-gui: Allow Larger Messages Through Ringbuffer

---
 src/UI/guimain.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/UI/guimain.cpp b/src/UI/guimain.cpp
index 5cbaaace..36cb9600 100644
--- a/src/UI/guimain.cpp
+++ b/src/UI/guimain.cpp
@@ -521,7 +521,7 @@ Fl_Osc_Interface *GUI::genOscInterface(MiddleWare *)
     return new UI_Interface();
 }
 
-rtosc::ThreadLink lo_buffer(4096, 1000);
+rtosc::ThreadLink lo_buffer(4096*2, 1000);
 
 static void liblo_error_cb(int i, const char *m, const char *loc)
 {
-- 
2.47.0

