From 030d68cd4416bde3ffe9ab43dd9f033634e070ce Mon Sep 17 00:00:00 2001
From: "Christopher A. Oliver" <caowasteland@gmail.com>
Date: Fri, 27 Nov 2015 20:40:52 -0500
Subject: [PATCH 0763/1936] Add missing test parameter

---
 src/Tests/guitar-adnote.xmz | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/Tests/guitar-adnote.xmz b/src/Tests/guitar-adnote.xmz
index dc34c334..86c2edd0 100644
--- a/src/Tests/guitar-adnote.xmz
+++ b/src/Tests/guitar-adnote.xmz
@@ -268,6 +268,7 @@ version-revision="2" ZynAddSubFX-author="Nasca Octavian Paul">
 <FREQUENCY_PARAMETERS>
 <par_bool name="fixed_freq" value="no" />
 <par name="fixed_freq_et" value="0" />
+<par name="bend_adjust" value="88" />
 <par name="detune" value="12040" />
 <par name="coarse_detune" value="0" />
 <par name="detune_type" value="0" />
@@ -367,6 +368,7 @@ version-revision="2" ZynAddSubFX-author="Nasca Octavian Paul">
 <par name="coarse_detune" value="2048" />
 <par name="detune_type" value="2" />
 <par_bool name="freq_envelope_enabled" value="yes" />
+<par_bool name="fixed_freq" value="no" />
 <FREQUENCY_ENVELOPE>
 <par_bool name="free_mode" value="no" />
 <par name="env_points" value="3" />
@@ -485,6 +487,7 @@ version-revision="2" ZynAddSubFX-author="Nasca Octavian Paul">
 <FREQUENCY_PARAMETERS>
 <par_bool name="fixed_freq" value="no" />
 <par name="fixed_freq_et" value="0" />
+<par name="bend_adjust" value="88" />
 <par name="detune" value="7902" />
 <par name="coarse_detune" value="0" />
 <par name="detune_type" value="0" />
@@ -568,6 +571,7 @@ version-revision="2" ZynAddSubFX-author="Nasca Octavian Paul">
 <FREQUENCY_PARAMETERS>
 <par_bool name="fixed_freq" value="no" />
 <par name="fixed_freq_et" value="0" />
+<par name="bend_adjust" value="88" />
 <par name="detune" value="8192" />
 <par name="coarse_detune" value="0" />
 <par name="detune_type" value="0" />
@@ -674,6 +678,7 @@ version-revision="2" ZynAddSubFX-author="Nasca Octavian Paul">
 <FREQUENCY_PARAMETERS>
 <par_bool name="fixed_freq" value="no" />
 <par name="fixed_freq_et" value="0" />
+<par name="bend_adjust" value="88" />
 <par name="detune" value="8192" />
 <par name="coarse_detune" value="0" />
 <par name="detune_type" value="0" />
@@ -997,6 +1002,7 @@ version-revision="2" ZynAddSubFX-author="Nasca Octavian Paul">
 <FREQUENCY_PARAMETERS>
 <par_bool name="fixed_freq" value="no" />
 <par name="fixed_freq_et" value="103" />
+<par name="bend_adjust" value="88" />
 <par name="detune" value="8192" />
 <par name="coarse_detune" value="0" />
 <par name="overtone_spread_type" value="0" />
@@ -3356,6 +3362,7 @@ version-revision="2" ZynAddSubFX-author="Nasca Octavian Paul">
 <FREQUENCY_PARAMETERS>
 <par name="fixed_freq" value="0" />
 <par name="fixed_freq_et" value="0" />
+<par name="bend_adjust" value="88" />
 <par name="detune" value="8192" />
 <par name="coarse_detune" value="0" />
 <par name="detune_type" value="1" />
-- 
2.47.0

