From e9b7af5586d9b0294ec9c07c79fc8989b58b0d50 Mon Sep 17 00:00:00 2001
From: Olivier Jolly <olivier@pcedev.com>
Date: Thu, 3 Dec 2015 23:30:54 +0100
Subject: [PATCH 0780/1936] Avoid recomputing time independant LFO variables

---
 src/Synth/LFO.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/Synth/LFO.cpp b/src/Synth/LFO.cpp
index aea7354f..b0d69104 100644
--- a/src/Synth/LFO.cpp
+++ b/src/Synth/LFO.cpp
@@ -124,6 +124,7 @@ float LFO::baseOut(const char waveShape, const float phase)
 float LFO::lfoout()
 {
     //update internals XXX TODO cleanup
+    if ( ! lfopars_.time || lfopars_.last_update_timestamp == lfopars_.time->time())
     {
         waveShape = lfopars_.PLFOtype;
         int stretch = lfopars_.Pstretch;
-- 
2.47.0

