From 6a4664c278ed78044db893ca03342e7a0f62820a Mon Sep 17 00:00:00 2001
From: falkTX <falktx@gmail.com>
Date: Sun, 13 Dec 2015 19:54:19 +0100
Subject: [PATCH 0793/1936] Fix host parameter count

---
 src/Plugin/AbstractFX.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Plugin/AbstractFX.hpp b/src/Plugin/AbstractFX.hpp
index 4d2e9af5..d82ca824 100644
--- a/src/Plugin/AbstractFX.hpp
+++ b/src/Plugin/AbstractFX.hpp
@@ -38,7 +38,7 @@ class AbstractPluginFX : public Plugin
 {
 public:
     AbstractPluginFX(const uint32_t params, const uint32_t programs)
-        : Plugin(params, programs, 0),
+        : Plugin(params-2, programs, 0),
           paramCount(params-2), // volume and pan handled by host
           programCount(programs),
           bufferSize(getBufferSize()),
-- 
2.47.0

