From 2ceaf534fe8149e350265a944cfacddeee5ed0ac Mon Sep 17 00:00:00 2001
From: falkTX <falktx@gmail.com>
Date: Mon, 14 Dec 2015 08:28:14 +0100
Subject: [PATCH 0794/1936] Remove useless comment

---
 src/Plugin/AbstractFX.hpp | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/Plugin/AbstractFX.hpp b/src/Plugin/AbstractFX.hpp
index d82ca824..2fbc5dd0 100644
--- a/src/Plugin/AbstractFX.hpp
+++ b/src/Plugin/AbstractFX.hpp
@@ -136,7 +136,6 @@ protected:
    /**
       Load a program.
       The host may call this function from any context, including realtime processing.
-      Must be implemented by your plugin class only if DISTRHO_PLUGIN_WANT_PROGRAMS is enabled.
     */
     void loadProgram(uint32_t index) override
     {
-- 
2.47.0

