From 239bf4956a6c72ab1378c8afb92b9cdc13b7cdc1 Mon Sep 17 00:00:00 2001
From: Johannes Lorenz <johannes89@ist-einmalig.de>
Date: Sat, 2 Jan 2016 09:47:34 +0100
Subject: [PATCH 0804/1936] Fixed compile error for clang 3.5.0 in DSSI plugin.

---
 src/Output/DSSIaudiooutput.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/Output/DSSIaudiooutput.cpp b/src/Output/DSSIaudiooutput.cpp
index 6927bc38..2ad9b87a 100644
--- a/src/Output/DSSIaudiooutput.cpp
+++ b/src/Output/DSSIaudiooutput.cpp
@@ -612,7 +612,7 @@ DSSIaudiooutput *DSSIaudiooutput::getInstance(LADSPA_Handle instance)
  * @param sampleRate [in] the sample rate to be used by the synth.
  * @return
  */
-DSSIaudiooutput::DSSIaudiooutput(unsigned long sampleRate) : dssi_control({dssi_control_description[0],
+DSSIaudiooutput::DSSIaudiooutput(unsigned long sampleRate) : dssi_control{dssi_control_description[0],
                                                                            dssi_control_description[1],
                                                                            dssi_control_description[2],
                                                                            dssi_control_description[3],
@@ -623,7 +623,7 @@ DSSIaudiooutput::DSSIaudiooutput(unsigned long sampleRate) : dssi_control({dssi_
                                                                            dssi_control_description[8],
                                                                            dssi_control_description[9],
                                                                            dssi_control_description[10],
-                                                                           dssi_control_description[11]})
+                                                                           dssi_control_description[11]}
 {
     SYNTH_T synth;
     synth.samplerate = sampleRate;
-- 
2.47.0

