From ec54033ad59833e1c037f87ebabe995c7eed71e7 Mon Sep 17 00:00:00 2001
From: "Christopher A. Oliver" <caowasteland@gmail.com>
Date: Sat, 2 Jan 2016 15:19:20 -0500
Subject: [PATCH 0805/1936] Round envelope segment lengths up to buffer size
 multiple.

---
 src/Synth/Envelope.cpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/Synth/Envelope.cpp b/src/Synth/Envelope.cpp
index 4199e495..6b3152c4 100644
--- a/src/Synth/Envelope.cpp
+++ b/src/Synth/Envelope.cpp
@@ -45,10 +45,11 @@ Envelope::Envelope(EnvelopeParams &pars, float basefreq, float bufferdt)
     if((mode == 2) && linearenvelope)
         mode = 1;                              //change to linear
 
-    for(int i = 0; i < MAX_ENVELOPE_POINTS; ++i) {
+    for(int i = 0; i < envpoints; ++i) {
         const float tmp = pars.getdt(i) / 1000.0f * envstretch;
         if(tmp > bufferdt)
-            envdt[i] = bufferdt / tmp;
+            envdt[i] =
+                i == envpoints ? bufferdt / tmp : 1 / ceil(tmp / bufferdt);
         else
             envdt[i] = 2.0f;  //any value larger than 1
 
-- 
2.47.0

