From a9fc7cdd5d9babeecd3fbc145c0a7ec3a86e8ff9 Mon Sep 17 00:00:00 2001
From: falkTX <falktx@gmail.com>
Date: Tue, 5 Jan 2016 06:01:58 +0000
Subject: [PATCH 0813/1936] Fix cmakefile

---
 src/Plugin/ZynAddSubFX/CMakeLists.txt | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/Plugin/ZynAddSubFX/CMakeLists.txt b/src/Plugin/ZynAddSubFX/CMakeLists.txt
index beff9014..8cb6b7ac 100644
--- a/src/Plugin/ZynAddSubFX/CMakeLists.txt
+++ b/src/Plugin/ZynAddSubFX/CMakeLists.txt
@@ -22,7 +22,7 @@ add_library(ZynAddSubFX_vst SHARED
     ZynAddSubFX.cpp
     ZynAddSubFX-UI.cpp)
 
-elseif(NtkGui)
+else()
 
 # UI Disabled
 add_library(ZynAddSubFX_lv2 SHARED
@@ -37,7 +37,7 @@ add_library(ZynAddSubFX_vst SHARED
     ${CMAKE_SOURCE_DIR}/DPF/distrho/DistrhoPluginMain.cpp
     ZynAddSubFX.cpp)
 
-endif(NtkGui)
+endif()
 
 set_target_properties(ZynAddSubFX_lv2 PROPERTIES COMPILE_DEFINITIONS "DISTRHO_PLUGIN_TARGET_LV2")
 set_target_properties(ZynAddSubFX_lv2 PROPERTIES LIBRARY_OUTPUT_DIRECTORY "lv2")
@@ -76,4 +76,4 @@ install(TARGETS ZynAddSubFX_lv2_ui LIBRARY DESTINATION lib/lv2/ZynAddSubFX.lv2/)
 install(FILES
         ${CMAKE_CURRENT_BINARY_DIR}/lv2/ZynAddSubFX_ui.ttl
         DESTINATION lib/lv2/ZynAddSubFX.lv2/)
-endif(NtkGui)
+endif()
-- 
2.47.0

