From a39dade5079c90e0a98ed980065df725a2e4d917 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sat, 16 Jan 2016 10:41:21 -0500
Subject: [PATCH 0824/1936] Middleware: Restore Midi Unlearn Functionality

---
 src/Misc/MiddleWare.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/Misc/MiddleWare.cpp b/src/Misc/MiddleWare.cpp
index 85318dc4..86df608b 100644
--- a/src/Misc/MiddleWare.cpp
+++ b/src/Misc/MiddleWare.cpp
@@ -1043,6 +1043,14 @@ static rtosc::Ports middwareSnoopPorts = {
         else
             midi.map(addr.c_str(), true);
         rEnd},
+    {"unlearn:s", 0, 0,
+        rBegin;
+        string addr = rtosc_argument(msg, 0).s;
+        auto &midi  = impl.midi_mapper;
+        auto map    = midi.getMidiMappingStrings();
+        midi.unMap(addr.c_str(), false);
+        midi.unMap(addr.c_str(), true);
+        rEnd},
     //drop this message into the abyss
     {"ui/title:", 0, 0, [](const char *msg, RtData &d) {}}
 };
-- 
2.47.0

