From b97f65d84b212879d5a4f9fc154ff623ba4f9b4e Mon Sep 17 00:00:00 2001
From: "Christopher A. Oliver" <caowasteland@gmail.com>
Date: Sat, 6 Feb 2016 19:10:50 -0500
Subject: [PATCH 0846/1936] Fix revision mismatch failure for plugin test

---
 src/Tests/guitar-adnote.xmz | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Tests/guitar-adnote.xmz b/src/Tests/guitar-adnote.xmz
index 20092530..ddb63d6c 100644
--- a/src/Tests/guitar-adnote.xmz
+++ b/src/Tests/guitar-adnote.xmz
@@ -2,7 +2,7 @@
 <?xml version="1.0f" encoding="UTF-8"?>
 <!DOCTYPE ZynAddSubFX-data>
 <ZynAddSubFX-data version-major="2" version-minor="5"
-version-revision="2" ZynAddSubFX-author="Nasca Octavian Paul">
+version-revision="3" ZynAddSubFX-author="Nasca Octavian Paul">
 <INFORMATION>
 <par_bool name="PADsynth_used" value="yes" />
 </INFORMATION>
-- 
2.47.0

