From 86d1fcde1c9a125fc0f59609390c86bff8f315fe Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Mon, 8 Feb 2016 09:37:13 -0500
Subject: [PATCH 0847/1936] Update Copyright Year

---
 src/main.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/main.cpp b/src/main.cpp
index b8fb8045..07116f80 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -136,7 +136,7 @@ int main(int argc, char *argv[])
     << "\nZynAddSubFX - Copyright (c) 2002-2013 Nasca Octavian Paul and others"
     << endl;
     cerr
-    << "                Copyright (c) 2009-2015 Mark McCurry [active maintainer]"
+    << "                Copyright (c) 2009-2016 Mark McCurry [active maintainer]"
     << endl;
     cerr << "Compiled: " << __DATE__ << " " << __TIME__ << endl;
     cerr << "This program is free software (GNU GPL v2 or later) and \n";
-- 
2.47.0

