From 169fe0c7424d252acd6ceff1d43a4c18de79e478 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Tue, 9 Feb 2016 10:21:33 -0500
Subject: [PATCH 0850/1936] CMake: Make DSSI Install Use PluginLibDir

---
 src/CMakeLists.txt | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index c7c0f5d2..365c129c 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -446,11 +446,7 @@ if (DssiEnable)
 		${OS_LIBRARIES}
         ${LIBLO_LIBRARIES}
 		)
-	if (${CMAKE_SIZEOF_VOID_P} EQUAL "8")
-		install(TARGETS zynaddsubfx_dssi LIBRARY DESTINATION lib64/dssi/)
-	else ()
-		install(TARGETS zynaddsubfx_dssi LIBRARY DESTINATION lib/dssi/)
-	endif ()
+    install(TARGETS zynaddsubfx_dssi LIBRARY DESTINATION ${PluginLibDir}/dssi/)
 endif()
 
 install(TARGETS zynaddsubfx
-- 
2.47.0

