From bda89cfdd70ef389e10871a12e8cd4149e306e2d Mon Sep 17 00:00:00 2001
From: Olivier Jolly <olivier@pcedev.com>
Date: Sat, 20 Feb 2016 23:31:04 +0100
Subject: [PATCH 0867/1936] Prevent unitialized use of SUBnote global filter
 sense

---
 src/Synth/SUBnote.cpp | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/src/Synth/SUBnote.cpp b/src/Synth/SUBnote.cpp
index 38bab514..2f0a2294 100644
--- a/src/Synth/SUBnote.cpp
+++ b/src/Synth/SUBnote.cpp
@@ -211,10 +211,6 @@ void SUBnote::setup(float freq,
             GlobalFilter->updateNoteFreq(basefreq);
     }
 
-    if(GlobalFilter)
-        GlobalFilter->updateSense(velocity, pars.PGlobalFilterVelocityScale,
-                                     pars.PGlobalFilterVelocityScaleFunction);
-
     oldamplitude = newamplitude;
 }
 
@@ -391,6 +387,9 @@ void SUBnote::initparameters(float freq)
 
         GlobalFilter = memory.alloc<ModFilter>(*pars.GlobalFilter, synth, time, memory, stereo, freq);
 
+        GlobalFilter->updateSense(velocity, pars.PGlobalFilterVelocityScale,
+                                  pars.PGlobalFilterVelocityScaleFunction);
+
         GlobalFilter->addMod(*GlobalFilterEnvelope);
     }
     computecurrentparameters();
-- 
2.47.0

