From 2e1d8a32325dc273a9099d3870fa1c194629ca1e Mon Sep 17 00:00:00 2001
From: "Christopher A. Oliver" <caowasteland@gmail.com>
Date: Sun, 21 Feb 2016 19:38:04 -0500
Subject: [PATCH 0868/1936] Have -N affect ALSA MIDI port name.

---
 src/Nio/AlsaEngine.cpp | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/src/Nio/AlsaEngine.cpp b/src/Nio/AlsaEngine.cpp
index a34357d3..04f4d873 100644
--- a/src/Nio/AlsaEngine.cpp
+++ b/src/Nio/AlsaEngine.cpp
@@ -28,6 +28,7 @@ using namespace std;
 #include "../Misc/Config.h"
 #include "InMgr.h"
 #include "AlsaEngine.h"
+#include "Nio.h"
 
 AlsaEngine::AlsaEngine(const SYNTH_T &synth)
     :AudioOut(synth)
@@ -212,7 +213,13 @@ bool AlsaEngine::openMidi()
     if(snd_seq_open(&midi.handle, "default", SND_SEQ_OPEN_INPUT, 0) != 0)
         return false;
 
-    snd_seq_set_client_name(midi.handle, "ZynAddSubFX");
+    string clientname = "ZynAddSubFX";
+    string postfix = Nio::getPostfix();
+    if (!postfix.empty())
+        clientname += "_" + postfix;
+    if(Nio::pidInClientName)
+        clientname += "_" + os_pid_as_padded_string();
+    snd_seq_set_client_name(midi.handle, clientname.c_str());
 
     alsaport = snd_seq_create_simple_port(
         midi.handle,
-- 
2.47.0

