From c5fd645de56f8772b7158394ae55c86dd5325a34 Mon Sep 17 00:00:00 2001
From: falkTX <falktx@gmail.com>
Date: Tue, 23 Feb 2016 09:59:10 +0100
Subject: [PATCH 0869/1936] Update DPF, fixes VST plugin state

---
 DPF | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/DPF b/DPF
index d1e12728..a82e260a 160000
--- a/DPF
+++ b/DPF
@@ -1 +1 @@
-Subproject commit d1e127289c4f3f51d34177ed3adb67f2e75925fe
+Subproject commit a82e260a1b88b031781b05e4554cd16fc37d342c
-- 
2.47.0

