From c90b4a6d6ff76a5e20f6a9d204f6af451cf14c09 Mon Sep 17 00:00:00 2001
From: falkTX <falktx@gmail.com>
Date: Thu, 17 Mar 2016 12:28:34 +0100
Subject: [PATCH 0890/1936] Build and install LV2 UI when using FLTK

---
 src/Plugin/ZynAddSubFX/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Plugin/ZynAddSubFX/CMakeLists.txt b/src/Plugin/ZynAddSubFX/CMakeLists.txt
index 9ad401b6..bc8e905f 100644
--- a/src/Plugin/ZynAddSubFX/CMakeLists.txt
+++ b/src/Plugin/ZynAddSubFX/CMakeLists.txt
@@ -85,7 +85,7 @@ install(FILES
     ${CMAKE_CURRENT_BINARY_DIR}/lv2/ZynAddSubFX.ttl
     DESTINATION ${PluginLibDir}/lv2/ZynAddSubFX.lv2/)
 
-if(NtkGui)
+if(NtkGui OR FltkGui)
 set_target_properties(ZynAddSubFX_lv2_ui PROPERTIES COMPILE_DEFINITIONS "DISTRHO_PLUGIN_TARGET_LV2")
 set_target_properties(ZynAddSubFX_lv2_ui PROPERTIES LIBRARY_OUTPUT_DIRECTORY "lv2")
 set_target_properties(ZynAddSubFX_lv2_ui PROPERTIES OUTPUT_NAME "ZynAddSubFX_ui")
-- 
2.47.0

