From 2b90e54bd5a3186f5fbef4a5796aeb917b97b3de Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Mon, 9 May 2016 16:17:06 -0400
Subject: [PATCH 0899/1936] FilterParams: Add Short Parameter Names

---
 src/Params/FilterParams.cpp | 31 +++++++++++++++++++------------
 1 file changed, 19 insertions(+), 12 deletions(-)

diff --git a/src/Params/FilterParams.cpp b/src/Params/FilterParams.cpp
index b2ac7cea..00910472 100644
--- a/src/Params/FilterParams.cpp
+++ b/src/Params/FilterParams.cpp
@@ -58,18 +58,25 @@ const rtosc::Ports FilterParams::ports = {
     rSelf(FilterParams),
     rPaste,
     rArrayPaste,
-    rParamZyn(Pcategory,   "Class of filter"),
-    rParamZyn(Ptype,       "Filter Type"),
-    rParamZyn(Pfreq,        "Center Freq"),
-    rParamZyn(Pq,           "Quality Factor (resonance/bandwidth)"),
-    rParamZyn(Pstages,      "Filter Stages + 1"),
-    rParamZyn(Pfreqtrack,   "Frequency Tracking amount"),
-    rParamZyn(Pgain,        "Output Gain"),
-    rParamZyn(Pnumformants, "Number of formants to be used"),
-    rParamZyn(Pformantslowness, "Rate that formants change"),
-    rParamZyn(Pvowelclearness, "Cost for mixing vowels"),
-    rParamZyn(Pcenterfreq,     "Center Freq (formant)"),
-    rParamZyn(Poctavesfreq,    "Number of octaves for formant"),
+    rParamZyn(Pcategory,        rShort("class"),    "Class of filter"),
+    rParamZyn(Ptype,            rShort("type"),     "Filter Type"),
+    rParamZyn(Pfreq,            rShort("cutoff"),   "Center Freq"),
+    rParamZyn(Pq,               rShort("q"),
+            "Quality Factor (resonance/bandwidth)"),
+    rParamZyn(Pstages,          rShort("stages"),   "Filter Stages + 1"),
+    rParamZyn(Pfreqtrack,       rShort("f.track"),
+            "Frequency Tracking amount"),
+    rParamZyn(Pgain,            rShort("gain"),     "Output Gain"),
+    rParamZyn(Pnumformants,     rShort("formants"),
+            "Number of formants to be used"),
+    rParamZyn(Pformantslowness, rShort("slew"),
+            "Rate that formants change"),
+    rParamZyn(Pvowelclearness,  rShort("clarity"),
+            "Cost for mixing vowels"),
+    rParamZyn(Pcenterfreq,      rShort("cutoff"),
+            "Center Freq (formant)"),
+    rParamZyn(Poctavesfreq,     rShort("octaves"),
+            "Number of octaves for formant"),
 
     //TODO check if FF_MAX_SEQUENCE is acutally expanded or not
     rParamZyn(Psequencesize, rMap(max, FF_MAX_SEQUENCE), "Length of vowel sequence"),
-- 
2.47.0

