From e2e35792393492b7141ad2109f2434d13b706812 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Mon, 23 May 2016 15:57:42 -0400
Subject: [PATCH 0915/1936] Watchpoint: Reset Deactivate Flag After Use

---
 src/Synth/WatchPoint.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/Synth/WatchPoint.cpp b/src/Synth/WatchPoint.cpp
index 97febfb2..5064c4b1 100644
--- a/src/Synth/WatchPoint.cpp
+++ b/src/Synth/WatchPoint.cpp
@@ -113,6 +113,7 @@ void WatchManager::tick(void)
         if(deactivate[i]) {
             memset(active_list[i], 0, 128);
             sample_list[i] = 0;
+            deactivate[i]  = false;
         }
     }
 
-- 
2.47.0

