From b4d3d59855328dbd4e853063e383fc4838891bb8 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Mon, 23 May 2016 15:58:30 -0400
Subject: [PATCH 0916/1936] EnvelopeParams: Add Short Names

---
 src/Params/EnvelopeParams.cpp | 23 +++++++++++++----------
 1 file changed, 13 insertions(+), 10 deletions(-)

diff --git a/src/Params/EnvelopeParams.cpp b/src/Params/EnvelopeParams.cpp
index c822f81a..37dc8cc3 100644
--- a/src/Params/EnvelopeParams.cpp
+++ b/src/Params/EnvelopeParams.cpp
@@ -37,16 +37,19 @@ static const rtosc::Ports localPorts = {
     rParamZyn(Penvsustain, rProp(internal), "Location of the sustain point"),
     rParams(Penvdt,  MAX_ENVELOPE_POINTS, "Envelope Delay Times"),
     rParams(Penvval, MAX_ENVELOPE_POINTS, "Envelope Values"),
-    rParamZyn(Penvstretch, "Stretch with respect to frequency"),
-    rToggle(Pforcedrelease, "Force Envelope to fully evaluate"),
-    rToggle(Plinearenvelope, "Linear or Logarithmic Envelopes"),
-    rParamZyn(PA_dt,  "Attack Time"),
-    rParamZyn(PA_val, "Attack Value"),
-    rParamZyn(PD_dt,  "Decay Time"),
-    rParamZyn(PD_val, "Decay Value"),
-    rParamZyn(PS_val, "Sustain Value"),
-    rParamZyn(PR_dt,  "Release Time"),
-    rParamZyn(PR_val, "Release Value"),
+    rParamZyn(Penvstretch,  rShort("stretch"),
+            "Stretch with respect to frequency"),
+    rToggle(Pforcedrelease, rShort("frcr"),
+            "Force Envelope to fully evaluate"),
+    rToggle(Plinearenvelope, rShort("lin/log"),
+            "Linear or Logarithmic Envelopes"),
+    rParamZyn(PA_dt,  rShort("a.dt"),  "Attack Time"),
+    rParamZyn(PA_val, rShort("a.val"), "Attack Value"),
+    rParamZyn(PD_dt,  rShort("d.dt"),  "Decay Time"),
+    rParamZyn(PD_val, rShort("d.val"), "Decay Value"),
+    rParamZyn(PS_val, rShort("s.val"), "Sustain Value"),
+    rParamZyn(PR_dt,  rShort("r.dt"),  "Release Time"),
+    rParamZyn(PR_val, rShort("r.val"), "Release Value"),
 
     {"addPoint:i", rProp(internal) rDoc("Add point to envelope"), NULL, [](const char *msg, RtData &d)
         {
-- 
2.47.0

