From 0247483f8e40bbc1742494ab91c3564d3ffa9f7b Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Thu, 9 Jun 2016 13:38:32 -0400
Subject: [PATCH 0923/1936] FilterParams: Add Response For Single Pole Filters

---
 src/Params/FilterParams.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/Params/FilterParams.cpp b/src/Params/FilterParams.cpp
index 0e141714..ba4b5379 100644
--- a/src/Params/FilterParams.cpp
+++ b/src/Params/FilterParams.cpp
@@ -135,6 +135,11 @@ const rtosc::Ports FilterParams::ports = {
                         (float)obj->Pstages,
                         cf.c[0], cf.c[1], cf.c[2],
                         0.0,     cf.d[1], cf.d[2]);
+            } else if(order == 1) {
+                d.reply(d.loc, "fffff",
+                        (float)obj->Pstages,
+                        cf.c[0], cf.c[1],
+                        0.0,     cf.d[1]);
             }
         }},
     //    "", NULL, [](){}},"/freq"
-- 
2.47.0

