From b7d086afcf0680535dd60b4ada3645b0aadb85c2 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Tue, 28 Jun 2016 17:43:59 -0400
Subject: [PATCH 0936/1936] ADnoteParameters: Add PDetuneType Metadata

---
 src/Params/ADnoteParameters.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/Params/ADnoteParameters.cpp b/src/Params/ADnoteParameters.cpp
index 4defe71b..8f84820d 100644
--- a/src/Params/ADnoteParameters.cpp
+++ b/src/Params/ADnoteParameters.cpp
@@ -241,7 +241,9 @@ static const Ports globalPorts = {
     //Frequency
     rParamI(PDetune,         rShort("fine"),   "Fine Detune"),
     rParamI(PCoarseDetune,   rShort("coarse"), "Coarse Detune"),
-    rParamZyn(PDetuneType,   rShort("type"),   "Detune Scaling Type"),
+    rParamZyn(PDetuneType,   rShort("type"),
+            rOptions(L35cents, L10cents, E100cents, E1200cents),
+            "Detune Scaling Type"),
     rParamZyn(PBandwidth,    rShort("bw."),    "Relative Fine Detune Gain"),
 
     //Amplitude
-- 
2.47.0

