From 4b456eece4528c9b4fa0ab2c9bb9ed2e557bb128 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Wed, 13 Jul 2016 21:01:37 -0400
Subject: [PATCH 0943/1936] Controller: Add OSC Metadata

---
 src/Params/Controller.cpp | 42 +++++++++++++++++++--------------------
 1 file changed, 21 insertions(+), 21 deletions(-)

diff --git a/src/Params/Controller.cpp b/src/Params/Controller.cpp
index efbf7d17..985ce266 100644
--- a/src/Params/Controller.cpp
+++ b/src/Params/Controller.cpp
@@ -27,31 +27,31 @@ using namespace rtosc;
 #undef rChangeCb
 #define rChangeCb if (obj->time) { obj->last_update_timestamp = obj->time->time(); }
 const rtosc::Ports Controller::ports = {
-    rParamZyn(panning.depth, "Depth of Panning MIDI Control"),
-    rParamZyn(filtercutoff.depth, "Depth of Filter Cutoff MIDI Control"),
-    rParamZyn(filterq.depth, "Depth of Filter Q MIDI Control"),
-    rParamZyn(bandwidth.depth, "Depth of Bandwidth MIDI Control"),
-    rToggle(bandwidth.exponential, "Bandwidth Exponential Mode"),
-    rParamZyn(modwheel.depth, "Depth of Modwheel MIDI Control"),
-    rToggle(modwheel.exponential, "Modwheel Exponential Mode"),
-    rToggle(pitchwheel.is_split, "If PitchWheel Has unified bendrange or not"),
+    rParamZyn(panning.depth,       rShort("pan.d"), "Depth of Panning MIDI Control"),
+    rParamZyn(filtercutoff.depth,  rShort("fc.d"), "Depth of Filter Cutoff MIDI Control"),
+    rParamZyn(filterq.depth,       rShort("fq.d"), "Depth of Filter Q MIDI Control"),
+    rParamZyn(bandwidth.depth,     rShort("bw.d"), "Depth of Bandwidth MIDI Control"),
+    rToggle(bandwidth.exponential, rShort("bw.exp"), "Bandwidth Exponential Mode"),
+    rParamZyn(modwheel.depth,      rShort("mdw.d"), "Depth of Modwheel MIDI Control"),
+    rToggle(modwheel.exponential,  rShort("mdw.exp"), "Modwheel Exponential Mode"),
+    rToggle(pitchwheel.is_split,   "If PitchWheel Has unified bendrange or not"),
     rParamI(pitchwheel.bendrange, "Range of MIDI Pitch Wheel"),
     rParamI(pitchwheel.bendrange_down, "Lower Range of MIDI Pitch Wheel"),
-    rToggle(expression.receive, "Expression MIDI Receive"),
-    rToggle(fmamp.receive,      "FM amplitude MIDI Receive"),
-    rToggle(volume.receive,     "Volume MIDI Receive"),
-    rToggle(sustain.receive,    "Sustain MIDI Receive"),
-    rToggle(portamento.receive, "Portamento MIDI Receive"),
+    rToggle(expression.receive, rShort("exp.rcv"), "Expression MIDI Receive"),
+    rToggle(fmamp.receive,      rShort("fma.rcv"), "FM amplitude MIDI Receive"),
+    rToggle(volume.receive,     rShort("vol.rcv"), "Volume MIDI Receive"),
+    rToggle(sustain.receive,    rShort("sus.rcv"), "Sustain MIDI Receive"),
+    rToggle(portamento.receive, rShort("prt.rcv"), "Portamento MIDI Receive"),
     rToggle(portamento.portamento, "UNDOCUMENTED"),
-    rParamZyn(portamento.time, "Portamento Length"),
-    rToggle(portamento.proportional, "If all portamentos are proportional to the distance they span"),
-    rParamZyn(portamento.propRate, "Portamento proportional rate"),
-    rParamZyn(portamento.propDepth, "Portamento proportional depth"),
-    rParamZyn(portamento.pitchthresh, "Threshold for portamento"),
-    rToggle(portamento.pitchthreshtype, "Type of threshold"),
+    rParamZyn(portamento.time,          rShort("time"), "Portamento Length"),
+    rToggle(portamento.proportional,    rShort("propt."), "If all portamentos are proportional to the distance they span"),
+    rParamZyn(portamento.propRate,      rShort("rate"), "Portamento proportional rate"),
+    rParamZyn(portamento.propDepth,     rShort("depth"), "Portamento proportional depth"),
+    rParamZyn(portamento.pitchthresh,   rShort("thresh"), "Threshold for portamento"),
+    rToggle(portamento.pitchthreshtype, rShort("tr.type"), "Type of threshold"),
     rParamZyn(portamento.updowntimestretch, "UNDOCUMENTED"),
-    rParamZyn(resonancecenter.depth, "Resonance Center MIDI Depth"),
-    rParamZyn(resonancebandwidth.depth, "Resonance Bandwidth MIDI Depth"),
+    rParamZyn(resonancecenter.depth,    rShort("rfc.d"), "Resonance Center MIDI Depth"),
+    rParamZyn(resonancebandwidth.depth, rShort("rbw.d"), "Resonance Bandwidth MIDI Depth"),
     rToggle(NRPN.receive, "NRPN MIDI Enable"),
     rAction(defaults),
 };
-- 
2.47.0

