From 4c58e5764a42145bf0548e781079885caed76afc Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Tue, 19 Jul 2016 14:07:44 -0400
Subject: [PATCH 0955/1936] Add Synth: Add Range To PDetune

---
 src/Params/ADnoteParameters.cpp | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/Params/ADnoteParameters.cpp b/src/Params/ADnoteParameters.cpp
index 8f84820d..78bb507b 100644
--- a/src/Params/ADnoteParameters.cpp
+++ b/src/Params/ADnoteParameters.cpp
@@ -94,7 +94,9 @@ static const Ports voicePorts = {
     rParamZyn(PfixedfreqET,       rShort("e.t."),   "Equal Tempermant Parameter"),
     rParamZyn(PBendAdjust,        rShort("bend"),   "Pitch bend adjustment"),
     rParamZyn(POffsetHz,          rShort("offset"), "Voice constant offset"),
-    rParamI(PDetune,              rShort("fine"),   "Fine Detune"),
+    //nominally -8192..8191
+    rParamI(PDetune,              rShort("fine"),
+            rLinear(0, 16383), "Fine Detune"),
     rParamI(PCoarseDetune,        rShort("coarse"), "Coarse Detune"),
     rParamZyn(PDetuneType,        rShort("type"),   "Magnitude of Detune"),
     rToggle(PFreqEnvelopeEnabled, rShort("enable"), "Frequency Envelope Enable"),
@@ -239,7 +241,9 @@ static const Ports globalPorts = {
     rToggle(PStereo, rShort("stereo"), "Mono/Stereo Enable"),
 
     //Frequency
-    rParamI(PDetune,         rShort("fine"),   "Fine Detune"),
+    //nominally -8192..8191
+    rParamI(PDetune,              rShort("fine"),
+            rLinear(0, 16383), "Fine Detune"),
     rParamI(PCoarseDetune,   rShort("coarse"), "Coarse Detune"),
     rParamZyn(PDetuneType,   rShort("type"),
             rOptions(L35cents, L10cents, E100cents, E1200cents),
-- 
2.47.0

