From 57ea725f5d67d8c8de28db8220fad62dd47fcb0b Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Tue, 19 Jul 2016 14:08:46 -0400
Subject: [PATCH 0957/1936] Add More Microtonal Metadata

---
 src/Misc/Microtonal.cpp | 24 ++++++++++++------------
 1 file changed, 12 insertions(+), 12 deletions(-)

diff --git a/src/Misc/Microtonal.cpp b/src/Misc/Microtonal.cpp
index 6cd936a7..a7796dea 100644
--- a/src/Misc/Microtonal.cpp
+++ b/src/Misc/Microtonal.cpp
@@ -39,25 +39,25 @@ using namespace rtosc;
  * A good lookup table should be a good finalization of this
  */
 const rtosc::Ports Microtonal::ports = {
-    rToggle(Pinvertupdown, "key mapping inverse"),
-    rParamZyn(Pinvertupdowncenter, "center of the inversion"),
-    rToggle(Penabled, "Enable for microtonal mode"),
-    rParamZyn(PAnote, "The note for 'A'"),
-    rParamF(PAfreq, "Frequency of the 'A' note"),
-    rParamZyn(Pscaleshift, "UNDOCUMENTED"),
-    rParamZyn(Pfirstkey, "First key to retune"),
-    rParamZyn(Plastkey,  "Last key to retune"),
-    rParamZyn(Pmiddlenote, "Scale degree 0 note"),
+    rToggle(Pinvertupdown, rShort("inv."), "key mapping inverse"),
+    rParamZyn(Pinvertupdowncenter, rShort("center"), "center of the inversion"),
+    rToggle(Penabled, rShort("enable"), "Enable for microtonal mode"),
+    rParamZyn(PAnote, rShort("A note"), "The note for 'A'"),
+    rParamF(PAfreq,   rShort("A freq"), "Frequency of the 'A' note"),
+    rParamZyn(Pscaleshift, rShort("shift"), "UNDOCUMENTED"),
+    rParamZyn(Pfirstkey, rShort("first key"), "First key to retune"),
+    rParamZyn(Plastkey,  rShort("last key"), "Last key to retune"),
+    rParamZyn(Pmiddlenote, rShort("middle"), "Scale degree 0 note"),
 
     //TODO check to see if this should be exposed
     rParamZyn(Pmapsize, "Size of key map"),
     rToggle(Pmappingenabled, "Mapping Enable"),
 
     rParams(Pmapping, 128, "Mapping of keys"),
-    rParamZyn(Pglobalfinedetune, "Fine detune for all notes"),
+    rParamZyn(Pglobalfinedetune, rShort("fine"), "Fine detune for all notes"),
 
-    rString(Pname, MICROTONAL_MAX_NAME_LEN, "Microtonal Name"),
-    rString(Pcomment, MICROTONAL_MAX_NAME_LEN, "Microtonal Name"),
+    rString(Pname, MICROTONAL_MAX_NAME_LEN,    rShort("name"), "Microtonal Name"),
+    rString(Pcomment, MICROTONAL_MAX_NAME_LEN, rShort("comment") "Microtonal Name"),
 
     {"octavesize:", rDoc("Get octave size"), 0, [](const char*, RtData &d)
         {
-- 
2.47.0

