From 7f453c5f52b780d1f0c1105def09ee9789ed884a Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Thu, 21 Jul 2016 13:51:10 -0400
Subject: [PATCH 0964/1936] Add Sanity Check For Sending OSC

---
 src/Misc/MiddleWare.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/Misc/MiddleWare.cpp b/src/Misc/MiddleWare.cpp
index 21c8c015..13161a1c 100644
--- a/src/Misc/MiddleWare.cpp
+++ b/src/Misc/MiddleWare.cpp
@@ -1539,6 +1539,11 @@ void MiddleWareImpl::broadcastToRemote(const char *rtmsg)
 
 void MiddleWareImpl::sendToRemote(const char *rtmsg, std::string dest)
 {
+    if(!rtmsg || rtmsg[0] != '/' || !rtosc_message_length(rtmsg, -1)) {
+        printf("[Warning] Invalid message in sendToRemote <%s>...\n", rtmsg);
+        return;
+    }
+
     //printf("sendToRemote(%s:%s,%s)\n", rtmsg, rtosc_argument_string(rtmsg),
     //        dest.c_str());
     if(dest == "GUI") {
-- 
2.47.0

