From fc1cb5bd0ad3ee06ddd8c63c5759cff5f7b407fe Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Thu, 21 Jul 2016 21:07:37 -0400
Subject: [PATCH 0965/1936] Pad Synth: Add More Parameter Metadata

---
 src/Params/PADnoteParameters.cpp | 34 ++++++++++++++++----------------
 1 file changed, 17 insertions(+), 17 deletions(-)

diff --git a/src/Params/PADnoteParameters.cpp b/src/Params/PADnoteParameters.cpp
index 8b5257bd..ff30a338 100644
--- a/src/Params/PADnoteParameters.cpp
+++ b/src/Params/PADnoteParameters.cpp
@@ -40,31 +40,31 @@ static const rtosc::Ports realtime_ports =
     rRecurp(GlobalFilter, "Post Filter"),
 
     //Volume
-    rToggle(PStereo, "Stereo/Mono Mode"),
-    rParamZyn(PPanning, "Left Right Panning"),
-    rParamZyn(PVolume, "Synth Volume"),
-    rParamZyn(PAmpVelocityScaleFunction, "Amplitude Velocity Sensing function"),
+    rToggle(PStereo,    rShort("stereo"), "Stereo/Mono Mode"),
+    rParamZyn(PPanning, rShort("panning"), "Left Right Panning"),
+    rParamZyn(PVolume,  rShort("vol"), "Synth Volume"),
+    rParamZyn(PAmpVelocityScaleFunction, rShort("sense"), "Amplitude Velocity Sensing function"),
 
     rParamZyn(Fadein_adjustment, "Adjustment for anti-pop strategy."),
 
     //Punch
-    rParamZyn(PPunchStrength, "Punch Strength"),
-    rParamZyn(PPunchTime, "UNKNOWN"),
-    rParamZyn(PPunchStretch, "How Punch changes with note frequency"),
-    rParamZyn(PPunchVelocitySensing, "Punch Velocity control"),
+    rParamZyn(PPunchStrength, rShort("strength"), "Punch Strength"),
+    rParamZyn(PPunchTime,     rShort("time"),     "Length of punch"),
+    rParamZyn(PPunchStretch,  rShort("stretch"), "How Punch changes with note frequency"),
+    rParamZyn(PPunchVelocitySensing, rShort("sense"), "Punch Velocity control"),
 
     //Filter
-    rParamZyn(PFilterVelocityScale, "Filter Velocity Magnitude"),
-    rParamZyn(PFilterVelocityScaleFunction, "Filter Velocity Function Shape"),
+    rParamZyn(PFilterVelocityScale,         rShort("scale"), "Filter Velocity Magnitude"),
+    rParamZyn(PFilterVelocityScaleFunction, rShort("sense"), "Filter Velocity Function Shape"),
 
     //Freq
-    rToggle(Pfixedfreq, "Base frequency fixed frequency enable"),
-    rParamZyn(PfixedfreqET, "Equal temeperate control for fixed frequency operation"),
+    rToggle(Pfixedfreq,     rShort("fixed"),  "Base frequency fixed frequency enable"),
+    rParamZyn(PfixedfreqET, rShort("f.ET"),   "Equal temeperate control for fixed frequency operation"),
     rParamZyn(PBendAdjust,          "Pitch bend adjustment"),
-    rParamZyn(POffsetHz,          "Voice constant offset"),
-    rParamI(PDetune,        "Fine Detune"),
-    rParamI(PCoarseDetune,  "Coarse Detune"),
-    rParamZyn(PDetuneType,  "Magnitude of Detune"),
+    rParamZyn(POffsetHz,    rShort("offset"), "Voice constant offset"),
+    rParamI(PDetune,        rShort("fine"),   "Fine Detune"),
+    rParamI(PCoarseDetune,  rShort("coarse"), "Coarse Detune"),
+    rParamZyn(PDetuneType,  rShort("type"),   "Magnitude of Detune"),
 
     {"sample#64:ifb", rProp(internal) rDoc("Nothing to see here"), 0,
         [](const char *m, rtosc::RtData &d)
@@ -86,7 +86,7 @@ static const rtosc::Ports realtime_ports =
             PADnoteParameters *obj = (PADnoteParameters *)d.obj;
             d.reply(d.loc, "f", getdetune(obj->PDetuneType, 0, obj->PDetune));
         }},
-    {"octave::c:i", rProp(parameter) rDoc("Octave note offset"), NULL,
+    {"octave::c:i", rProp(parameter) rShort("octave") rDoc("Octave note offset"), NULL,
         [](const char *msg, RtData &d)
         {
             PADnoteParameters *obj = (PADnoteParameters *)d.obj;
-- 
2.47.0

