From 8211bd244a02869f0b9271780f4e932abbe62a67 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Fri, 22 Jul 2016 09:54:19 -0400
Subject: [PATCH 0966/1936] Add Some Default Bank Tags

---
 src/Misc/MiddleWare.cpp | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/src/Misc/MiddleWare.cpp b/src/Misc/MiddleWare.cpp
index 13161a1c..02422f8e 100644
--- a/src/Misc/MiddleWare.cpp
+++ b/src/Misc/MiddleWare.cpp
@@ -944,6 +944,25 @@ const rtosc::Ports bankPorts = {
         }
         d.replyArray("/bank/types", t, args);
         rEnd},
+    {"tags:", 0, 0,
+        rBegin;
+        const char *types[8];
+        types[ 0] = "fast";
+        types[ 1] = "slow";
+        types[ 2] = "saw";
+        types[ 3] = "bell";
+        types[ 4] = "lead";
+        types[ 5] = "ambient";
+        types[ 6] = "horn";
+        types[ 7] = "alarm";
+        char        t[8+1]={0};
+        rtosc_arg_t args[8];
+        for(int i=0; i<8; ++i) {
+            t[i]      = 's';
+            args[i].s = types[i];
+        }
+        d.replyArray(d.loc, t, args);
+        rEnd},
     {"slot#1024:", 0, 0,
         rBegin;
         const int loc = extractInt(msg);
-- 
2.47.0

