From d593e897a1f0ce8eb4118884fc5a6e79203501fb Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Fri, 22 Jul 2016 12:31:05 -0400
Subject: [PATCH 0970/1936] Add Distortion Type Metadata

---
 src/Effects/Distorsion.cpp | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/Effects/Distorsion.cpp b/src/Effects/Distorsion.cpp
index fe5c127f..dc9801cf 100644
--- a/src/Effects/Distorsion.cpp
+++ b/src/Effects/Distorsion.cpp
@@ -32,7 +32,12 @@ rtosc::Ports Distorsion::ports = {
     rEffPar(Plrcross, 2, rShort("l/r") "Left/Right Crossover"),
     rEffPar(Pdrive,   3, rShort("drive"), "Input amplification"),
     rEffPar(Plevel,   4, rShort("output"), "Output amplification"),
-    rEffPar(Ptype,    5, rShort("type"), "Distortion Shape"),
+    rEffPar(Ptype,    5, rShort("type"),
+            rOptions(Undistorted,
+                Arctangent, Asymmetric, Pow, Sine, Quantisize,
+                Zigzag, Limiter, Upper Limiter, Lower Limiter,
+                Inverse Limiter, Clip, Asym2, Pow2, sigmoid),
+            "Distortion Shape"),
     rEffParTF(Pnegate, 6, rShort("neg"), "Negate Signal"),
     rEffPar(Plpf, 7, rShort("lpf"), "Low Pass Cutoff"),
     rEffPar(Phpf, 8, rShort("hpf"), "High Pass Cutoff"),
-- 
2.47.0

