From f096f7355d2d0b774a0532d0f5143cfcd4693cdb Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Tue, 26 Jul 2016 17:02:52 -0400
Subject: [PATCH 0979/1936] SVFilter: Fix Compile Issue After Merge

---
 src/DSP/SVFilter.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/DSP/SVFilter.cpp b/src/DSP/SVFilter.cpp
index 7222c6d8..ec6a29cf 100644
--- a/src/DSP/SVFilter.cpp
+++ b/src/DSP/SVFilter.cpp
@@ -19,6 +19,7 @@
 #include "SVFilter.h"
 
 #define errx(...)
+#define warnx(...)
 #ifndef errx
 #include <err.h>
 #endif
-- 
2.47.0

