From 6f073935facc78af54fe8c9d8f3ab1e4705a0d53 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Mon, 1 Aug 2016 10:30:50 -0400
Subject: [PATCH 0991/1936] Add Workaround For Windows Home Dir Paths

---
 src/Misc/MiddleWare.cpp | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/src/Misc/MiddleWare.cpp b/src/Misc/MiddleWare.cpp
index 24ae7d21..71ca65a8 100644
--- a/src/Misc/MiddleWare.cpp
+++ b/src/Misc/MiddleWare.cpp
@@ -1198,7 +1198,14 @@ static rtosc::Ports middwareSnoopPorts = {
         rEnd},
     {"file_home_dir:", 0, 0,
         rBegin;
-        d.reply(d.loc, "s", getenv("HOME"));
+        const char *home = getenv("HOME");
+        if(!home)
+            home = getenv("USERPROFILE");
+        if(!home)
+            home = getenv("HOMEPATH");
+        if(!home)
+            home = "/";
+        d.reply(d.loc, "s", home);
         rEnd},
     {"file_list_files:s", 0, 0,
         rBegin;
-- 
2.47.0

