From c7678a6680a2a8ba908c6ba3e7250703de5a881a Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Mon, 1 Aug 2016 22:18:23 -0400
Subject: [PATCH 0993/1936] Middleware: Add OSC Damage On Clear Actions

---
 src/Misc/MiddleWare.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/Misc/MiddleWare.cpp b/src/Misc/MiddleWare.cpp
index 71ca65a8..f8d8fb2f 100644
--- a/src/Misc/MiddleWare.cpp
+++ b/src/Misc/MiddleWare.cpp
@@ -1272,6 +1272,7 @@ static rtosc::Ports middwareSnoopPorts = {
     {"reset_master:", 0, 0,
         rBegin;
         impl.loadMaster(NULL);
+        d.reply("/damage", "s", "/");
         rEnd},
     {"load_xiz:is", 0, 0,
         rBegin;
@@ -1309,7 +1310,9 @@ static rtosc::Ports middwareSnoopPorts = {
         rEnd},
     {"part#16/clear:", 0, 0,
         rBegin;
-        impl.loadClearPart(extractInt(msg));
+        int id = extractInt(msg);
+        impl.loadClearPart(id);
+        d.reply("/damage", "s", ("/part"+to_s(id)).c_str());
         rEnd},
     {"undo:", 0, 0,
         rBegin;
-- 
2.47.0

