From d7cc9391d0e1cbb701774610cb639e2269f85b39 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Wed, 10 Aug 2016 13:33:00 -0400
Subject: [PATCH 1002/1936] CMake: Fix LV2 Build

---
 src/Plugin/ZynAddSubFX/CMakeLists.txt | 32 +++++++++++++--------------
 1 file changed, 16 insertions(+), 16 deletions(-)

diff --git a/src/Plugin/ZynAddSubFX/CMakeLists.txt b/src/Plugin/ZynAddSubFX/CMakeLists.txt
index a39ea3fc..2cfc7f0b 100644
--- a/src/Plugin/ZynAddSubFX/CMakeLists.txt
+++ b/src/Plugin/ZynAddSubFX/CMakeLists.txt
@@ -73,11 +73,11 @@ add_library(ZynAddSubFX_vst SHARED
 else()
 
 # UI Disabled
-#add_library(ZynAddSubFX_lv2 SHARED
-#    ${CMAKE_SOURCE_DIR}/src/globals.cpp
-#    ${CMAKE_SOURCE_DIR}/src/UI/ConnectionDummy.cpp
-#    ${CMAKE_SOURCE_DIR}/DPF/distrho/DistrhoPluginMain.cpp
-#    ZynAddSubFX.cpp)
+add_library(ZynAddSubFX_lv2 SHARED
+    ${CMAKE_SOURCE_DIR}/src/globals.cpp
+    ${CMAKE_SOURCE_DIR}/src/UI/ConnectionDummy.cpp
+    ${CMAKE_SOURCE_DIR}/DPF/distrho/DistrhoPluginMain.cpp
+    ZynAddSubFX.cpp)
 
 add_library(ZynAddSubFX_vst SHARED
     ${CMAKE_SOURCE_DIR}/src/globals.cpp
@@ -87,10 +87,10 @@ add_library(ZynAddSubFX_vst SHARED
 
 endif()
 
-#set_target_properties(ZynAddSubFX_lv2 PROPERTIES COMPILE_DEFINITIONS "DISTRHO_PLUGIN_TARGET_LV2")
-#set_target_properties(ZynAddSubFX_lv2 PROPERTIES LIBRARY_OUTPUT_DIRECTORY "lv2")
-#set_target_properties(ZynAddSubFX_lv2 PROPERTIES OUTPUT_NAME "ZynAddSubFX")
-#set_target_properties(ZynAddSubFX_lv2 PROPERTIES PREFIX "")
+set_target_properties(ZynAddSubFX_lv2 PROPERTIES COMPILE_DEFINITIONS "DISTRHO_PLUGIN_TARGET_LV2")
+set_target_properties(ZynAddSubFX_lv2 PROPERTIES LIBRARY_OUTPUT_DIRECTORY "lv2")
+set_target_properties(ZynAddSubFX_lv2 PROPERTIES OUTPUT_NAME "ZynAddSubFX")
+set_target_properties(ZynAddSubFX_lv2 PROPERTIES PREFIX "")
 
 set_target_properties(ZynAddSubFX_vst PROPERTIES COMPILE_DEFINITIONS "DISTRHO_PLUGIN_TARGET_VST")
 set_target_properties(ZynAddSubFX_vst PROPERTIES LIBRARY_OUTPUT_DIRECTORY "vst")
@@ -107,17 +107,17 @@ else()
     set(PLATFORM_LIBRARIES X11 GL)
 endif()
 
-#target_link_libraries(ZynAddSubFX_lv2 zynaddsubfx_core ${OS_LIBRARIES} ${LIBLO_LIBRARIES}
-#    ${PLATFORM_LIBRARIES})
+target_link_libraries(ZynAddSubFX_lv2 zynaddsubfx_core ${OS_LIBRARIES} ${LIBLO_LIBRARIES}
+    ${PLATFORM_LIBRARIES})
 target_link_libraries(ZynAddSubFX_vst zynaddsubfx_core ${OS_LIBRARIES} ${LIBLO_LIBRARIES}
     ${PLATFORM_LIBRARIES})
 
-#install(TARGETS ZynAddSubFX_lv2 LIBRARY DESTINATION ${PluginLibDir}/lv2/ZynAddSubFX.lv2/)
-#install(TARGETS ZynAddSubFX_vst LIBRARY DESTINATION ${PluginLibDir}/vst/)
+install(TARGETS ZynAddSubFX_lv2 LIBRARY DESTINATION ${PluginLibDir}/lv2/ZynAddSubFX.lv2/)
+install(TARGETS ZynAddSubFX_vst LIBRARY DESTINATION ${PluginLibDir}/vst/)
 
-#add_custom_command(TARGET ZynAddSubFX_lv2 POST_BUILD
-#    COMMAND ../../lv2-ttl-generator $<TARGET_FILE:ZynAddSubFX_lv2>
-#    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/lv2)
+add_custom_command(TARGET ZynAddSubFX_lv2 POST_BUILD
+    COMMAND ../../lv2-ttl-generator $<TARGET_FILE:ZynAddSubFX_lv2>
+    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/lv2)
 
 install(FILES
     ${CMAKE_CURRENT_BINARY_DIR}/lv2/manifest.ttl
-- 
2.47.0

