From 512852d670971b9cfe170303fcae12a64b5043f7 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Wed, 17 Aug 2016 15:37:20 -0400
Subject: [PATCH 1004/1936] Middleware: Add More Liblo Error Handling

---
 src/Misc/MiddleWare.cpp | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/Misc/MiddleWare.cpp b/src/Misc/MiddleWare.cpp
index d062d95c..98b86e46 100644
--- a/src/Misc/MiddleWare.cpp
+++ b/src/Misc/MiddleWare.cpp
@@ -1614,6 +1614,10 @@ void MiddleWareImpl::sendToRemote(const char *rtmsg, std::string dest)
     } else if(!dest.empty()) {
         lo_message msg  = lo_message_deserialise((void*)rtmsg,
                 rtosc_message_length(rtmsg, bToU->buffer_size()), NULL);
+        if(!msg) {
+            printf("[ERROR] OSC to <%s> Failed To Parse In Liblo\n", rtmsg);
+            return;
+        }
 
         //Send to known url
         lo_address addr = lo_address_new_from_url(dest.c_str());
@@ -1650,6 +1654,11 @@ void MiddleWareImpl::bToUhandle(const char *rtmsg)
     MwDataObj d(this);
     middlewareReplyPorts.dispatch(rtmsg, d, true);
 
+    if(!rtmsg) {
+        fprintf(stderr, "[ERROR] Unexpected Null OSC In Zyn\n");
+        return;
+    }
+
     in_order = true;
     //Normal message not captured by the ports
     if(d.matches == 0) {
-- 
2.47.0

