From 123486d09d499c0b5aeb7fdff992f1982c9a989f Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Wed, 17 Aug 2016 15:38:14 -0400
Subject: [PATCH 1005/1936] Add Handler For No Args Bank Search

---
 src/Misc/MiddleWare.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/Misc/MiddleWare.cpp b/src/Misc/MiddleWare.cpp
index 98b86e46..28a25ae1 100644
--- a/src/Misc/MiddleWare.cpp
+++ b/src/Misc/MiddleWare.cpp
@@ -1075,6 +1075,10 @@ const rtosc::Ports bankPorts = {
         d.replyArray("/bank/search_results", res_type, res_dat);
 #undef MAX_SEARCH
         rEnd},
+    {"search_results:", 0, 0,
+        rBegin;
+        d.reply("/bank/search_results", "");
+        rEnd},
 };
 
 /******************************************************************************
-- 
2.47.0

