From 69e3ff2d306c78ac4c1389f632b648ba40e3f85d Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Tue, 30 Aug 2016 18:44:53 -0400
Subject: [PATCH 1013/1936] Envelope: Add Penvsustain To The External API

---
 src/Params/EnvelopeParams.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Params/EnvelopeParams.cpp b/src/Params/EnvelopeParams.cpp
index 14efc560..daefe882 100644
--- a/src/Params/EnvelopeParams.cpp
+++ b/src/Params/EnvelopeParams.cpp
@@ -39,7 +39,7 @@ static const rtosc::Ports localPorts = {
 #define rChangeCb if(!obj->Pfreemode) obj->converttofree(); \
                   if(obj->time) { obj->last_update_timestamp = obj->time->time(); }
     rParamZyn(Penvpoints, rProp(internal), "Number of points in complex definition"),
-    rParamZyn(Penvsustain, rProp(internal), "Location of the sustain point"),
+    rParamZyn(Penvsustain, "Location of the sustain point"),
     rParams(Penvdt,  MAX_ENVELOPE_POINTS, "Envelope Delay Times"),
     rParams(Penvval, MAX_ENVELOPE_POINTS, "Envelope Values"),
     rParamZyn(Penvstretch,  rShort("stretch"),
-- 
2.47.0

