From cb5933ef0012320bd561253f1b8a47e532f9c987 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Tue, 30 Aug 2016 18:46:11 -0400
Subject: [PATCH 1014/1936] Master: Fix Pinsparts Options Metadata

---
 src/Misc/Master.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Misc/Master.cpp b/src/Misc/Master.cpp
index 2fd91c59..abe21646 100644
--- a/src/Misc/Master.cpp
+++ b/src/Misc/Master.cpp
@@ -116,7 +116,7 @@ static const Ports master_ports = {
     rRecursp(insefx, 8, "Insertion Effect"),//NUM_INS_EFX
     rRecur(microtonal, "Micrtonal Mapping Functionality"),
     rRecur(ctl, "Controller"),
-    rArrayI(Pinsparts, NUM_INS_EFX, rOpt(-1, Master),
+    rArrayI(Pinsparts, NUM_INS_EFX, rOpt(-2, Master), rOpt(-1, Off)
             rOptions(Part1, Part2, Part3, Part4,  Part5, Part6,
                  Part7, Part8, Part9, Part10, Part11, Part12,
                  Part13, Part14, Part15, Part16),
-- 
2.47.0

