From 6a9efbe8f317813b57e21054f9729fad8edf088a Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Thu, 8 Sep 2016 10:25:07 -0400
Subject: [PATCH 1018/1936] FilterParams: Add More Short Names To OSC Metadata

---
 src/Params/FilterParams.cpp | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/src/Params/FilterParams.cpp b/src/Params/FilterParams.cpp
index ac72e13a..dffbffd7 100644
--- a/src/Params/FilterParams.cpp
+++ b/src/Params/FilterParams.cpp
@@ -32,9 +32,9 @@ constexpr int sizeof_pvowels = sizeof(FilterParams::Pvowels);
 #define rObject FilterParams::Pvowels_t::formants_t
 
 static const rtosc::Ports subsubports = {
-    rParamZyn(freq, "Formant frequency"),
-    rParamZyn(amp,  "Strength of formant"),
-    rParamZyn(q,    "Quality Factor"),
+    rParamZyn(freq, rShort("f.freq"), "Formant frequency"),
+    rParamZyn(amp,  rShort("f.str"),  "Strength of formant"),
+    rParamZyn(q,    rShort("f.q"),    "Quality Factor"),
 };
 #undef rObject
 
@@ -69,6 +69,7 @@ const rtosc::Ports FilterParams::ports = {
     rParamZyn(Pq,               rShort("q"),
             "Quality Factor (resonance/bandwidth)"),
     rOption(Pstages,            rShort("stages"),
+            rLinear(1,5),
             rOptions(1, 2, 3, 4, 5), "Filter Stages"),
     rParamZyn(Pfreqtrack,       rShort("f.track"),
             "Frequency Tracking amount"),
@@ -85,9 +86,9 @@ const rtosc::Ports FilterParams::ports = {
             "Number of octaves for formant"),
 
     //TODO check if FF_MAX_SEQUENCE is acutally expanded or not
-    rParamZyn(Psequencesize, rMap(max, FF_MAX_SEQUENCE), "Length of vowel sequence"),
-    rParamZyn(Psequencestretch, "How modulators stretch the sequence"),
-    rToggle(Psequencereversed, "If the modulator input is inverted"),
+    rParamZyn(Psequencesize,    rShort("seq.size"), rMap(max, FF_MAX_SEQUENCE), "Length of vowel sequence"),
+    rParamZyn(Psequencestretch, rShort("seq.str"), "How modulators stretch the sequence"),
+    rToggle(Psequencereversed,  rShort("reverse"), "If the modulator input is inverted"),
 
     //{"Psequence#" FF_MAX_SEQUENCE "/nvowel", "", NULL, [](){}},
     {"type-svf::i", rProp(parameter) rShort("type")
-- 
2.47.0

