From 56fa3683b483d20e93ca626e947934b64e662985 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Mon, 19 Sep 2016 16:55:21 -0400
Subject: [PATCH 1021/1936] Update Rtosc To Fix Metadata Bug

---
 rtosc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rtosc b/rtosc
index 0771cb1f..6c9aaf90 160000
--- a/rtosc
+++ b/rtosc
@@ -1 +1 @@
-Subproject commit 0771cb1f29a68d13bb8ec00abaaafcb54cb1366e
+Subproject commit 6c9aaf902debf0849e5ffc4588249225ccad46b3
-- 
2.47.0

