From f03de1cb69f7e9721e8ed8fe139d4c1b41fab918 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Mon, 19 Sep 2016 16:56:11 -0400
Subject: [PATCH 1023/1936] PADnoteParameters: Add Anti-Pop Short Metadata

---
 src/Params/PADnoteParameters.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Params/PADnoteParameters.cpp b/src/Params/PADnoteParameters.cpp
index 41063fc2..b0fae260 100644
--- a/src/Params/PADnoteParameters.cpp
+++ b/src/Params/PADnoteParameters.cpp
@@ -45,7 +45,7 @@ static const rtosc::Ports realtime_ports =
     rParamZyn(PVolume,  rShort("vol"), "Synth Volume"),
     rParamZyn(PAmpVelocityScaleFunction, rShort("sense"), "Amplitude Velocity Sensing function"),
 
-    rParamZyn(Fadein_adjustment, "Adjustment for anti-pop strategy."),
+    rParamZyn(Fadein_adjustment, rShort("a.pop."), "Adjustment for anti-pop strategy."),
 
     //Punch
     rParamZyn(PPunchStrength, rShort("strength"), "Punch Strength"),
-- 
2.47.0

