From 0c1ccba9cffbf843c755f76f9a0783d636dc8cd9 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Mon, 19 Sep 2016 16:56:34 -0400
Subject: [PATCH 1024/1936] SUBnoteParameters: Fix Detune Metadata

---
 src/Params/SUBnoteParameters.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/Params/SUBnoteParameters.cpp b/src/Params/SUBnoteParameters.cpp
index 705db1c8..b9da5fbb 100644
--- a/src/Params/SUBnoteParameters.cpp
+++ b/src/Params/SUBnoteParameters.cpp
@@ -43,7 +43,8 @@ static const rtosc::Ports SUBnotePorts = {
     rParamI(PDetune,       rShort("detune"), rLinear(0, 16383), "Detune in detune type units"),
     rParamI(PCoarseDetune, rShort("cdetune"), "Coarse Detune"),
     //Real values needed
-    rOption(PDetuneType,               rShort("det. scl."), rOptions(100 cents, 200 cents, 500 cents), "Detune Scale"),
+    rOption(PDetuneType,               rShort("det. scl."),
+            rOptions(L35 cents, L10 cents, E100 cents, E1200 cents), "Detune Scale"),
     rToggle(PFreqEnvelopeEnabled,      rShort("enable"), "Enable for Frequency Envelope"),
     rToggle(PBandWidthEnvelopeEnabled, rShort("enable"), "Enable for Bandwidth Envelope"),
     rToggle(PGlobalFilterEnabled,                 rShort("enable"), "Enable for Global Filter"),
-- 
2.47.0

