From 8b7f62fc26bda8f4a504bfeab7a851e95ab2a5cc Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Tue, 20 Sep 2016 21:25:06 -0400
Subject: [PATCH 1027/1936] Remove Debug Printfs

---
 src/Effects/EQ.cpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/Effects/EQ.cpp b/src/Effects/EQ.cpp
index 5c2e84d8..9575ab1b 100644
--- a/src/Effects/EQ.cpp
+++ b/src/Effects/EQ.cpp
@@ -65,9 +65,7 @@ rtosc::Ports EQ::ports = {
     {"filter#8/", 0, &filterports,
         rBegin;
         (void)obj;
-        printf("got a filter\n");
         SNIP;
-        printf("dispatch to <%s>\n", msg);
         filterports.dispatch(msg, d);
         rEnd},
     {"coeff:", rProp(internal) rDoc("Get equalizer Coefficients"), NULL,
-- 
2.47.0

