From 849392909eff393db5a08c2c73e22bf5ed77c334 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Tue, 20 Sep 2016 21:25:20 -0400
Subject: [PATCH 1028/1936] Default To Using $PWD As File Home Dir

---
 src/Misc/MiddleWare.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/Misc/MiddleWare.cpp b/src/Misc/MiddleWare.cpp
index 2c22314c..c74887bc 100644
--- a/src/Misc/MiddleWare.cpp
+++ b/src/Misc/MiddleWare.cpp
@@ -1239,7 +1239,9 @@ static rtosc::Ports middwareSnoopPorts = {
         rEnd},
     {"file_home_dir:", 0, 0,
         rBegin;
-        const char *home = getenv("HOME");
+        const char *home = getenv("PWD");
+        if(!home)
+            home = getenv("HOME");
         if(!home)
             home = getenv("USERPROFILE");
         if(!home)
-- 
2.47.0

