From db749e86f7a1a432250f77d80636c8c908a75370 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sun, 2 Oct 2016 15:52:58 -0400
Subject: [PATCH 1037/1936] Increase Poll Rate For No-UI/Zest-UI

---
 src/UI/ConnectionDummy.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/UI/ConnectionDummy.cpp b/src/UI/ConnectionDummy.cpp
index 7284344f..17c17766 100644
--- a/src/UI/ConnectionDummy.cpp
+++ b/src/UI/ConnectionDummy.cpp
@@ -27,7 +27,7 @@ void raiseUi(ui_handle_t, const char *, const char *, ...)
 }
 void tickUi(ui_handle_t)
 {
-    usleep(100000);
+    usleep(1000);
 }
 Fl_Osc_Interface *genOscInterface(MiddleWare*)
 {
-- 
2.47.0

