From 324ad564b72ef1aed61ec4b177dbb04d73462b1b Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sun, 2 Oct 2016 15:57:08 -0400
Subject: [PATCH 1041/1936] Remove Debug #error Mistakenly Left In the Code

---
 src/Plugin/ZynAddSubFX/ZynAddSubFX-UI.cpp | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/Plugin/ZynAddSubFX/ZynAddSubFX-UI.cpp b/src/Plugin/ZynAddSubFX/ZynAddSubFX-UI.cpp
index 1daa4a81..b5215c9f 100644
--- a/src/Plugin/ZynAddSubFX/ZynAddSubFX-UI.cpp
+++ b/src/Plugin/ZynAddSubFX/ZynAddSubFX-UI.cpp
@@ -27,7 +27,6 @@ public:
     {
         setTitle("ZynAddSubFX");
 
-#error this should not be compiled
 #if DISTRHO_OS_MAC
         extUiPath  = bpath;
         extUiPath += "/zynaddsubfx-ext-gui";
-- 
2.47.0

