From 4b95ad25aebe87c79edfcfde86bd03ac46031d48 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sun, 2 Oct 2016 16:03:42 -0400
Subject: [PATCH 1042/1936] Fix Ext-Gui Compile

---
 src/UI/guimain.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/UI/guimain.cpp b/src/UI/guimain.cpp
index fac13889..7190dffb 100644
--- a/src/UI/guimain.cpp
+++ b/src/UI/guimain.cpp
@@ -46,7 +46,7 @@ bool fileexists(const char *filename)
     return false;
 }
 
-extern int Pexitprogram;
+int Pexitprogram=0;
 
 
 #include "Connection.h"
-- 
2.47.0

