From 445ec10d3408c7c6b15e0955bb6708a82232a453 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Thu, 6 Oct 2016 21:45:54 -0400
Subject: [PATCH 1047/1936] Middleware: Increase Max Msg. Size In Ringbuffers

---
 src/Misc/MiddleWare.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/Misc/MiddleWare.cpp b/src/Misc/MiddleWare.cpp
index 027ffc69..4491d725 100644
--- a/src/Misc/MiddleWare.cpp
+++ b/src/Misc/MiddleWare.cpp
@@ -1491,8 +1491,8 @@ MiddleWareImpl::MiddleWareImpl(MiddleWare *mw, SYNTH_T synth_,
                 int res = master->saveXML(save_file.c_str());
                 (void)res;});})
 {
-    bToU = new rtosc::ThreadLink(4096*2,1024);
-    uToB = new rtosc::ThreadLink(4096*2,1024);
+    bToU = new rtosc::ThreadLink(4096*2*16,1024/16);
+    uToB = new rtosc::ThreadLink(4096*2*16,1024/16);
     midi_mapper.base_ports = &Master::ports;
     midi_mapper.rt_cb      = [this](const char *msg){handleMsg(msg);};
     if(preferrred_port != -1)
-- 
2.47.0

