From 04e1af5159a9d0b6d2cbed6586a2f56685adafd0 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sun, 16 Oct 2016 09:29:40 -0400
Subject: [PATCH 1063/1936] Fix Some Controller Metadata

---
 src/Params/Controller.cpp | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/src/Params/Controller.cpp b/src/Params/Controller.cpp
index c424ce1d..40e1a0b4 100644
--- a/src/Params/Controller.cpp
+++ b/src/Params/Controller.cpp
@@ -42,10 +42,11 @@ const rtosc::Ports Controller::ports = {
     rToggle(volume.receive,     rShort("vol.rcv"), "Volume MIDI Receive"),
     rToggle(sustain.receive,    rShort("sus.rcv"), "Sustain MIDI Receive"),
     rToggle(portamento.receive, rShort("prt.rcv"), "Portamento MIDI Receive"),
-    rToggle(portamento.portamento, "UNDOCUMENTED"),
+    rToggle(portamento.portamento, "Portamento Enable"),
     rParamZyn(portamento.time,          rShort("time"), "Portamento Length"),
-    rToggle(portamento.proportional,    rShort("propt."), "If all portamentos are proportional to the distance they span"),
-    rParamZyn(portamento.propRate,      rShort("rate"), "Portamento proportional rate"),
+    rToggle(portamento.proportional,    rShort("propt."), "Whether the portamento time is proportional"
+            "to the size of the interval between two notes."),
+    rParamZyn(portamento.propRate,      rShort("scale"), "Portamento proportional scale"),
     rParamZyn(portamento.propDepth,     rShort("depth"), "Portamento proportional depth"),
     rParamZyn(portamento.pitchthresh,   rShort("thresh"), "Threshold for portamento"),
     rToggle(portamento.pitchthreshtype, rShort("tr.type"), "Type of threshold"),
-- 
2.47.0

