From eb07b6fabe42c9259a9d0b7c540c228d55dc82a2 Mon Sep 17 00:00:00 2001
From: Johannes Lorenz <johannes89@ist-einmalig.de>
Date: Sun, 16 Oct 2016 18:34:12 +0200
Subject: [PATCH 1068/1936] Suppress errors from system headers.

---
 src/CMakeLists.txt | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index e5ee7962..b1af22a4 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -262,12 +262,18 @@ add_definitions(
 	 -Wextra
 	 )
 
-CHECK_CXX_COMPILER_FLAG(-Wno-inconsistent-missing-override
+check_cxx_compiler_flag(-Wno-inconsistent-missing-override
 	COMPILER_SUPPORTS_OVERRIDE)
 if(COMPILER_SUPPORTS_OVERRIDE)
     add_definitions(-Wno-inconsistent-missing-override)
 endif()
 
+check_cxx_compiler_flag(--system-header-prefix
+	COMPILER_SUPPORTS_SYSTEM_HDR_PREFIX)
+if(COMPILER_SUPPORTS_SYSTEM_HDR_PREFIX)
+    add_definitions(--system-header-prefix="FL/")
+endif()
+
 if(NOT AVOID_ASM)
 	message(STATUS "Compiling with x86 opcode support")
     add_definitions(-DASM_F2I_YES)
-- 
2.47.0

