From 4b2077b3a216bc22b75d0dc30f88f1bdc0eb8ac7 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Tue, 15 Nov 2016 18:50:28 -0500
Subject: [PATCH 1073/1936] FilterParams: Update Q Metadata

---
 src/Params/FilterParams.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Params/FilterParams.cpp b/src/Params/FilterParams.cpp
index eb852196..d392845d 100644
--- a/src/Params/FilterParams.cpp
+++ b/src/Params/FilterParams.cpp
@@ -34,7 +34,7 @@ constexpr int sizeof_pvowels = sizeof(FilterParams::Pvowels);
 static const rtosc::Ports subsubports = {
     rParamZyn(freq, rShort("f.freq"), "Formant frequency"),
     rParamZyn(amp,  rShort("f.str"),  "Strength of formant"),
-    rParamZyn(q,    rShort("f.q"),    "Quality Factor"),
+    rParamZyn(q,    rShort("f.q"),    "The formant's quality factor, also known as resonance bandwith or Q for short"),
 };
 #undef rObject
 
-- 
2.47.0

