From ab6fbf640dc40cae0499f0594e014125fb80e367 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sat, 26 Nov 2016 13:01:41 -0500
Subject: [PATCH 1096/1936] OscilGen: Fix Padaptiveharmonicspar Range MetaData

---
 src/Synth/OscilGen.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Synth/OscilGen.cpp b/src/Synth/OscilGen.cpp
index 481ccc30..b718e711 100644
--- a/src/Synth/OscilGen.cpp
+++ b/src/Synth/OscilGen.cpp
@@ -199,7 +199,7 @@ const rtosc::Ports OscilGen::realtime_ports{
             "Base frequency of adaptive harmonic (30..3000Hz)"),
     rParamI(Padaptiveharmonicspower, rShort("amount"), rLinear(0,200),
             "Adaptive Harmonic Strength"),
-    rParamZyn(Padaptiveharmonicspar, rShort("power"),
+    rParamI(Padaptiveharmonicspar, rShort("power"), rLinear(0,100),
             "Adaptive Harmonics Postprocessing Power"),
     {"waveform:", rDoc("Returns waveform points"),
         NULL, [](const char *, rtosc::RtData &d) {
-- 
2.47.0

