From f4bcd4deba7f9e4866e4d8dd9711fa00287452bc Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Mon, 28 Nov 2016 18:31:12 -0500
Subject: [PATCH 1101/1936] Avoid OpenGL Dep In Non-Zest Plugin

---
 src/Plugin/ZynAddSubFX/CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/Plugin/ZynAddSubFX/CMakeLists.txt b/src/Plugin/ZynAddSubFX/CMakeLists.txt
index 160209b9..07751250 100644
--- a/src/Plugin/ZynAddSubFX/CMakeLists.txt
+++ b/src/Plugin/ZynAddSubFX/CMakeLists.txt
@@ -107,8 +107,10 @@ if(${CMAKE_SYSTEM_NAME} STREQUAL "Windows")
             wsock32
             "-static" iphlpapi
             "-static" winpthread)
-else()
+elseif(ZestGui)
     set(PLATFORM_LIBRARIES X11 GL rt)
+else()
+    set(PLATFORM_LIBRARIES X11 rt)
 endif()
 
 target_link_libraries(ZynAddSubFX_lv2 zynaddsubfx_core ${OS_LIBRARIES} ${LIBLO_LIBRARIES}
-- 
2.47.0

