From 39e903d30cdddd3dd44c5f4ea866dc4b9a571881 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sat, 3 Dec 2016 11:25:23 -0500
Subject: [PATCH 1104/1936] Set 0 Volume Voices To -Inf dB

---
 src/Synth/ADnote.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/Synth/ADnote.cpp b/src/Synth/ADnote.cpp
index 9a9109ca..a6e5e792 100644
--- a/src/Synth/ADnote.cpp
+++ b/src/Synth/ADnote.cpp
@@ -651,6 +651,8 @@ void ADnote::legatonote(LegatoParams lpars)
                  * (1.0f - pars.VoicePar[nvoice].PVolume / 127.0f))             // -60 dB .. 0 dB
             * VelF(velocity,
                    pars.VoicePar[nvoice].PAmpVelocityScaleFunction); //velocity
+        if(pars.VoicePar[nvoice].PVolume == 0)
+            NoteVoicePar[nvoice].Volume = 0;
 
         if(pars.VoicePar[nvoice].PVolumeminus != 0)
             NoteVoicePar[nvoice].Volume = -NoteVoicePar[nvoice].Volume;
@@ -808,6 +810,8 @@ void ADnote::initparameters(WatchManager *wm, const char *prefix)
         /* Voice Amplitude Parameters Init */
         vce.Volume = powf(0.1f, 3.0f * (1.0f - param.PVolume / 127.0f)) // -60dB..0dB
                      * VelF(velocity, param.PAmpVelocityScaleFunction);
+        if(param.PVolume == 0)
+            vce.Volume = 0;
 
         if(param.PVolumeminus)
             vce.Volume = -vce.Volume;
-- 
2.47.0

