From aba35f5d3ff23000154b3706d549fc0a946ab801 Mon Sep 17 00:00:00 2001
From: Olivier Humbert <trebmuh@users.noreply.github.com>
Date: Wed, 28 Dec 2016 02:18:33 +0100
Subject: [PATCH 1111/1936] Duplicate status lines in cmakelist.txt

---
 src/CMakeLists.txt | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 2fed9510..2ecef60e 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -564,9 +564,6 @@ package_status(JACK_FOUND       "JACK     " "found"   ${Yellow})
 package_status(PORTAUDIO_FOUND  "PA       " "found"   ${Yellow})
 package_status(LASH_FOUND       "Lash     " "found"   ${Yellow})
 package_status(DSSI_FOUND       "DSSI     " "found"   ${Yellow})
-package_status(FLTK_FOUND       "fltk     " "found"   ${Yellow})
-package_status(NTK_FOUND        "ntk      " "found"   ${Yellow})
-package_status(X11_FOUND        "x11      " "found"   ${Yellow})
 package_status(CXXTEST_FOUND    "CxxTest  " "found"   ${Yellow})
 package_status(LashEnable       "Lash     " "enabled" ${Yellow})
 package_status(DssiEnable       "DSSI     " "enabled" ${Yellow})
-- 
2.47.0

