From 30dfc6d92182b4fd9241703b94cf7e420c1e4f3b Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Fri, 6 Jan 2017 10:28:47 -0500
Subject: [PATCH 1113/1936] Distortion: Fix Out-Of-Bounds Waveshaping Viz

---
 src/Effects/Distorsion.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/Effects/Distorsion.cpp b/src/Effects/Distorsion.cpp
index b71772c6..da1047b3 100644
--- a/src/Effects/Distorsion.cpp
+++ b/src/Effects/Distorsion.cpp
@@ -60,7 +60,8 @@ rtosc::Ports Distorsion::ports = {
             for(int i=0; i<128; ++i)
                 buffer[i] = 2*(i/128.0)-1;
 
-            waveShapeSmps(sizeof(buffer), buffer, dd.Ptype + 1, dd.Pdrive);
+            waveShapeSmps(sizeof(buffer)/sizeof(buffer[0]), buffer,
+                    dd.Ptype + 1, dd.Pdrive);
 
             for(int i=0; i<128; ++i) {
                 arg_str[i] = 'f';
-- 
2.47.0

