From 247e2337c4cb0de818303bfd48c568aab8f747be Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Tue, 31 Jan 2017 12:07:11 -0500
Subject: [PATCH 1116/1936] Update RTOSC - Fixes Relearning MIDI CC

---
 rtosc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rtosc b/rtosc
index 9d388756..d184b254 160000
--- a/rtosc
+++ b/rtosc
@@ -1 +1 @@
-Subproject commit 9d388756ecc46393fe77cfd5ff388bf234372996
+Subproject commit d184b254e8f52bd19a4b466b54234992e6e31928
-- 
2.47.0

