From f671cf5b29a17759b9599bb6b5e6a4f0d0c00814 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Thu, 23 Feb 2017 19:18:35 -0500
Subject: [PATCH 1117/1936] Microtonal: Adjust Metadata Labels

---
 src/Misc/Microtonal.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/Misc/Microtonal.cpp b/src/Misc/Microtonal.cpp
index 3ce81dc0..871c35d0 100644
--- a/src/Misc/Microtonal.cpp
+++ b/src/Misc/Microtonal.cpp
@@ -43,8 +43,8 @@ const rtosc::Ports Microtonal::ports = {
     rToggle(Pinvertupdown, rShort("inv."), "key mapping inverse"),
     rParamZyn(Pinvertupdowncenter, rShort("center"), "center of the inversion"),
     rToggle(Penabled, rShort("enable"), "Enable for microtonal mode"),
-    rParamZyn(PAnote, rShort("A note"), "The note for 'A'"),
-    rParamF(PAfreq,   rShort("A freq"), "Frequency of the 'A' note"),
+    rParamZyn(PAnote, rShort("1/1 midi note"), "The note for 'A'"),
+    rParamF(PAfreq,   rShort("ref freq"), "Frequency of the 'A' note"),
     rParamZyn(Pscaleshift, rShort("shift"), "UNDOCUMENTED"),
     rParamZyn(Pfirstkey, rShort("first key"), "First key to retune"),
     rParamZyn(Plastkey,  rShort("last key"), "Last key to retune"),
-- 
2.47.0

