From c0a62ddeee953f0ad4f211ed7079602a9ab8dcca Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sun, 12 Mar 2017 21:57:05 -0400
Subject: [PATCH 1119/1936] Exclude '.' In File Browser

---
 src/Misc/MiddleWare.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Misc/MiddleWare.cpp b/src/Misc/MiddleWare.cpp
index 904d8b1b..a20f898f 100644
--- a/src/Misc/MiddleWare.cpp
+++ b/src/Misc/MiddleWare.cpp
@@ -890,7 +890,7 @@ static std::vector<std::string> getFiles(const char *folder, bool finddir)
         printf("error = %x\n", INVALID_FILE_ATTRIBUTES);
         bool is_dir = GetFileAttributes(darn_windows.c_str()) & FILE_ATTRIBUTE_DIRECTORY;
 #endif
-        if(finddir == is_dir)
+        if(finddir == is_dir && strcmp(".", fn->d_name))
             files.push_back(fn->d_name);
     }
 
-- 
2.47.0

