From 6540188d33bb0b252e9f1b21412ac9d3e3518b53 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Thu, 23 Mar 2017 13:57:12 -0400
Subject: [PATCH 1124/1936] Fix Possible Crash With Too Many PAD Samples

---
 src/Params/PADnoteParameters.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/Params/PADnoteParameters.cpp b/src/Params/PADnoteParameters.cpp
index b0fae260..197e2c04 100644
--- a/src/Params/PADnoteParameters.cpp
+++ b/src/Params/PADnoteParameters.cpp
@@ -848,6 +848,9 @@ void PADnoteParameters::sampleGenerator(PADnoteParameters::callback cb,
     if(samplemax == 0)
         samplemax = 1;
 
+    if(samplemax > PAD_MAX_SAMPLES)
+        samplemax = PAD_MAX_SAMPLES;
+
     //prepare a BIG FFT
     FFTwrapper *fft      = new FFTwrapper(samplesize);
     fft_t      *fftfreqs = new fft_t[samplesize / 2];
-- 
2.47.0

