From 1b46561488f251582f5afe0c6b00238f2de975e5 Mon Sep 17 00:00:00 2001
From: Daniel Sheeler <dsheeler@pobox.com>
Date: Mon, 8 May 2017 16:45:45 -0500
Subject: [PATCH 1137/1936] Make plugin respond to midi program change

---
 src/Plugin/ZynAddSubFX/ZynAddSubFX.cpp | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/src/Plugin/ZynAddSubFX/ZynAddSubFX.cpp b/src/Plugin/ZynAddSubFX/ZynAddSubFX.cpp
index 459338ef..030af122 100644
--- a/src/Plugin/ZynAddSubFX/ZynAddSubFX.cpp
+++ b/src/Plugin/ZynAddSubFX/ZynAddSubFX.cpp
@@ -395,6 +395,17 @@ protected:
                 master->setController(channel, control, value);
             } break;
 
+            case 0xC0: {
+                const int program = midiEvent.data[1];
+
+                for(int i=0; i < NUM_MIDI_PARTS; ++i) {
+                    //set the program of the parts assigned to the midi channel
+                    if(master->part[i]->Prcvchn == channel) {
+                        middleware->pendingSetProgram(i, program);
+                    }
+                }
+            } break;
+
             case 0xE0: {
                 const uint8_t lsb = midiEvent.data[1];
                 const uint8_t msb = midiEvent.data[2];
-- 
2.47.0

