From bb92bcccb79cf7f8e31fe62bc35ace5d7dd12d46 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Fri, 26 May 2017 16:24:13 -0400
Subject: [PATCH 1139/1936] Fix Segfaults With Invalid .scl/.kbm Filenames

---
 src/Misc/Microtonal.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/Misc/Microtonal.cpp b/src/Misc/Microtonal.cpp
index 496669fc..1560eb0b 100644
--- a/src/Misc/Microtonal.cpp
+++ b/src/Misc/Microtonal.cpp
@@ -572,6 +572,9 @@ int Microtonal::loadscl(SclInfo &scl, const char *filename)
     char  tmp[500];
     OctaveTuning tmpoctave[MAX_OCTAVE_SIZE];
 
+    if(!file)
+        return 2;
+
     fseek(file, 0, SEEK_SET);
 
     //loads the short description
@@ -623,6 +626,9 @@ int Microtonal::loadkbm(KbmInfo &kbm, const char *filename)
     float tmpPAfreq = 440.0f;
     char  tmp[500];
 
+    if(!file)
+        return 2;
+
     fseek(file, 0, SEEK_SET);
     //loads the mapsize
     if(loadline(file, tmp) != 0 || sscanf(tmp, "%d", &x) == 0)
-- 
2.47.0

