From fa6232fbce625cf9a160eaa4a67e0595d0fed53a Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sun, 25 Jun 2017 18:21:23 -0400
Subject: [PATCH 1149/1936] Add compatiability port to fix test

---
 rtosc               | 2 +-
 src/Misc/Master.cpp | 8 ++++++++
 2 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/rtosc b/rtosc
index 35fad3e1..fb3bc5f4 160000
--- a/rtosc
+++ b/rtosc
@@ -1 +1 @@
-Subproject commit 35fad3e169685fd41bb84f7ffa2e59e93dbbe856
+Subproject commit fb3bc5f41f50cd5fd794db0514a304e47a70d564
diff --git a/src/Misc/Master.cpp b/src/Misc/Master.cpp
index b436aa57..db5a0a28 100644
--- a/src/Misc/Master.cpp
+++ b/src/Misc/Master.cpp
@@ -485,6 +485,14 @@ static const Ports master_ports = {
         rBOIL_END},
     {"bank/", rDoc("Controls for instrument banks"), &bankPorts,
             [](const char*,RtData&) {}},
+    {"learn:s", rProp(depricated) rDoc("MIDI Learn"), 0,
+        rBegin;
+        int free_slot = m->automate.free_slot();
+        if(free_slot >= 0) {
+            m->automate.createBinding(free_slot, rtosc_argument(msg, 0).s, true);
+            m->automate.active_slot = free_slot;
+        }
+        rEnd},
 };
 
 #undef rBegin
-- 
2.47.0

