From df5635ad2f8f5d0b3248a15a8c04a195a50797a3 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Fri, 21 Jul 2017 15:14:38 -0400
Subject: [PATCH 1167/1936] Disable Autosave By Default

---
 src/main.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/main.cpp b/src/main.cpp
index 0a741c6d..24556daf 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -318,7 +318,7 @@ int main(int argc, char *argv[])
     opterr = 0;
     int option_index = 0, opt, exitwithhelp = 0, exitwithversion = 0;
     int prefered_port = -1;
-    int auto_save_interval = 60;
+    int auto_save_interval = 0;
 int wmidi = -1;
 
     string loadfile, loadinstrument, execAfterInit, loadmidilearn;
-- 
2.47.0

