From 9e8100eb0fce68f3fcc038e2e312f9869927708d Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sun, 23 Jul 2017 11:10:17 -0400
Subject: [PATCH 1171/1936] Enhance Automation SubSlot Reset

---
 rtosc               | 2 +-
 src/Misc/Master.cpp | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/rtosc b/rtosc
index d2868a00..b84ca63b 160000
--- a/rtosc
+++ b/rtosc
@@ -1 +1 @@
-Subproject commit d2868a0031c01066dfcd21de02aa309b2bb7c1e9
+Subproject commit b84ca63bf12153e4531f88a320b1b9debf9b76ae
diff --git a/src/Misc/Master.cpp b/src/Misc/Master.cpp
index 1ae4994e..afdb790e 100644
--- a/src/Misc/Master.cpp
+++ b/src/Misc/Master.cpp
@@ -131,7 +131,7 @@ static int get_next_int(const char *msg)
 }
 
 static const Ports mapping_ports = {
-    {"offset::f", rProp(parameter) rShort("off") rLinear(-50, 50) rMap(unit, percent), 0,
+    {"offset::f", rProp(parameter) rDefault(0) rShort("off") rLinear(-50, 50) rMap(unit, percent), 0,
         rBegin;
         int slot = d.idx[1];
         int param = d.idx[0];
@@ -142,7 +142,7 @@ static const Ports mapping_ports = {
         } else
             d.reply(d.loc, "f", a.getSlotSubOffset(slot, param));
         rEnd},
-    {"gain::f", rProp(parameter) rShort("gain") rLinear(-200, 200) rMap(unit, percent), 0,
+    {"gain::f", rProp(parameter) rDefault(100) rShort("gain") rLinear(-200, 200) rMap(unit, percent), 0,
         rBegin;
         int slot = d.idx[1];
         int param = d.idx[0];
-- 
2.47.0

