From 879fc0243ae7e1a60357adc98e830496a72da510 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sun, 23 Jul 2017 13:29:56 -0400
Subject: [PATCH 1172/1936] Pad parameters: Fix Spelling Error

---
 src/Params/PADnoteParameters.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Params/PADnoteParameters.cpp b/src/Params/PADnoteParameters.cpp
index 25c95507..3db86bac 100644
--- a/src/Params/PADnoteParameters.cpp
+++ b/src/Params/PADnoteParameters.cpp
@@ -192,7 +192,7 @@ static const rtosc::Ports non_realtime_ports =
     //Harmonic Bandwidth
     rOption(Pbwscale, rShort("bw scale"),
             rOptions(Normal,
-              EqualHz, Quater,
+              EqualHz, Quarter,
               Half, 75%, 150%,
               Double, Inv. Half),
             rDefault(Normal),
-- 
2.47.0

