From a8e6ca76f559c9c4a5f9990e3440f43e2f4f6469 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Fri, 28 Jul 2017 11:30:35 -0400
Subject: [PATCH 1185/1936] Pass Two At Fixing Zyn-Fusion Effect Saving

---
 src/Effects/EffectMgr.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Effects/EffectMgr.cpp b/src/Effects/EffectMgr.cpp
index 35982248..4a29cff6 100644
--- a/src/Effects/EffectMgr.cpp
+++ b/src/Effects/EffectMgr.cpp
@@ -468,7 +468,7 @@ void EffectMgr::add2XML(XMLwrapper& xml)
     for(int n = 0; n < 128; ++n) {
         int par = 0;
         if(efx)
-            par = efx->getpar(par);
+            par = efx->getpar(n);
         else if(n<128)
             par = settings[n];
 
-- 
2.47.0

