From 00fb8c00391ab82773bad2cf316add3823043dc6 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sun, 30 Jul 2017 22:33:02 -0400
Subject: [PATCH 1188/1936] Filter: Fix Freq Tracking Param Loading

---
 src/Params/FilterParams.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Params/FilterParams.cpp b/src/Params/FilterParams.cpp
index 2d0d1eb3..e170f1da 100644
--- a/src/Params/FilterParams.cpp
+++ b/src/Params/FilterParams.cpp
@@ -570,7 +570,7 @@ void FilterParams::getfromXML(XMLwrapper& xml)
         basefreq     = xml.getparreal("basefreq",   1000);
         baseq        = xml.getparreal("baseq",      10);
         gain         = xml.getparreal("gain",       0);
-        freqtracking = xml.getparreal("freq_track", 0);
+        freqtracking = xml.getparreal("freq_tracking", 0);
     }
 
     //formant filter parameters
-- 
2.47.0

