From ab564231e0c6e8a0ae3893d6c7dcc87a197d1f38 Mon Sep 17 00:00:00 2001
From: Hans Petter Selasky <hps@selasky.org>
Date: Wed, 2 Aug 2017 22:47:17 +0200
Subject: [PATCH 1192/1936] Fix for clang compile warnings.

Signed-off-by: Hans Petter Selasky <hps@selasky.org>
---
 src/Misc/MiddleWare.cpp | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/Misc/MiddleWare.cpp b/src/Misc/MiddleWare.cpp
index 1bce7737..6a031457 100644
--- a/src/Misc/MiddleWare.cpp
+++ b/src/Misc/MiddleWare.cpp
@@ -1043,8 +1043,8 @@ const rtosc::Ports bankPorts = {
         rBegin;
         auto res = impl.search(rtosc_argument(msg, 0).s);
 #define MAX_SEARCH 300
-        char res_type[MAX_SEARCH+1] = {0};
-        rtosc_arg_t res_dat[MAX_SEARCH] = {0};
+        char res_type[MAX_SEARCH+1] = {};
+        rtosc_arg_t res_dat[MAX_SEARCH] = {};
         for(unsigned i=0; i<res.size() && i<MAX_SEARCH; ++i) {
             res_type[i]  = 's';
             res_dat[i].s = res[i].c_str();
@@ -1056,8 +1056,8 @@ const rtosc::Ports bankPorts = {
         rBegin;
         auto res = impl.blist(rtosc_argument(msg, 0).s);
 #define MAX_SEARCH 300
-        char res_type[MAX_SEARCH+1] = {0};
-        rtosc_arg_t res_dat[MAX_SEARCH] = {0};
+        char res_type[MAX_SEARCH+1] = {};
+        rtosc_arg_t res_dat[MAX_SEARCH] = {};
         for(unsigned i=0; i<res.size() && i<MAX_SEARCH; ++i) {
             res_type[i]  = 's';
             res_dat[i].s = res[i].c_str();
@@ -1354,7 +1354,7 @@ static rtosc::Ports middwareSnoopPorts = {
     //    //cc-id, path, min, max
 //#define MAX_MIDI 32
     //    rtosc_arg_t args[MAX_MIDI*4];
-    //    char        argt[MAX_MIDI*4+1] = {0};
+    //    char        argt[MAX_MIDI*4+1] = {};
     //    int j=0;
     //    for(unsigned i=0; i<key.size() && i<MAX_MIDI; ++i) {
     //        auto val = midi.inv_map[key[i]];
-- 
2.47.0

