From 6738914755d76aa0a7eec0404565a421c1aafa6b Mon Sep 17 00:00:00 2001
From: Johannes Lorenz <johannes89@ist-einmalig.de>
Date: Sun, 12 Nov 2017 01:29:32 +0100
Subject: [PATCH 1225/1936] Update submodule rtosc

---
 rtosc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rtosc b/rtosc
index 74d15bf8..c6ca9d21 160000
--- a/rtosc
+++ b/rtosc
@@ -1 +1 @@
-Subproject commit 74d15bf80421007e8ce11c40b5b84635ee249a9b
+Subproject commit c6ca9d2124c0cbc314db7d3f194ec4bea5fedf63
-- 
2.47.0

