From c4d04c5aa93432c0e7b2052401e35d77c21d7895 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sun, 21 Jan 2018 13:52:47 -0500
Subject: [PATCH 1246/1936] Bank: Fix const issue on Apple build

---
 src/Misc/Bank.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/Misc/Bank.cpp b/src/Misc/Bank.cpp
index b9f01cb1..f2c18735 100644
--- a/src/Misc/Bank.cpp
+++ b/src/Misc/Bank.cpp
@@ -37,7 +37,7 @@
 #ifdef __APPLE__
 #include <dlfcn.h>
 
-static char* my_dylib_path () {
+static const char* my_dylib_path () {
     static Dl_info info;
     if (dladdr((const void*) &my_dylib_path, &info)) {
         return info.dli_fname;
@@ -391,7 +391,7 @@ void Bank::rescanforbanks()
 #endif
 #ifdef __APPLE__
    {
-        char* path = my_dylib_path ();
+        const char* path = my_dylib_path ();
         if (path && strstr(path, "ZynAddSubFX.dylib") && strlen (path) < 1000) {
             char tmp[1024];
             strcpy (tmp, path);
-- 
2.47.0

