From 9a7be8c75448418852ae37476d5ec734ceafd940 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sat, 3 Mar 2018 14:01:28 -0500
Subject: [PATCH 1252/1936] Move instruments repostitory to github

---
 .gitmodules | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/.gitmodules b/.gitmodules
index 7c4c1e96..abd2cee6 100644
--- a/.gitmodules
+++ b/.gitmodules
@@ -1,6 +1,6 @@
 [submodule "instruments"]
 	path = instruments
-	url = git://git.code.sf.net/p/zynaddsubfx/instruments
+	url = https://github.com/zynaddsubfx/instruments
 [submodule "rtosc"]
 	path = rtosc
 	url = https://github.com/fundamental/rtosc
-- 
2.47.0

