From 2400e883942b64c468849b17707082d57774a39f Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sun, 11 Mar 2018 11:09:45 -0400
Subject: [PATCH 1255/1936] Microtonal: Add PAfreq range metadata

---
 src/Misc/Microtonal.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/Misc/Microtonal.cpp b/src/Misc/Microtonal.cpp
index 7fcff3e3..1e09884c 100644
--- a/src/Misc/Microtonal.cpp
+++ b/src/Misc/Microtonal.cpp
@@ -51,6 +51,7 @@ const rtosc::Ports Microtonal::ports = {
     rParamZyn(PAnote, rShort("1/1 midi note"), rDefault(69),
         "The note for 'A'"),
     rParamF(PAfreq,   rShort("ref freq"), rDefault(440.0f),
+         rLog(1.0,20000.0),
         "Frequency of the 'A' note"),
     rParamZyn(Pscaleshift, rShort("shift"), rDefault(64),
         "UNDOCUMENTED"),
-- 
2.47.0

