From 1ae86177a7b0c885cd3c3f5142e4a7ad380b798f Mon Sep 17 00:00:00 2001
From: Joseph Herlant <herlantj@gmail.com>
Date: Wed, 14 Mar 2018 17:36:26 -0700
Subject: [PATCH 1257/1936] Move the generation of chunked html pages to
 asciidoctor

---
 doc/Makefile | 16 +++++-----------
 1 file changed, 5 insertions(+), 11 deletions(-)

diff --git a/doc/Makefile b/doc/Makefile
index a81b571e..acc2b319 100644
--- a/doc/Makefile
+++ b/doc/Makefile
@@ -13,21 +13,15 @@ xhtml: zynaddsubfx.txt envelope.txt intro.txt lfo.txt
 		zynaddsubfx.txt
 
 chunked:
-	asciidoctor --doctype=article   -b docbook5 \
-	          zynaddsubfx.txt
 	rm -rf ./zynaddsubfx.chunked
 	mkdir ./zynaddsubfx.chunked
 	rm -f "./zynaddsubfx.chunked/*.html"
 	ln -s ../images/ zynaddsubfx.chunked/images
-	xsltproc --stringparam html.stylesheet "./docbook-xsl.css" \
-	         --stringparam callout.graphics 0 \
-		 --stringparam navig.graphics 0 \
-		 --stringparam admon.textlabel 1 \
-		 --stringparam admon.graphics 0 \
-		 --nonet \
-	         --stringparam base.dir "zynaddsubfx.chunked/" \
-	         "/etc/asciidoc/docbook-xsl/chunked.xsl" \
-		 "./zynaddsubfx.xml" \
+	cp *.txt zynaddsubfx.chunked/
+	@curl https://raw.githubusercontent.com/asciidoctor/asciidoctor-extensions-lab/master/lib/multipage-html5-converter.rb -o ./zynaddsubfx.chunked/multipage-html5-converter.rb
+	cd ./zynaddsubfx.chunked; asciidoctor -r ./multipage-html5-converter.rb -b multipage_html5 zynaddsubfx.txt
+	ln -s zynaddsubfx-chunked.html ./zynaddsubfx.chunked/index.html
+	rm ./zynaddsubfx.chunked/multipage-html5-converter.rb ./zynaddsubfx.chunked/*.txt
 
 pdf:
 	asciidoctor-pdf zynaddsubfx.txt
-- 
2.47.0

