From 95a4588a68d24e1fa007efde14d8f7a86b01a0c4 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Fri, 6 Apr 2018 23:37:48 -0400
Subject: [PATCH 1260/1936] Allocator: Fix allocating 0 size array

---
 src/Misc/Allocator.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Misc/Allocator.h b/src/Misc/Allocator.h
index 8b1011e8..8274f2e3 100644
--- a/src/Misc/Allocator.h
+++ b/src/Misc/Allocator.h
@@ -61,7 +61,7 @@ class Allocator
         T *valloc(size_t len, Ts&&... ts)
         {
             T *data = (T*)alloc_mem(len*sizeof(T));
-            if(!data) {
+            if(!data && len != 0) {
                 rollbackTransaction();
                 throw std::bad_alloc();
             }
-- 
2.47.0

