From 7d0b3d9a8dddb47c4e8c07fd6aae692e670d3c20 Mon Sep 17 00:00:00 2001
From: nick87720z <nick87720z@gmail.com>
Date: Fri, 25 May 2018 18:21:41 +0500
Subject: [PATCH 1265/1936] gitmodules: fix unsecured url

---
 .gitmodules | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/.gitmodules b/.gitmodules
index abd2cee6..e0423424 100644
--- a/.gitmodules
+++ b/.gitmodules
@@ -7,4 +7,4 @@
 	branch = default_values
 [submodule "DPF"]
 	path = DPF
-	url = git://github.com/DISTRHO/DPF
+	url = https://github.com/DISTRHO/DPF
-- 
2.47.0

