From 8216cf975ed808e6222b23fc7820600b433f94b5 Mon Sep 17 00:00:00 2001
From: Johannes Lorenz <j.git@lorenz-ho.me>
Date: Sun, 3 Jun 2018 18:53:03 +0200
Subject: [PATCH 1268/1936] Update rtosc: Make port-checker optional

---
 rtosc                    | 2 +-
 src/Tests/CMakeLists.txt | 6 ++++--
 2 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/rtosc b/rtosc
index 161679aa..ed4c8c7f 160000
--- a/rtosc
+++ b/rtosc
@@ -1 +1 @@
-Subproject commit 161679aabfef4c4ee38e1ed81d1d4f723a508d7d
+Subproject commit ed4c8c7fe445d9a576ee709c40299422f70f1f42
diff --git a/src/Tests/CMakeLists.txt b/src/Tests/CMakeLists.txt
index 43b269e2..aeeddd2c 100644
--- a/src/Tests/CMakeLists.txt
+++ b/src/Tests/CMakeLists.txt
@@ -69,8 +69,10 @@ target_link_libraries(EffectTest ${test_lib})
 add_executable(ins-test InstrumentStats.cpp)
 target_link_libraries(ins-test ${test_lib} rt)
 
-cp_script(check-ports.rb)
-add_test(PortChecker check-ports.rb)
+if(LIBLO_FOUND)
+    cp_script(check-ports.rb)
+    add_test(PortChecker check-ports.rb)
+endif()
 
 add_executable(save-osc SaveOSC.cpp)
 target_link_libraries(save-osc
-- 
2.47.0

