From ff7af7cfda2b66c53aa6138c8605daebb43a389d Mon Sep 17 00:00:00 2001
From: Patrick Desaulniers <desaulniers.patrick@carrefour.cegepvicto.ca>
Date: Fri, 13 Jul 2018 22:48:01 -0400
Subject: [PATCH 1275/1936] Save and load slot names

---
 src/Misc/Master.cpp | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/Misc/Master.cpp b/src/Misc/Master.cpp
index 9babbcc3..2097cc0e 100644
--- a/src/Misc/Master.cpp
+++ b/src/Misc/Master.cpp
@@ -639,6 +639,7 @@ void Master::saveAutomation(XMLwrapper &xml, const rtosc::AutomationMgr &midi)
             xml.beginbranch("slot", i);
             XmlNode params("params");
             params["midi-cc"] = to_s(slot.midi_cc);
+            params["name"] = to_s(slot.name);
             xml.add(params);
             for(int j=0; j<midi.per_slot; ++j) {
                 const auto &au = slot.automations[j];
@@ -689,8 +690,16 @@ void Master::loadAutomation(XMLwrapper &xml, rtosc::AutomationMgr &midi)
                     }
                 }
                 for(auto node:xml.getBranch())
+                {
                     if(node.name == "params")
+                    {
                         slot.midi_cc = atoi(node["midi-cc"].c_str());
+                        if(node["name"] != "")
+                        {
+                            strncpy(slot.name, node["name"].c_str(), sizeof(slot.name) - 1);
+                        }
+                    }
+                }
                 xml.exitbranch();
             }
         }
-- 
2.47.0

