From e7a281d5a6dab00ecf47e223fe54e4440ce84e79 Mon Sep 17 00:00:00 2001
From: Johannes Lorenz <j.git@lorenz-ho.me>
Date: Sat, 21 Jul 2018 03:25:48 +0200
Subject: [PATCH 1276/1936] Move to new rtosc, tests run again

---
 rtosc                   |  2 +-
 src/Misc/MiddleWare.cpp | 39 +++++++++++++++++++--------------------
 2 files changed, 20 insertions(+), 21 deletions(-)

diff --git a/rtosc b/rtosc
index fdd628fe..006b11e5 160000
--- a/rtosc
+++ b/rtosc
@@ -1 +1 @@
-Subproject commit fdd628fefd2d80610619c041885455d299f2dc5a
+Subproject commit 006b11e50e13e875176f7a7e916abb723b856995
diff --git a/src/Misc/MiddleWare.cpp b/src/Misc/MiddleWare.cpp
index 9883f0cc..4c9971b2 100644
--- a/src/Misc/MiddleWare.cpp
+++ b/src/Misc/MiddleWare.cpp
@@ -117,25 +117,24 @@ static int handler_function(const char *path, const char *types, lo_arg **argv,
     lo_message_serialise(msg, path, buffer, &size);
 
     if(!strcmp(buffer, "/path-search") &&
-       !strcmp("ss", rtosc_argument_string(buffer))) {
-        auto reply_cb = [](const char* url, const char* types, const rtosc_arg_t* args)
-        {
-            char buffer[1024*20];
-            size_t length = rtosc_amessage(buffer, sizeof(buffer),
-                                           "/paths", types, args);
-            if(length) {
-                lo_message msg  = lo_message_deserialise((void*)buffer,
-                                                         length, NULL);
-                lo_address addr = lo_address_new_from_url(url);
-                if(addr)
-                    lo_send_message(addr, buffer, msg);
-                lo_address_free(addr);
-                lo_message_free(msg);
-            }
-        };
-        rtosc::path_search(Master::ports, buffer, mw->activeUrl().c_str(),
-                           reply_cb);
-    } else if(buffer[0]=='/' && strrchr(buffer, '/')[1]) {
+       !strcmp("ss", rtosc_argument_string(buffer)))
+    {
+        char reply_buffer[1024*20];
+        std::size_t length =
+            rtosc::path_search(Master::ports, buffer, 128,
+                               reply_buffer, sizeof(reply_buffer));
+        if(length) {
+            lo_message msg  = lo_message_deserialise((void*)reply_buffer,
+                                                     length, NULL);
+            lo_address addr = lo_address_new_from_url(mw->activeUrl().c_str());
+            if(addr)
+                lo_send_message(addr, reply_buffer, msg);
+            lo_address_free(addr);
+            lo_message_free(msg);
+        }
+    }
+    else if(buffer[0]=='/' && strrchr(buffer, '/')[1])
+    {
         mw->transmitMsg(rtosc::Ports::collapsePath(buffer));
     }
 
@@ -1724,7 +1723,7 @@ void MiddleWareImpl::heartBeat(Master *master)
     //Last provided beat
     //Last acknowledged beat
     //Current offline status
-    
+
     struct timespec time;
     monotonic_clock_gettime(&time);
     uint32_t now = (time.tv_sec-start_time_sec)*100 +
-- 
2.47.0

