From 264341ac588cea186b82a598451a3ef33a7dbfe9 Mon Sep 17 00:00:00 2001
From: Daniel Sheeler <dsheeler@pobox.com>
Date: Sun, 9 Sep 2018 11:26:26 -0500
Subject: [PATCH 1280/1936] fixed firsttime error for legato notes

---
 src/Synth/PADnote.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Synth/PADnote.cpp b/src/Synth/PADnote.cpp
index 5140ac09..57c64eef 100644
--- a/src/Synth/PADnote.cpp
+++ b/src/Synth/PADnote.cpp
@@ -73,7 +73,7 @@ void PADnote::setup(float freq,
         BendAdjust = BendAdj / 24.0f;
     float offset_val = (pars.POffsetHz - 64)/64.0f;
     OffsetHz = 15.0f*(offset_val * sqrtf(fabsf(offset_val)));
-    firsttime = true;
+    if(!legato) firsttime = true;
     realfreq  = basefreq;
     if(!legato)
         NoteGlobalPar.Detune = getdetune(pars.PDetuneType, pars.PCoarseDetune,
-- 
2.47.0

