From 0134afc2ae315364c09e03e64a6d582362af6728 Mon Sep 17 00:00:00 2001
From: Olivier Humbert <trebmuh@users.noreply.github.com>
Date: Wed, 12 Sep 2018 23:19:28 +0200
Subject: [PATCH 1283/1936] Update zynaddsubfx-jack-multi.desktop

---
 zynaddsubfx-jack-multi.desktop | 1 +
 1 file changed, 1 insertion(+)

diff --git a/zynaddsubfx-jack-multi.desktop b/zynaddsubfx-jack-multi.desktop
index 1db7e2d6..39e56735 100644
--- a/zynaddsubfx-jack-multi.desktop
+++ b/zynaddsubfx-jack-multi.desktop
@@ -1,5 +1,6 @@
 [Desktop Entry]
 Name=ZynAddSubFX - Jack (multi channel)
+Name=ZynAddSubFX - Jack (multi canaux)
 Comment=A powerful realtime software synthesizer
 Comment[fr]=Un synthétiseur logiciel temps-réel puissant
 Comment[pl]=Funkcjonalny syntezator wirtualny czasu rzeczywistego
-- 
2.47.0

