From 59823b4296c6f5f4c365bf04b5c1424bbe70304d Mon Sep 17 00:00:00 2001
From: Olivier Humbert <trebmuh@users.noreply.github.com>
Date: Wed, 12 Sep 2018 23:20:05 +0200
Subject: [PATCH 1284/1936] Update zynaddsubfx-jack-multi.desktop

---
 zynaddsubfx-jack-multi.desktop | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/zynaddsubfx-jack-multi.desktop b/zynaddsubfx-jack-multi.desktop
index 39e56735..d44d88b6 100644
--- a/zynaddsubfx-jack-multi.desktop
+++ b/zynaddsubfx-jack-multi.desktop
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Name=ZynAddSubFX - Jack (multi channel)
-Name=ZynAddSubFX - Jack (multi canaux)
+Name[fr]=ZynAddSubFX - Jack (multi canaux)
 Comment=A powerful realtime software synthesizer
 Comment[fr]=Un synthétiseur logiciel temps-réel puissant
 Comment[pl]=Funkcjonalny syntezator wirtualny czasu rzeczywistego
-- 
2.47.0

