From 96795dbf6b1e21f768e9cc39eb7df7df4a492516 Mon Sep 17 00:00:00 2001
From: Johannes Lorenz <j.git@lorenz-ho.me>
Date: Sat, 22 Sep 2018 23:17:32 +0200
Subject: [PATCH 1285/1936] Remove invalid branch name in .gitmodules

---
 .gitmodules | 1 -
 1 file changed, 1 deletion(-)

diff --git a/.gitmodules b/.gitmodules
index e0423424..dc4ce98e 100644
--- a/.gitmodules
+++ b/.gitmodules
@@ -4,7 +4,6 @@
 [submodule "rtosc"]
 	path = rtosc
 	url = https://github.com/fundamental/rtosc
-	branch = default_values
 [submodule "DPF"]
 	path = DPF
 	url = https://github.com/DISTRHO/DPF
-- 
2.47.0

