From b6736b315263909618660b03c3c240e2d7dc5370 Mon Sep 17 00:00:00 2001
From: Johannes Lorenz <j.git@lorenz-ho.me>
Date: Wed, 26 Sep 2018 16:09:14 +0200
Subject: [PATCH 1286/1936] Update rtosc

---
 rtosc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rtosc b/rtosc
index 006b11e5..487d16b4 160000
--- a/rtosc
+++ b/rtosc
@@ -1 +1 @@
-Subproject commit 006b11e50e13e875176f7a7e916abb723b856995
+Subproject commit 487d16b4d9809be8fa463450683101fb39399b15
-- 
2.47.0

