From 2395a052c3a412b757bfa397c51b160dd1390646 Mon Sep 17 00:00:00 2001
From: Olivier Humbert <trebmuh@users.noreply.github.com>
Date: Wed, 26 Sep 2018 21:15:22 +0200
Subject: [PATCH 1287/1936] adds a third location where to check for the
 zyn-fusion binary

---
 src/main.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/main.cpp b/src/main.cpp
index 60b721f4..871219a9 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -681,6 +681,7 @@ int wmidi = -1;
                 }
             }
             exec_fusion("./zyn-fusion");
+            exec_fusion("/opt/zyn-fusion/zyn-fusion");
             exec_fusion("zyn-fusion");
             err(1,"Failed to launch Zyn-Fusion");
         }
-- 
2.47.0

