From 90d9f3134b962a37b23acbf812dd7284a9b2154e Mon Sep 17 00:00:00 2001
From: Daniel Sheeler <dsheeler@pobox.com>
Date: Sun, 30 Sep 2018 07:00:46 -0500
Subject: [PATCH 1291/1936] overzealous renaming

---
 src/Params/ADnoteParameters.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Params/ADnoteParameters.cpp b/src/Params/ADnoteParameters.cpp
index 2b7580cc..b13ba9fc 100644
--- a/src/Params/ADnoteParameters.cpp
+++ b/src/Params/ADnoteParameters.cpp
@@ -222,7 +222,7 @@ static const Ports voicePorts = {
                 obj->PCoarseDetune = k + (obj->PCoarseDetune/1024)*1024;
             }
         }},
-    {"PFMVolume::i", rShort("vol.") rLinear(0,127)
+    {"FMVolume::i", rShort("vol.") rLinear(0,127)
         rDoc("Modulator Magnitude"), NULL,
         [](const char *msg, RtData &d)
         {
-- 
2.47.0

