From ac451cbd554714cc95a778517d307e3900287077 Mon Sep 17 00:00:00 2001
From: Johannes Lorenz <j.git@lorenz-ho.me>
Date: Sat, 13 Oct 2018 08:24:19 +0200
Subject: [PATCH 1295/1936] Fix bash completion install path

---
 cmake/BashCompletion.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/BashCompletion.cmake b/cmake/BashCompletion.cmake
index 0dc01617..769f7678 100644
--- a/cmake/BashCompletion.cmake
+++ b/cmake/BashCompletion.cmake
@@ -79,7 +79,7 @@ if [ -w \"${BASHCOMP_PKG_PATH}\" ]; then\n\
 fi\n\
 echo -e \"\\nInstalling bash completion...\\n\"\n\
 mkdir -p \"\$BASHCOMP_PKG_PATH\"\n\
-cp \"${CMAKE_CURRENT_SOURCE_DIR}/${SCRIPT_NAME}\" \"\$BASHCOMP_PKG_PATH\"\n\
+cp \"${CMAKE_CURRENT_BINARY_DIR}/${SCRIPT_NAME}\" \"\$BASHCOMP_PKG_PATH\"\n\
 chmod a+r \"\$BASHCOMP_PKG_PATH/${SCRIPT_NAME}\"\n\
 echo -e \"Bash completion for ${SCRIPT_NAME} has been installed to \$BASHCOMP_PKG_PATH/${SCRIPT_NAME}\"\n\
 ")
-- 
2.47.0

