From ac513137dd1482b59df1ea38a6ae1cb48224703d Mon Sep 17 00:00:00 2001
From: Patrick Desaulniers <desaulniers.patrick@carrefour.cegepvicto.ca>
Date: Mon, 10 Dec 2018 22:22:00 -0500
Subject: [PATCH 1298/1936] ran => random

---
 src/Params/LFOParams.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Params/LFOParams.cpp b/src/Params/LFOParams.cpp
index d8a8ceb8..31e8d41b 100644
--- a/src/Params/LFOParams.cpp
+++ b/src/Params/LFOParams.cpp
@@ -57,7 +57,7 @@ static const rtosc::Ports _ports = {
               rDefaultDepends(loc), rDefault(64), rPreset(ad_voice_freq, 0),
               "Starting Phase"),
     rOption(PLFOtype, rShort("type"), rOptions(sine, triangle, square, up, down,
-                exp1, exp2, ran), rDefault(sine), "Shape of LFO"),
+                exp1, exp2, random), rDefault(sine), "Shape of LFO"),
     rParamZyn(Prandomness, rShort("a.r."), rSpecial(disable), rDefault(0),
             "Amplitude Randomness (calculated uniformly at each cycle)"),
     rParamZyn(Pfreqrand, rShort("f.r."), rSpecial(disable), rDefault(0),
-- 
2.47.0

