From 56ffc34b3ae3362e7ee1dc947f3e53f918dfec89 Mon Sep 17 00:00:00 2001
From: Patrick Desaulniers <desaulniers.patrick@carrefour.cegepvicto.ca>
Date: Mon, 17 Dec 2018 17:40:59 -0500
Subject: [PATCH 1299/1936] Add default metadata to ADDSynth's coarse note
 detune

---
 src/Params/ADnoteParameters.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Params/ADnoteParameters.cpp b/src/Params/ADnoteParameters.cpp
index 51f3b282..49cf99dc 100644
--- a/src/Params/ADnoteParameters.cpp
+++ b/src/Params/ADnoteParameters.cpp
@@ -207,7 +207,7 @@ static const Ports voicePorts = {
                 obj->PCoarseDetune = k*1024 + obj->PCoarseDetune%1024;
             }
         }},
-    {"coarsedetune::c:i", rProp(parameter) rShort("coarse") rLinear(-64,63)
+    {"coarsedetune::c:i", rProp(parameter) rShort("coarse") rLinear(-64,63) rDefault(0)
         rDoc("Coarse note detune"), NULL,
         [](const char *msg, RtData &d)
         {
-- 
2.47.0

