From acfa27c20e86b5184a024621df76e0d4d35373b1 Mon Sep 17 00:00:00 2001
From: Patrick Desaulniers <desaulniers.patrick@carrefour.cegepvicto.ca>
Date: Wed, 19 Dec 2018 07:07:51 -0500
Subject: [PATCH 1301/1936] Set old filter defaults in UnisonTest

---
 src/Tests/UnisonTest.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/Tests/UnisonTest.h b/src/Tests/UnisonTest.h
index 855febb4..5c4ae88c 100644
--- a/src/Tests/UnisonTest.h
+++ b/src/Tests/UnisonTest.h
@@ -23,6 +23,7 @@
 #include "../Synth/ADnote.h"
 #include "../Synth/OscilGen.h"
 #include "../Params/Presets.h"
+#include "../Params/FilterParams.h"
 #include "../DSP/FFTwrapper.h"
 #include "../globals.h"
 using namespace std;
@@ -64,6 +65,9 @@ class AdNoteTest:public CxxTest::TestSuite
             //sawtooth to make things a bit more interesting
             params->VoicePar[0].OscilSmp->Pcurrentbasefunc = 3;
 
+            params->GlobalPar.PFilterVelocityScale = 64;
+            params->GlobalPar.GlobalFilter->basefreq = 5076.203125;
+
             controller = new Controller(*synth, time);
 
             //lets go with.... 50! as a nice note
-- 
2.47.0

