From 3d4d80eb7a820b10d630cc321d6dd12ad9db813d Mon Sep 17 00:00:00 2001
From: Hans Petter Selasky <hps@selasky.org>
Date: Sun, 3 Feb 2019 20:00:51 +0100
Subject: [PATCH 1304/1936] Fix style.

Signed-off-by: Hans Petter Selasky <hps@selasky.org>
---
 src/Containers/NotePool.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Containers/NotePool.cpp b/src/Containers/NotePool.cpp
index 6a8a8e03..decaa882 100644
--- a/src/Containers/NotePool.cpp
+++ b/src/Containers/NotePool.cpp
@@ -249,7 +249,7 @@ int NotePool::getRunningNotes(void) const
     for(auto &desc:activeDesc()) {
         if(desc.playing() == false && desc.sustained() == false)
             continue;
-        if (running[desc.note] != false)
+        if(running[desc.note] != false)
             continue;
         running[desc.note] = true;
         running_count++;
-- 
2.47.0

