From 47aef93a64a2bf7383f3563ea151ca36e8d0a381 Mon Sep 17 00:00:00 2001
From: Hans Petter Selasky <hps@selasky.org>
Date: Wed, 6 Feb 2019 11:24:54 +0100
Subject: [PATCH 1312/1936] Add default reference note value to microtonal
 tooltip.

Signed-off-by: Hans Petter Selasky <hps@selasky.org>
---
 src/UI/MicrotonalUI.fl | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/UI/MicrotonalUI.fl b/src/UI/MicrotonalUI.fl
index f8fb4e56..dc84ca3f 100644
--- a/src/UI/MicrotonalUI.fl
+++ b/src/UI/MicrotonalUI.fl
@@ -202,7 +202,7 @@ if (true) {
  else o->textcolor(FL_BLACK);*/
 
 o->redraw();}
-          tooltip {The "A" note (the reference note for which freq. ("A" freq) is given)} xywh {173 17 65 20} labelfont 1 labelsize 10 align 129 minimum 0 maximum 127 step 1 value 69 textfont 1 textsize 10
+          tooltip {The "A" note (the reference note for which freq. ("A" freq) is given, default=69)} xywh {173 17 65 20} labelfont 1 labelsize 10 align 129 minimum 0 maximum 127 step 1 value 69 textfont 1 textsize 10
           code0 {o->lstep(12);}
           code1 {o->init("PAnote");}
           class Fl_Osc_Counter
-- 
2.47.0

