From 09d5835ec07a8c41675b202bc7ad2d0cda77dbbc Mon Sep 17 00:00:00 2001
From: Daniel Sheeler <dsheeler@pobox.com>
Date: Sun, 17 Mar 2019 22:46:32 -0500
Subject: [PATCH 1338/1936] Remove debug printf statements

---
 src/Misc/Master.cpp | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/src/Misc/Master.cpp b/src/Misc/Master.cpp
index a0397b6d..393db54b 100644
--- a/src/Misc/Master.cpp
+++ b/src/Misc/Master.cpp
@@ -1586,12 +1586,9 @@ void Master::getfromXML(XMLwrapper& xml)
 {
     if (xml.hasparreal("volume")) {
         xml.getparreal("volume", volume);
-        printf("getparreal\n");
     } else {
         setPvolume(xml.getpar127("volume", Pvolume));
-        printf("getpar127\n");
     }
-    printf("volume %f\n", volume);
     setPkeyshift(xml.getpar127("key_shift", Pkeyshift));
     ctl.NRPN.receive = xml.getparbool("nrpn_receive", ctl.NRPN.receive);
 
-- 
2.47.0

