From 00e198e7a7924419525821c4bf32c0505a5e07bb Mon Sep 17 00:00:00 2001
From: Daniel Sheeler <dsheeler@pobox.com>
Date: Tue, 26 Mar 2019 20:35:50 -0500
Subject: [PATCH 1346/1936] maybe could have been a bug?

---
 src/Misc/Master.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/Misc/Master.cpp b/src/Misc/Master.cpp
index 6831534c..d191af77 100644
--- a/src/Misc/Master.cpp
+++ b/src/Misc/Master.cpp
@@ -1312,7 +1312,6 @@ bool Master::AudioOut(float *outr, float *outl)
             outl[i] *= vol;
             outr[i] *= vol;
         }
-        oldVolume = Volume;
     }
     else {
         // No interpolation
@@ -1322,6 +1321,8 @@ bool Master::AudioOut(float *outr, float *outl)
             outr[i] *= vol;
         }
     }
+    oldVolume = Volume;
+
     vuUpdate(outl, outr);
 
     //Shutup if it is asked (with fade-out)
-- 
2.47.0

