From 380c44ae05795431e1fd2b86aaf4d65c0df74c5b Mon Sep 17 00:00:00 2001
From: Daniel Sheeler <dsheeler@pobox.com>
Date: Wed, 27 Mar 2019 10:52:24 -0500
Subject: [PATCH 1348/1936] Remove application of master volume to part VU
 meters

---
 src/Misc/Master.cpp | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/src/Misc/Master.cpp b/src/Misc/Master.cpp
index d191af77..6579c7c7 100644
--- a/src/Misc/Master.cpp
+++ b/src/Misc/Master.cpp
@@ -1019,9 +1019,6 @@ void Master::vuUpdate(const float *outr, const float *outl)
                 if (fabs(outr[i]) > vuoutpeakpartr[npart])
                     vuoutpeakpartr[npart] = fabs(outr[i]);
             }
-            float v = dB2rap(Volume);
-            vuoutpeakpartl[npart] *= v;
-            vuoutpeakpartr[npart] *= v;
         }
         else
         if(fakepeakpart[npart] > 1)
-- 
2.47.0

