From e1090f3f3b8403da2cdaf1ed6a9aa5f68d2da14b Mon Sep 17 00:00:00 2001
From: Friedolino <mkirchn@freenet.de>
Date: Sun, 7 Apr 2019 16:48:45 +0200
Subject: [PATCH 1360/1936] added support for 14 bit nrpn messages to
 automation

---
 src/Misc/Master.cpp | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/Misc/Master.cpp b/src/Misc/Master.cpp
index a439894d..675aa65f 100644
--- a/src/Misc/Master.cpp
+++ b/src/Misc/Master.cpp
@@ -256,6 +256,16 @@ static const Ports slot_ports = {
         } else
             d.reply(d.loc, "i", a.slots[slot].midi_cc);
 
+        rEnd},
+    {"midi-nrpn::i", rProp(parameter) rMap(default, -1) rDoc("Access assigned midi NRPN slot") , 0,
+        rBegin;
+        int slot = d.idx[0];
+        if(rtosc_narguments(msg)) {
+            a.slots[slot].midi_nrpn = rtosc_argument(msg, 0).i;
+            d.broadcast(d.loc, "i", a.slots[slot].midi_nrpn);
+        } else
+            d.reply(d.loc, "i", a.slots[slot].midi_nrpn);
+
         rEnd},
     {"active::T:F",  rProp(parameter) rMap(default, F) rDoc("If Slot is enabled"), 0,
         rBegin;
-- 
2.47.0

