From 696a83d076299e1107280224fe448977df228311 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sat, 25 May 2019 09:52:01 -0400
Subject: [PATCH 1361/1936] Update rtosc submodule

Should fix NRPN support
---
 rtosc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rtosc b/rtosc
index c4f131c9..eed63261 160000
--- a/rtosc
+++ b/rtosc
@@ -1 +1 @@
-Subproject commit c4f131c96e95d23ecbfbc5f1794c5d349ca564c9
+Subproject commit eed63261d3bbf2ff16e3336842eee01da4d41c3c
-- 
2.47.0

