From 783ca72df9efbaedbe9802d874d3b68bf22e1ab2 Mon Sep 17 00:00:00 2001
From: michiboo <chanmickyyun@gmail.com>
Date: Sun, 9 Jun 2019 11:10:37 +0300
Subject: [PATCH 1364/1936] add watch point test to adNote

---
 src/Tests/AdNoteTest.h | 19 +++++++++++++++++--
 1 file changed, 17 insertions(+), 2 deletions(-)

diff --git a/src/Tests/AdNoteTest.h b/src/Tests/AdNoteTest.h
index f661d9f4..896f11ae 100644
--- a/src/Tests/AdNoteTest.h
+++ b/src/Tests/AdNoteTest.h
@@ -25,6 +25,7 @@
 #include "../Params/Presets.h"
 #include "../DSP/FFTwrapper.h"
 #include "../globals.h"
+#include <rtosc/thread-link.h>
 
 using namespace std;
 using namespace zyn;
@@ -35,6 +36,7 @@ SYNTH_T *synth;
 class AdNoteTest:public CxxTest::TestSuite
 {
     public:
+        rtosc::ThreadLink *tr;
         ADnote       *note;
         AbsTime      *time;
         FFTwrapper   *fft;
@@ -42,7 +44,7 @@ class AdNoteTest:public CxxTest::TestSuite
         Controller   *controller;
         Alloc         memory;
         unsigned char testnote;
-
+        WatchManager *w;
         float *outR, *outL;
 
         void setUp() {
@@ -59,6 +61,8 @@ class AdNoteTest:public CxxTest::TestSuite
             for(int i = 0; i < synth->buffersize; ++i)
                 *(outR + i) = 0;
 
+            tr  = new rtosc::ThreadLink(1024,3);
+            w   = new WatchManager(tr);
 
             fft = new FFTwrapper(synth->oscilsize);
             //prepare the default settings
@@ -93,7 +97,7 @@ class AdNoteTest:public CxxTest::TestSuite
             float freq = 440.0f * powf(2.0f, (testnote - 69.0f) / 12.0f);
             SynthParams pars{memory, *controller, *synth, *time, freq, 120, 0, testnote / 12.0f, false, prng()};
 
-            note = new ADnote(defaultPreset, pars);
+            note = new ADnote(defaultPreset, pars,w);
 
         }
 
@@ -145,6 +149,17 @@ class AdNoteTest:public CxxTest::TestSuite
             sampleCount += synth->buffersize;
             TS_ASSERT_DELTA(outL[255], -0.0901f, 0.0001f);
 
+        
+            TS_ASSERT(!tr->hasNext());
+            w->add_watch("out1");
+            note->noteout(outL, outR);
+            sampleCount += synth->buffersize;
+            w->tick();
+            TS_ASSERT(tr->hasNext());
+            TS_ASSERT_EQUALS(string("out1"), tr->read());
+            TS_ASSERT(!tr->hasNext());
+        
+
             while(!note->finished()) {
                 note->noteout(outL, outR);
 #ifdef WRITE_OUTPUT
-- 
2.47.0

