From 506c1009b68df9aa7acf4a9530cf8e2eb2235770 Mon Sep 17 00:00:00 2001
From: michiboo <chanmickyyun@gmail.com>
Date: Tue, 11 Jun 2019 17:19:30 +0300
Subject: [PATCH 1365/1936] add test for second watch point

---
 src/Tests/AdNoteTest.h | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/Tests/AdNoteTest.h b/src/Tests/AdNoteTest.h
index 896f11ae..3ae200ed 100644
--- a/src/Tests/AdNoteTest.h
+++ b/src/Tests/AdNoteTest.h
@@ -132,9 +132,14 @@ class AdNoteTest:public CxxTest::TestSuite
 
             note->releasekey();
 
-
+            TS_ASSERT(!tr->hasNext());
+            w->add_watch("out");
             note->noteout(outL, outR);
             sampleCount += synth->buffersize;
+            w->tick();
+            TS_ASSERT(tr->hasNext());
+            TS_ASSERT_EQUALS(string("out"), tr->read());
+            TS_ASSERT(!tr->hasNext());
             TS_ASSERT_DELTA(outL[255], -0.4688f, 0.0001f);
 
             note->noteout(outL, outR);
-- 
2.47.0

