From eaae0aaf8470db1e7eefab4697104275382a280c Mon Sep 17 00:00:00 2001
From: michiboo <chanmickyyun@gmail.com>
Date: Mon, 24 Jun 2019 18:47:04 +0300
Subject: [PATCH 1378/1936] add line to clear wp after deactivate

---
 src/Synth/WatchPoint.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/Synth/WatchPoint.cpp b/src/Synth/WatchPoint.cpp
index 16278462..ab128940 100644
--- a/src/Synth/WatchPoint.cpp
+++ b/src/Synth/WatchPoint.cpp
@@ -105,11 +105,10 @@ void WatchManager::tick(void)
         if(sample_list[i] >= framesize-1) {
             char        arg_types[MAX_SAMPLE+1] = {0};
             rtosc_arg_t arg_val[MAX_SAMPLE];
-            printf("\n current accum index:%d  \n",sample_list[i]);
+    
             for(int j=0; j<sample_list[i]; ++j) {
                 arg_types[j] = 'f';
                 arg_val[j].f = data_list[i][j];
-                printf("%f ",arg_val[i].f);
             }
             
             write_back->writeArray(active_list[i], arg_types, arg_val);
@@ -124,6 +123,7 @@ void WatchManager::tick(void)
     for(int i=0; i<MAX_WATCH; ++i) {
         if(deactivate[i]) {
             memset(active_list[i], 0, 128);
+            memset(data_list[i], 0, sizeof(float)*128);
             sample_list[i] = 0;
             deactivate[i]  = false;
         }
-- 
2.47.0

