From 3ab91249b12a7b74a900a8381987447be4fd57a3 Mon Sep 17 00:00:00 2001
From: michiboo <chanmickyyun@gmail.com>
Date: Mon, 24 Jun 2019 19:12:44 +0300
Subject: [PATCH 1379/1936] fix minor bug

---
 src/Synth/WatchPoint.cpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/Synth/WatchPoint.cpp b/src/Synth/WatchPoint.cpp
index ab128940..f2e6d13e 100644
--- a/src/Synth/WatchPoint.cpp
+++ b/src/Synth/WatchPoint.cpp
@@ -105,7 +105,6 @@ void WatchManager::tick(void)
         if(sample_list[i] >= framesize-1) {
             char        arg_types[MAX_SAMPLE+1] = {0};
             rtosc_arg_t arg_val[MAX_SAMPLE];
-    
             for(int j=0; j<sample_list[i]; ++j) {
                 arg_types[j] = 'f';
                 arg_val[j].f = data_list[i][j];
@@ -181,7 +180,6 @@ void WatchManager::satisfy(const char *id, float *f, int n)
             data_list[selected][start] = f[i];
             start++;
         }
-
         sample_list[selected] += space;
     }
 }
-- 
2.47.0

