From a7a674bfdf6c783600ec1ab2aef51e4b03a81fed Mon Sep 17 00:00:00 2001
From: michiboo <chanmickyyun@gmail.com>
Date: Mon, 24 Jun 2019 19:17:39 +0300
Subject: [PATCH 1380/1936] minor bug fix

---
 src/Synth/WatchPoint.h | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/Synth/WatchPoint.h b/src/Synth/WatchPoint.h
index 79b06f01..d4d4e92f 100644
--- a/src/Synth/WatchPoint.h
+++ b/src/Synth/WatchPoint.h
@@ -39,10 +39,9 @@ struct WatchManager
     thrlnk *write_back;
     bool    new_active;
     char    active_list[MAX_WATCH][MAX_WATCH_PATH];
-    float   data_list[MAX_SAMPLE][MAX_WATCH];
+    float   data_list[MAX_WATCH][MAX_SAMPLE];
     int     sample_list[MAX_WATCH];
     bool    deactivate[MAX_WATCH];
-    int   accumulate_index[MAX_WATCH];
 
     //External API
     WatchManager(thrlnk *link=0);
-- 
2.47.0

