From 6e838ba43db848a0645e96d8609310b23db7ea7b Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sat, 20 Jul 2019 13:52:47 -0400
Subject: [PATCH 1395/1936] Fix failure to load master volume

---
 src/Misc/Master.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Misc/Master.cpp b/src/Misc/Master.cpp
index 675aa65f..64023cab 100644
--- a/src/Misc/Master.cpp
+++ b/src/Misc/Master.cpp
@@ -1617,7 +1617,7 @@ int Master::loadXML(const char *filename)
 void Master::getfromXML(XMLwrapper& xml)
 {
     if (xml.hasparreal("volume")) {
-        xml.getparreal("volume", Volume);
+        Volume = xml.getparreal("volume", Volume);
     } else {
         Volume  = volume127ToFloat(xml.getpar127("volume", 0));
         oldVolume = Volume;
-- 
2.47.0

