From b5a096640e4c35e1b7ac7a39d5eb2f2391d5087f Mon Sep 17 00:00:00 2001
From: michiboo <chanmickyyun@gmail.com>
Date: Mon, 22 Jul 2019 15:52:17 +0300
Subject: [PATCH 1397/1936] try fix triggerTest

---
 src/Tests/TriggerTest.h | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/src/Tests/TriggerTest.h b/src/Tests/TriggerTest.h
index 0241e223..24aa8dc8 100644
--- a/src/Tests/TriggerTest.h
+++ b/src/Tests/TriggerTest.h
@@ -120,24 +120,21 @@ class TriggerTest:public CxxTest::TestSuite
 #endif
             sampleCount += synth->buffersize;
             note->releasekey();
-
             TS_ASSERT(!tr->hasNext());
             w->add_watch("noteout");
             w->add_watch("noteout1");
             note->noteout(outL, outR);
             sampleCount += synth->buffersize;
             w->tick();
-            TS_ASSERT(w->trigger_active("noteout"));
-            TS_ASSERT(w->trigger_active("noteout1"));
             note->noteout(outL, outR);
             sampleCount += synth->buffersize;
             w->tick();
-
+            TS_ASSERT(w->trigger_active("noteout"));
+            TS_ASSERT(w->trigger_active("noteout1"));
             note->noteout(outL, outR);
             sampleCount += synth->buffersize;
             w->tick();
             TS_ASSERT_EQUALS(string("noteout1"), tr->read());
-            
             note->noteout(outL, outR);
             sampleCount += synth->buffersize;
             w->tick();
-- 
2.47.0

