From 33d2963372c05bb0ce2fb5c64dcea158dd722240 Mon Sep 17 00:00:00 2001
From: michiboo <chanmickyyun@gmail.com>
Date: Tue, 23 Jul 2019 00:44:50 +0300
Subject: [PATCH 1402/1936] fix trigger test3

---
 src/Tests/TriggerTest.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Tests/TriggerTest.h b/src/Tests/TriggerTest.h
index 2225534d..14ecb056 100644
--- a/src/Tests/TriggerTest.h
+++ b/src/Tests/TriggerTest.h
@@ -136,7 +136,7 @@ class TriggerTest:public CxxTest::TestSuite
             note->noteout(outL, outR);
             sampleCount += synth->buffersize;
             w->tick();
-            TS_ASSERT_EQUALS(string("noteout1"), tr->read());
+            TS_ASSERT_EQUALS(string("noteout"), tr->read());
             
             while(!note->finished()) {
                 note->noteout(outL, outR);
-- 
2.47.0

