From 872477e172ae4f9e22d435df3b8cd828448c8ae4 Mon Sep 17 00:00:00 2001
From: michiboo <chanmickyyun@gmail.com>
Date: Mon, 12 Aug 2019 22:40:19 +0300
Subject: [PATCH 1407/1936] fix off by one error in watch point

---
 src/Synth/WatchPoint.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Synth/WatchPoint.cpp b/src/Synth/WatchPoint.cpp
index 45ad0701..ca20b4a4 100644
--- a/src/Synth/WatchPoint.cpp
+++ b/src/Synth/WatchPoint.cpp
@@ -115,7 +115,7 @@ void WatchManager::tick(void)
         int framesize = 2;
         call_count[i] = 0;
         if(strstr(active_list[i], "noteout") != NULL)
-            framesize = MAX_SAMPLE;
+            framesize = MAX_SAMPLE -1;
         if(sample_list[i] >= framesize && call_count[i]==0) {
             char        arg_types[MAX_SAMPLE+1] = {0};
             rtosc_arg_t arg_val[MAX_SAMPLE];
-- 
2.47.0

