From 5a5c179f0de65d4a6cbb47503eed02e55186e87e Mon Sep 17 00:00:00 2001
From: Hans Petter Selasky <hps@selasky.org>
Date: Sat, 14 Sep 2019 13:03:50 +0200
Subject: [PATCH 1424/1936] Remove unused field.

Signed-off-by: Hans Petter Selasky <hps@selasky.org>
---
 src/UI/EnvelopeFreeEdit.h | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/UI/EnvelopeFreeEdit.h b/src/UI/EnvelopeFreeEdit.h
index ce01f7e4..61c8d101 100644
--- a/src/UI/EnvelopeFreeEdit.h
+++ b/src/UI/EnvelopeFreeEdit.h
@@ -58,6 +58,5 @@ class EnvelopeFreeEdit : public Fl_Box, public Fl_Osc_Widget
         char Penvval[MAX_ENVELOPE_POINTS];
         //The Sustain point
         char Penvsustain;
-        int button_state;
         int mod_state;
 };
-- 
2.47.0

