From 914b9b18af61730e4a311dc13f2bd048ecef17ab Mon Sep 17 00:00:00 2001
From: Hans Petter Selasky <hps@selasky.org>
Date: Thu, 2 Jan 2020 19:18:53 +0100
Subject: [PATCH 1459/1936] Fix order of includes when both FLTK and NTK is
 installed.

Signed-off-by: Hans Petter Selasky <hps@selasky.org>
---
 src/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index e93a1430..e7065c98 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -421,7 +421,7 @@ if(NtkGui)
 
 	message(STATUS "Will build NTK gui")
 
-	include_directories(
+	include_directories(BEFORE
 			${NTK_INCLUDE_DIRS}
 			"${CMAKE_CURRENT_SOURCE_DIR}/UI"
 			"${CMAKE_CURRENT_BINARY_DIR}/UI"
-- 
2.47.0

