From 2cdd01ce3a0fc7dbace5aeeae95671bd93325885 Mon Sep 17 00:00:00 2001
From: Hans Petter Selasky <hps@selasky.org>
Date: Thu, 2 Jan 2020 19:31:34 +0100
Subject: [PATCH 1460/1936] Place const keyword correctly.

Signed-off-by: Hans Petter Selasky <hps@selasky.org>
---
 src/Synth/OscilGen.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/Synth/OscilGen.cpp b/src/Synth/OscilGen.cpp
index 97635664..4f76c46f 100644
--- a/src/Synth/OscilGen.cpp
+++ b/src/Synth/OscilGen.cpp
@@ -1632,7 +1632,7 @@ FUNC(circle)
 
 base_func_t *getBaseFunction(unsigned char func)
 {
-    static const base_func_t *functions[] = {
+    static base_func_t * const functions[] = {
         basefunc_triangle,
         basefunc_pulse,
         basefunc_saw,
@@ -1792,7 +1792,7 @@ FILTER(s)
 
 filter_func_t *getFilter(unsigned char func)
 {
-    static const filter_func_t *functions[] = {
+    static filter_func_t * const functions[] = {
         osc_lp,
         osc_hp1,
         osc_hp1b,
-- 
2.47.0

