From ec317c1162c582e8658eeccc736d0e823d9994f3 Mon Sep 17 00:00:00 2001
From: Hans Petter Selasky <hps@selasky.org>
Date: Sat, 22 Feb 2020 10:19:26 +0100
Subject: [PATCH 1469/1936] Remove unused fields from SUBnote class.

Signed-off-by: Hans Petter Selasky <hps@selasky.org>
---
 src/Synth/SUBnote.h | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/Synth/SUBnote.h b/src/Synth/SUBnote.h
index c475ffe8..0f82c378 100644
--- a/src/Synth/SUBnote.h
+++ b/src/Synth/SUBnote.h
@@ -103,9 +103,7 @@ class SUBnote:public SynthNote
         float overtone_freq[MAX_SUB_HARMONICS];
 
         int   oldpitchwheel, oldbandwidth;
-        float globalfiltercenterq;
         float velocity;
-        WatchManager *wm;
 };
 
 }
-- 
2.47.0

