From 1c52964f9fd447b0cecbbfb1c18e6a7d4eebef09 Mon Sep 17 00:00:00 2001
From: nori li <50680474+thenorili@users.noreply.github.com>
Date: Tue, 18 Feb 2020 11:12:41 -0500
Subject: [PATCH 1472/1936] Corrected syntax of Enumerations.H include.

---
 src/UI/MasterUI.fl | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/UI/MasterUI.fl b/src/UI/MasterUI.fl
index f72f513e..54647645 100644
--- a/src/UI/MasterUI.fl
+++ b/src/UI/MasterUI.fl
@@ -94,7 +94,11 @@ decl {\#if !defined(PLUGINVERSION) && HAS_X11
 
 decl {\#if !defined(PLUGINVERSION) && HAS_X11
 \#include <FL/FL.H>
-\#include "Enumerations.H"
+\#endif} {private local
+} 
+
+decl {\#if !defined(PLUGINVERSION) && HAS_X11
+\#include <FL/Enumerations.H>
 \#endif} {private local
 } 
 
-- 
2.47.0

