From 344fe4e479dbb11b056195ab81e4ef00489a8c60 Mon Sep 17 00:00:00 2001
From: Strongheart <strongheart_1@hotmail.com>
Date: Tue, 17 Mar 2020 00:50:06 +0100
Subject: [PATCH 1480/1936] unison-size  minimum: 0 corrected to 1

---
 src/Params/ADnoteParameters.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Params/ADnoteParameters.cpp b/src/Params/ADnoteParameters.cpp
index 0dd9c166..d7bb8420 100644
--- a/src/Params/ADnoteParameters.cpp
+++ b/src/Params/ADnoteParameters.cpp
@@ -71,7 +71,7 @@ static const Ports voicePorts = {
     rRecurp(VoiceFilter,    "Optional Voice Filter"),
 
 //    rToggle(Enabled,       rShort("enable"), "Voice Enable"),
-    rParamI(Unison_size, rShort("size"), rMap(min, 0), rMap(max, 50),
+    rParamI(Unison_size, rShort("size"), rMap(min, 1), rMap(max, 50),
         rDefault(1), "Number of subvoices"),
     rParamZyn(Unison_phase_randomness, rShort("ph.rnd."), rDefault(127),
         "Phase Randomness"),
-- 
2.47.0

