From a7182360270fb152964e6e85c89af8afc6cb6b45 Mon Sep 17 00:00:00 2001
From: Hans Petter Selasky <hps@selasky.org>
Date: Fri, 3 Apr 2020 12:04:12 +0200
Subject: [PATCH 1492/1936] Use floating point constant.

Signed-off-by: Hans Petter Selasky <hps@selasky.org>
---
 src/Params/FilterParams.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Params/FilterParams.cpp b/src/Params/FilterParams.cpp
index 2b805b92..3fe2261f 100644
--- a/src/Params/FilterParams.cpp
+++ b/src/Params/FilterParams.cpp
@@ -469,7 +469,7 @@ float FilterParams::getq() const
 }
 float FilterParams::getfreqtracking(float notefreq) const
 {
-    return log2f(notefreq / 440.0f) * (freqtracking / 100.0);
+    return log2f(notefreq / 440.0f) * (freqtracking / 100.0f);
 }
 
 float FilterParams::getgain() const
-- 
2.47.0

