From 9870a589a3b3b3beaa74eea04066d7b8e5117e46 Mon Sep 17 00:00:00 2001
From: Hans Petter Selasky <hps@selasky.org>
Date: Fri, 3 Apr 2020 13:21:33 +0200
Subject: [PATCH 1493/1936] Remove redundant initializers.

Signed-off-by: Hans Petter Selasky <hps@selasky.org>
---
 src/Synth/ModFilter.cpp | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/Synth/ModFilter.cpp b/src/Synth/ModFilter.cpp
index 40d66716..002da31e 100644
--- a/src/Synth/ModFilter.cpp
+++ b/src/Synth/ModFilter.cpp
@@ -30,7 +30,6 @@ ModFilter::ModFilter(const FilterParams &pars_,
                            bool         stereo,
                            float        notefreq)
     :pars(pars_), synth(synth_), time(time_), alloc(alloc_),
-    baseQ(pars.getq()), baseFreq(pars.getfreq()),
     noteFreq(notefreq),
     left(nullptr), 
     right(nullptr),
-- 
2.47.0

