From 0c54493208d5e61bac82eca6a26f232e544b9995 Mon Sep 17 00:00:00 2001
From: Hans Petter Selasky <hps@selasky.org>
Date: Fri, 3 Apr 2020 13:23:54 +0200
Subject: [PATCH 1495/1936] Implement smart float class which return smooth
 values.

Signed-off-by: Hans Petter Selasky <hps@selasky.org>
---
 src/globals.h | 29 +++++++++++++++++++++++++++++
 1 file changed, 29 insertions(+)

diff --git a/src/globals.h b/src/globals.h
index 363b901e..3955f041 100644
--- a/src/globals.h
+++ b/src/globals.h
@@ -347,5 +347,34 @@ struct SYNTH_T {
     static float numRandom(void); //defined in Util.cpp for now
 };
 
+class smooth_float {
+private:
+    bool init;
+    float curr_value;
+    float next_value;
+public:
+    smooth_float() {
+        init = false;
+	next_value = curr_value = 0.0f;
+    };
+    smooth_float(const float value) {
+        init = true;
+        next_value = curr_value = value;
+    };
+    operator float() {
+        const float delta = (next_value - curr_value) / 128.0f;
+        curr_value += delta;
+        return (curr_value);
+    };
+    void operator =(const float value) {
+      if (init) {
+          next_value = value;
+      } else {
+          next_value = curr_value = value;
+          init = true;
+      }
+    };
+};
+
 }
 #endif
-- 
2.47.0

