From ccd81935af91c32d033d275a2ef9180665dece14 Mon Sep 17 00:00:00 2001
From: Hans Petter Selasky <hps@selasky.org>
Date: Fri, 3 Apr 2020 13:38:14 +0200
Subject: [PATCH 1497/1936] Remove dead code.

Signed-off-by: Hans Petter Selasky <hps@selasky.org>
---
 src/Synth/SUBnote.cpp | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/src/Synth/SUBnote.cpp b/src/Synth/SUBnote.cpp
index e7e4c036..192ba3f4 100644
--- a/src/Synth/SUBnote.cpp
+++ b/src/Synth/SUBnote.cpp
@@ -185,11 +185,6 @@ void SUBnote::setup(float freq,
             initparameters(basefreq / 440.0f * freq, wm, prefix);
     }
     else {
-        if(pars.Pfixedfreq == 0)
-            freq = basefreq;
-        else
-            freq *= basefreq / 440.0f;
-
         if(GlobalFilter)
             GlobalFilter->updateNoteFreq(basefreq);
     }
-- 
2.47.0

