From 980c36087a0275f77bfa328b7f661edbe46f86ad Mon Sep 17 00:00:00 2001
From: Hans Petter Selasky <hps@selasky.org>
Date: Fri, 3 Apr 2020 13:38:40 +0200
Subject: [PATCH 1498/1936] Smooth ModFilter parameters.

Signed-off-by: Hans Petter Selasky <hps@selasky.org>
---
 src/Synth/ModFilter.h | 11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/src/Synth/ModFilter.h b/src/Synth/ModFilter.h
index 2f3b3532..1dca0c63 100644
--- a/src/Synth/ModFilter.h
+++ b/src/Synth/ModFilter.h
@@ -53,12 +53,11 @@ class ModFilter
         Allocator          &alloc; //RT Memory Pool
 
 
-
-        float baseQ;    //filter sharpness
-        float baseFreq; //base filter frequency
-        float noteFreq; //frequency note was initialized to
-        float tracking; //shift due to note frequency
-        float sense;    //shift due to note velocity
+        smooth_float baseQ;    //filter sharpness
+        smooth_float baseFreq; //base filter frequency
+        smooth_float noteFreq; //frequency note was initialized to
+        smooth_float tracking; //shift due to note frequency
+        smooth_float sense;    //shift due to note velocity
 
 
         Filter       *left; //left  channel filter
-- 
2.47.0

