From ebbbfe8a7a69823143ee218d2e49ab0ad3136619 Mon Sep 17 00:00:00 2001
From: Hans Petter Selasky <hps@selasky.org>
Date: Sat, 4 Apr 2020 21:01:26 +0200
Subject: [PATCH 1514/1936] Remove empty line.

Signed-off-by: Hans Petter Selasky <hps@selasky.org>
---
 src/Params/Controller.cpp | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/Params/Controller.cpp b/src/Params/Controller.cpp
index b2144cf7..9c84dcd3 100644
--- a/src/Params/Controller.cpp
+++ b/src/Params/Controller.cpp
@@ -283,7 +283,6 @@ int Controller::initportamento(float oldfreq,
     else     // No legato, do the original if...return
     if((portamento.used != 0) || (portamento.portamento == 0))
         return 0;
-    ;
 
     float portamentotime = powf(100.0f, portamento.time / 127.0f) / 50.0f; //portamento time in seconds
 
-- 
2.47.0

