From 852e6544e6bd42453ddaaebd8b5d74864f0ed84b Mon Sep 17 00:00:00 2001
From: Hans Petter Selasky <hps@selasky.org>
Date: Sun, 5 Apr 2020 13:37:50 +0200
Subject: [PATCH 1521/1936] Fix Advanced/Beginner Selection

Signed-off-by: Hans Petter Selasky <hps@selasky.org>
---
 src/UI/MasterUI.fl | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/UI/MasterUI.fl b/src/UI/MasterUI.fl
index 05347282..dd7d6a74 100644
--- a/src/UI/MasterUI.fl
+++ b/src/UI/MasterUI.fl
@@ -506,7 +506,7 @@ if (fl_choice("The file *might* exist. \\nOverwrite it?","No","Yes",NULL)) {
 	masterwindow->hide();
 	refresh_master_ui();
 	simplemasterwindow->show();
-    osc->write("/cfg.UserInterfaceMode", "i", 2);
+    osc->write("/config/cfg.UserInterfaceMode", "i", 2);
 };}
               xywh {15 15 100 20}
             }
@@ -1072,7 +1072,7 @@ if (result==-10) fl_alert("Error: Could not load the file\\nbecause it is not an
 	simplemasterwindow->hide();
 	refresh_master_ui();
 	masterwindow->show();
-    osc->write("/cfg.UserInterfaceMode", "i", 1);
+    osc->write("/config/cfg.UserInterfaceMode", "i", 1);
 };}
             xywh {0 0 100 20}
           }
-- 
2.47.0

