From 307bc45c51e87647e448490565b0ebd11e0e454c Mon Sep 17 00:00:00 2001
From: Friedolino <mkirchn@freenet.de>
Date: Sun, 28 Jun 2020 21:48:33 +0200
Subject: [PATCH 1544/1936] print the filename to console at savexml port

---
 src/Misc/Part.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/Misc/Part.cpp b/src/Misc/Part.cpp
index 7773efd6..46d280dc 100644
--- a/src/Misc/Part.cpp
+++ b/src/Misc/Part.cpp
@@ -176,6 +176,7 @@ static const Ports partPorts = {
                 strftime (p->loaded_file,23,"%F_%R.xiz",timeinfo); 
             }
             p->saveXML(p->loaded_file);
+            fprintf(stderr, "Part %d saved to %s\n", (p->partno + 1), p->loaded_file);
         }},
     //{"kit#16::T:F", "::Enables or disables kit item", 0,
     //    [](const char *m, RtData &d) {
-- 
2.47.0

