From 4e1ac7aff9cddff829afc275e51caa39ef9d5eb0 Mon Sep 17 00:00:00 2001
From: Carlo Bramini <30959007+carlo-bramini@users.noreply.github.com>
Date: Wed, 8 Jul 2020 16:02:32 +0200
Subject: [PATCH 1576/1936] [CMAKE] Remove duplicated check on PORTAUDIO.

---
 src/CMakeLists.txt | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 804831f3..9659f2f0 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -34,8 +34,6 @@ if(PKG_CONFIG_FOUND AND NOT (${CMAKE_SYSTEM_NAME} STREQUAL "Windows"))
     pkg_check_modules(FFTW REQUIRED fftw3)
     pkg_check_modules(MXML REQUIRED mxml)
 
-    pkg_check_modules(PORTAUDIO portaudio-2.0>=19)
-
     pkg_search_module(LASH lash-1.0)
     mark_as_advanced(LASH_LIBRARIES)
     pkg_search_module(DSSI dssi>=0.9.0)
-- 
2.47.0

