From d2672b50ac3fac511f295251f630561f9d316bf6 Mon Sep 17 00:00:00 2001
From: Friedolino <mkirchn@freenet.de>
Date: Fri, 2 Oct 2020 22:37:33 +0200
Subject: [PATCH 1594/1936] fix of fixed savexml port

---
 src/Misc/Part.cpp | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/src/Misc/Part.cpp b/src/Misc/Part.cpp
index afdaf473..786f916e 100644
--- a/src/Misc/Part.cpp
+++ b/src/Misc/Part.cpp
@@ -176,9 +176,14 @@ static const Ports partPorts = {
                 time (&rawtime);
                 const struct tm* timeinfo = localtime (&rawtime);
                 strftime (filename,23,"%F_%R.xiz",timeinfo); 
+                p->saveXML(filename);
+                fprintf(stderr, "Part %d saved to %s\n", (p->partno + 1), filename);
+            }
+            else
+            {
+                p->saveXML(p->loaded_file);
+                fprintf(stderr, "Part %d saved to %s\n", (p->partno + 1), p->loaded_file);
             }
-            p->saveXML(p->loaded_file);
-            fprintf(stderr, "Part %d saved to %s\n", (p->partno + 1), p->loaded_file);
         }},
     //{"kit#16::T:F", "::Enables or disables kit item", 0,
     //    [](const char *m, RtData &d) {
@@ -1092,8 +1097,6 @@ int Part::saveXML(const char *filename)
     xml.endbranch();
 
     int result = xml.saveXMLfile(filename, gzip_compression);
-    strncpy(loaded_file,filename, sizeof(loaded_file));
-    loaded_file[sizeof(loaded_file)-1] = '\0';
     return result;
 }
 
-- 
2.47.0

