From 2f5c978e0cb47461fa3c1e1bf2b132a8da7bd16e Mon Sep 17 00:00:00 2001
From: Johannes Lorenz <j.git@lorenz-ho.me>
Date: Sun, 25 Oct 2020 19:15:56 +0100
Subject: [PATCH 1639/1936] Remove unused MultiPseudoStack::dump

---
 src/Containers/MultiPseudoStack.h | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/Containers/MultiPseudoStack.h b/src/Containers/MultiPseudoStack.h
index 22e59de7..e9e63e11 100644
--- a/src/Containers/MultiPseudoStack.h
+++ b/src/Containers/MultiPseudoStack.h
@@ -11,7 +11,6 @@
 */
 #pragma once
 #include <atomic>
-#include <cassert>
 
 namespace zyn {
 
@@ -58,7 +57,6 @@ class MultiQueue
 public:
     MultiQueue(void);
     ~MultiQueue(void);
-    void dump(void);
     //! return a fresh item, e.g. to write to
     qli_t *alloc(void)   { return m_free.read();   }
     //! free an item
-- 
2.47.0

