From 0c0ffda19245e064e156ac74a58b656eded675eb Mon Sep 17 00:00:00 2001
From: Johannes Lorenz <j.git@lorenz-ho.me>
Date: Sun, 25 Oct 2020 19:22:13 +0100
Subject: [PATCH 1640/1936] Master.h: add comments about handling OSC events

---
 src/Misc/Master.h | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/Misc/Master.h b/src/Misc/Master.h
index ae58365e..91108c5d 100644
--- a/src/Misc/Master.h
+++ b/src/Misc/Master.h
@@ -125,6 +125,8 @@ class Master
         void vuUpdate(const float *outl, const float *outr);
 
         //Process a set of OSC events in the bToU buffer
+        //This may be called by MiddleWare if we are offline
+        //(in this case, the param offline is true)
         bool runOSC(float *outl, float *outr, bool offline=false,
                     Master* master_from_mw = nullptr);
 
@@ -248,8 +250,10 @@ class Master
         //Used by loadOSC and saveOSC
         int loadOSCFromStr(const char *file_content,
                            rtosc::savefile_dispatcher_t* dispatcher);
-        //!applyOscEvent with a DataObj parameter
-        //!@return false iff master has been changed
+        //! apply an OSC event with a DataObj parameter
+        //! @note This may be called by MiddleWare if we are offline
+        //!   (in this case, the param offline is true)
+        //! @return false iff master has been changed
         bool applyOscEvent(const char *event, float *outl, float *outr,
                            bool offline, bool nio,
                            class DataObj& d, int msg_id = -1,
-- 
2.47.0

