From 1d277cd6d449266aba32639f913791bd5a29b946 Mon Sep 17 00:00:00 2001
From: Johannes Lorenz <j.git@lorenz-ho.me>
Date: Sat, 31 Oct 2020 15:19:11 +0100
Subject: [PATCH 1644/1936] MessageTest: Add more Copy/Paste checks

---
 src/Tests/MessageTest.h | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/src/Tests/MessageTest.h b/src/Tests/MessageTest.h
index e74a8b8c..5e644735 100644
--- a/src/Tests/MessageTest.h
+++ b/src/Tests/MessageTest.h
@@ -99,6 +99,23 @@ class MessageTest:public CxxTest::TestSuite
             start_realtime();
             mw->transmitMsg("/presets/copy", "s", "/part0/kit0/adpars/VoicePar0/FMSmp/");
 
+            TS_ASSERT_EQUALS(mw->getPresetsStore().clipboard.type.c_str(), string("Poscilgen"));
+            // a regex would be better here...
+            // hopefully, mxml will not change its whitespace behavior
+            TS_ASSERT(strstr(mw->getPresetsStore().clipboard.data.c_str(), "<par name=\"base_function_par\" value=\"32\" />"));
+
+            /* // better test this without string comparison:
+            {
+                XMLwrapper xml;
+                bool couldPutXml = xml.putXMLdata(mw->getPresetsStore().clipboard.data.c_str());
+                TS_ASSERT(couldPutXml);
+                unsigned char copiedBasefuncPar = xml.getpar127("base_function_par", 0);
+                TS_ASSERT_EQUALS(copiedBasefuncPar, 32);
+            }*/
+
+            //printf("clipboard type: %s\n",mw->getPresetsStore().clipboard.type.c_str());
+            //printf("clipboard data:\n%s\n",mw->getPresetsStore().clipboard.data.c_str());
+
             TS_ASSERT_EQUALS(osc_dst.Pbasefuncpar, 64);
             TS_ASSERT_EQUALS(osc_oth.Pbasefuncpar, 64);
 
-- 
2.47.0

