From c33cb65e1ad34bb6e8644c3f8e5d0cd0ad4d48be Mon Sep 17 00:00:00 2001
From: Johannes Lorenz <j.git@lorenz-ho.me>
Date: Sat, 7 Nov 2020 22:35:24 +0100
Subject: [PATCH 1645/1936] DynFilter: Fix VocalMorph2, was named VocalMorph1

---
 src/Effects/DynamicFilter.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/Effects/DynamicFilter.cpp b/src/Effects/DynamicFilter.cpp
index 8ed5497e..aef989e4 100644
--- a/src/Effects/DynamicFilter.cpp
+++ b/src/Effects/DynamicFilter.cpp
@@ -27,7 +27,7 @@ namespace zyn {
 #define rEnd }
 
 rtosc::Ports DynamicFilter::ports = {
-    {"preset::i", rOptions(WahWah, AutoWah, Sweep, VocalMorph1, VocalMorph1)
+    {"preset::i", rOptions(WahWah, AutoWah, Sweep, VocalMorph1, VocalMorph2)
                   rDoc("Instrument Presets"), 0,
                   rBegin;
                   rObject *o = (rObject*)d.obj;
@@ -286,7 +286,7 @@ unsigned char DynamicFilter::getpresetpar(unsigned char npreset, unsigned int np
         {100, 64, 30, 0, 0, 50, 80, 0,  0, 60},
         //VocalMorph1
         {110, 64, 80, 0, 0, 64, 0,  64, 0, 60},
-        //VocalMorph1
+        //VocalMorph2
         {127, 64, 50, 0, 0, 96, 64, 0,  0, 60}
     };
     if(npreset < NUM_PRESETS && npar < PRESET_SIZE) {
-- 
2.47.0

