From f0e940cb55d2ea31c804beccbe6ee05ea98e88f3 Mon Sep 17 00:00:00 2001
From: Johannes Lorenz <j.git@lorenz-ho.me>
Date: Sun, 8 Nov 2020 00:23:50 +0100
Subject: [PATCH 1647/1936] Master: Add comments

---
 src/Misc/Master.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/Misc/Master.cpp b/src/Misc/Master.cpp
index 22be28ae..33b0dc45 100644
--- a/src/Misc/Master.cpp
+++ b/src/Misc/Master.cpp
@@ -68,15 +68,15 @@ static const Ports sysefxPort =
             assert(isdigit(*index_1));
             if(isdigit(index_1[-1]))
                 index_1--;
-            int ind1 = atoi(index_1);
+            int ind1 = atoi(index_1); //efx
 
             //Now get the second index like normal
             while(!isdigit(*m)) m++;
-            int ind2 = atoi(m);
+            int ind2 = atoi(m); //part
             Master &mast = *(Master*)d.obj;
 
             if(rtosc_narguments(m)) {
-                mast.setPsysefxvol(ind2, ind1, rtosc_argument(m,0).i);
+                mast.setPsysefxvol(ind2, ind1, rtosc_argument(m,0).i /*vol*/);
                 d.broadcast(d.loc, "i", mast.Psysefxvol[ind1][ind2]);
             } else
                 d.reply(d.loc, "i", mast.Psysefxvol[ind1][ind2]);
-- 
2.47.0

