From 79c5043f43599802029fd8499291ed6b0a907182 Mon Sep 17 00:00:00 2001
From: Johannes Lorenz <j.git@lorenz-ho.me>
Date: Sat, 21 Nov 2020 22:39:23 +0100
Subject: [PATCH 1657/1936] check-ports.rb: Add sleep to let zyn setup

---
 src/Tests/check-ports.rb | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/Tests/check-ports.rb b/src/Tests/check-ports.rb
index 7da771f5..b39a1c0a 100755
--- a/src/Tests/check-ports.rb
+++ b/src/Tests/check-ports.rb
@@ -6,11 +6,15 @@ rval=0
 my_path=File.dirname(__FILE__)
 
 # start zyn, grep the lo server port, and connect the port checker to it
+# NOTE: If you add too much debug output to zyn, it will be blocked when writing into the pipe,
+#       leading to MiddleWare to not reply and to port-checker to fail.
+#       This script should be rewritten to keep reading (and printing) zyn's messages.
 Open3.popen3(my_path + "/../zynaddsubfx -O null --no-gui") do |stdin, stdout, stderr, wait_thr|
   pid = wait_thr[:pid]
   while line=stderr.gets do 
     # print "line: " + line;
     if /^lo server running on (\d+)$/.match(line) then
+      sleep 3 # give zyn more time to setup
       port_checker_rval = system(my_path + "/../../rtosc/port-checker 'osc.udp://localhost:" + $1 + "/'")
       if port_checker_rval != true then
         puts "Error: port-checker has returned #{$?.exitstatus}."
-- 
2.47.0

