From fe7de5f132957a0215b9cee97f4997356ae0a0d4 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Tue, 8 Dec 2020 21:11:08 -0500
Subject: [PATCH 1666/1936] FLTK-GUI: Correct Oscilgen win title for padsynth

---
 src/UI/OscilGenUI.fl | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/UI/OscilGenUI.fl b/src/UI/OscilGenUI.fl
index e3c79857..b7249d3b 100644
--- a/src/UI/OscilGenUI.fl
+++ b/src/UI/OscilGenUI.fl
@@ -990,6 +990,8 @@ basefuncdisplaygroup = NULL;
 bfpar = NULL;
 bftype = NULL;
 make_window();
+if(!adnotep)
+    osceditUI->label("PADsynth Harmonic Content Editor");
 bftype->init("Pcurrentbasefunc");
 initialized = true;
 
-- 
2.47.0

