From 2ab265ba366bd6de1086f21c55cb399f70745e8c Mon Sep 17 00:00:00 2001
From: Johannes Lorenz <j.git@lorenz-ho.me>
Date: Tue, 15 Dec 2020 16:56:11 +0100
Subject: [PATCH 1675/1936] Fix assert

---
 src/UI/FormantFilterGraph.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/UI/FormantFilterGraph.cpp b/src/UI/FormantFilterGraph.cpp
index 25d3a7b5..dfe44cf6 100644
--- a/src/UI/FormantFilterGraph.cpp
+++ b/src/UI/FormantFilterGraph.cpp
@@ -63,7 +63,7 @@ void FormantFilterGraph::OSC_value(int x, const char *loc)
 }
 void FormantFilterGraph::OSC_value(unsigned x, void *v)
 {
-    assert(x = sizeof(Pvowels));
+    assert(x == sizeof(Pvowels));
     memcpy(&Pvowels[0], v, x);
     redraw();
 }
-- 
2.47.0

