From f6f61a5b40c5d48ddc6da22159dbb6eac2d4ceb5 Mon Sep 17 00:00:00 2001
From: Johannes Lorenz <j.git@lorenz-ho.me>
Date: Tue, 15 Dec 2020 17:02:36 +0100
Subject: [PATCH 1677/1936] Remove multi-line comment

---
 src/Synth/OscilGen.cpp | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/src/Synth/OscilGen.cpp b/src/Synth/OscilGen.cpp
index 7ade5924..630acf18 100644
--- a/src/Synth/OscilGen.cpp
+++ b/src/Synth/OscilGen.cpp
@@ -200,9 +200,6 @@ const rtosc::Ports OscilGen::non_realtime_ports = {
             d.broadcast("/damage", "s", repath);
         }}};
 
-//unused:
-//#define rForwardCb [](const char *msg, rtosc::RtData &d) {\
-//    printf("forwarding...\n"); d.forward();}
 const rtosc::Ports OscilGen::realtime_ports{
     rSelf(OscilGen),
     rPresetType,
-- 
2.47.0

