From d36c18abd3dc4125057765c6f8b0a5cee3fe50bd Mon Sep 17 00:00:00 2001
From: Johannes Lorenz <j.git@lorenz-ho.me>
Date: Fri, 5 Feb 2021 06:28:51 +0100
Subject: [PATCH 1709/1936] [skip ci] Implement ZynAddSubFX::getDescription

---
 src/Plugin/ZynAddSubFX/ZynAddSubFX.cpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/Plugin/ZynAddSubFX/ZynAddSubFX.cpp b/src/Plugin/ZynAddSubFX/ZynAddSubFX.cpp
index 47862b0b..8615a38d 100644
--- a/src/Plugin/ZynAddSubFX/ZynAddSubFX.cpp
+++ b/src/Plugin/ZynAddSubFX/ZynAddSubFX.cpp
@@ -155,8 +155,9 @@ protected:
     */
     const char* getDescription() const noexcept override
     {
-        // TODO
-        return "";
+        return "Synthesizer featuring additive, subtractive, and Fourier "
+               "synthesis methods, a variety of modulators, powerful "
+               "oscillator editors, and a variety of built-in effects.";
     }
 
    /**
-- 
2.47.0

