From c7bc72f023d9643bbac8fae362b165a9bee0a9ba Mon Sep 17 00:00:00 2001
From: Friedolino <mkirchn@freenet.de>
Date: Fri, 26 Feb 2021 22:36:55 +0100
Subject: [PATCH 1714/1936] fix envstretch at forced release

---
 src/Synth/Envelope.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Synth/Envelope.cpp b/src/Synth/Envelope.cpp
index a06a2361..9ce524fe 100644
--- a/src/Synth/Envelope.cpp
+++ b/src/Synth/Envelope.cpp
@@ -170,7 +170,7 @@ float Envelope::envout(bool doWatch)
         else
             out = envoutval + (envval[releaseindex] - envoutval) * t; // linear interpolation envoutval and envval[releaseindex]
 
-        t += envdt[releaseindex] * envstretch;
+        t += envdt[releaseindex];
 
         if(t >= 1.0f) { // move to the next segment
             currentpoint = envsustain + 2;
-- 
2.47.0

