From 73babcb2fac771176b4178b7e515d387dfe7b7e1 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sun, 25 Apr 2021 16:40:06 -0400
Subject: [PATCH 1725/1936] Remove final stray cxxtest include

---
 src/Tests/SaveOSC.cpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/Tests/SaveOSC.cpp b/src/Tests/SaveOSC.cpp
index 6803b971..1d5898f8 100644
--- a/src/Tests/SaveOSC.cpp
+++ b/src/Tests/SaveOSC.cpp
@@ -7,8 +7,6 @@
 #include <rtosc/thread-link.h>
 #include <rtosc/rtosc-time.h>
 
-#include <cxxtest/TestSuite.h>
-
 #include "../Misc/Master.h"
 #include "../Misc/MiddleWare.h"
 #include "../UI/NSM.H"
-- 
2.47.0

