From 4a3f7d9a13b6de460c49f9eb5de0bde42e5a4767 Mon Sep 17 00:00:00 2001
From: Friedolino <mkirchn@freenet.de>
Date: Tue, 8 Jun 2021 22:19:28 +0200
Subject: [PATCH 1741/1936] add fake visualisation to for ladder filter

---
 src/Params/FilterParams.cpp | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/src/Params/FilterParams.cpp b/src/Params/FilterParams.cpp
index 3b3a0b09..41e61281 100644
--- a/src/Params/FilterParams.cpp
+++ b/src/Params/FilterParams.cpp
@@ -209,6 +209,19 @@ const rtosc::Ports FilterParams::ports = {
                         (float)obj->Pstages,
                         cf.b[0], cf.b[1], cf.b[2],
                         0.0,     -cf.a[1], -cf.a[2]);
+            } else if(obj->Pcategory == 3) {
+                int order = 0;
+                float gain = dB2rap(obj->getgain());
+                if(obj->Ptype != 6 && obj->Ptype != 7 && obj->Ptype != 8)
+                    gain = 1.0;
+                auto cf = AnalogFilter::computeCoeff(obj->Ptype,
+                        Filter::getrealfreq(obj->getfreq()),
+                        obj->getq(), obj->Pstages,
+                        gain, 48000, order);
+                d.reply(d.loc, "fffffff",
+                        (float)obj->Pstages,
+                        cf.c[0], cf.c[1], cf.c[2],
+                        0.0,     cf.d[1], cf.d[2]);
             }
         }},
     //    "", NULL, [](){}},"/freq"
-- 
2.47.0

