From 9507e7b4745af35312f980467031ff06633dd76b Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Mon, 21 Jun 2021 17:14:21 -0400
Subject: [PATCH 1746/1936] Clear old automation slots when reloading settings

---
 src/Misc/Master.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/Misc/Master.cpp b/src/Misc/Master.cpp
index 1eab6453..1ed50ab1 100644
--- a/src/Misc/Master.cpp
+++ b/src/Misc/Master.cpp
@@ -708,6 +708,10 @@ void Master::saveAutomation(XMLwrapper &xml, const rtosc::AutomationMgr &midi)
 
 void Master::loadAutomation(XMLwrapper &xml, rtosc::AutomationMgr &midi)
 {
+    //Clear out old data
+    for(int i=0; i<midi.nslots; ++i)
+        midi.clearSlot(i);
+
     if(xml.enterbranch("automation")) {
         for(int i=0; i<midi.nslots; ++i) {
             auto &slot = midi.slots[i];
-- 
2.47.0

