From d14b29a90445a524d236fb319dacb4c676848002 Mon Sep 17 00:00:00 2001
From: Friedolino <mkirchn@freenet.de>
Date: Tue, 22 Jun 2021 18:02:45 +0200
Subject: [PATCH 1749/1936] fix type mapping for fake bode plots

---
 src/Params/FilterParams.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Params/FilterParams.cpp b/src/Params/FilterParams.cpp
index 41e61281..0a17d2a7 100644
--- a/src/Params/FilterParams.cpp
+++ b/src/Params/FilterParams.cpp
@@ -214,7 +214,7 @@ const rtosc::Ports FilterParams::ports = {
                 float gain = dB2rap(obj->getgain());
                 if(obj->Ptype != 6 && obj->Ptype != 7 && obj->Ptype != 8)
                     gain = 1.0;
-                auto cf = AnalogFilter::computeCoeff(obj->Ptype,
+                auto cf = AnalogFilter::computeCoeff(4-obj->Ptype,
                         Filter::getrealfreq(obj->getfreq()),
                         obj->getq(), obj->Pstages,
                         gain, 48000, order);
-- 
2.47.0

