From 418c09247799cbf93590bea56bfd0dc03417e3e5 Mon Sep 17 00:00:00 2001
From: Friedolino <mkirchn@freenet.de>
Date: Wed, 23 Jun 2021 21:26:13 +0200
Subject: [PATCH 1752/1936] use shorter names for zest in LFO

---
 src/Params/LFOParams.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/Params/LFOParams.cpp b/src/Params/LFOParams.cpp
index 6ec9a77b..371bd446 100644
--- a/src/Params/LFOParams.cpp
+++ b/src/Params/LFOParams.cpp
@@ -80,11 +80,11 @@ static const rtosc::Ports _ports = {
               rLinear(0.0, 4.0), rDefaultDepends(loc), rDefault(0),
               rPreset(ad_voice_amp, 0.94),
               "Delay before LFO start\n0..4 second delay"),
-    rParamF(fadein, rShort("fadein"), rSpecial(disable), rUnit(S),
+    rParamF(fadein, rShort("f.in"), rSpecial(disable), rUnit(S),
               rLinear(0.0, 10.0f), rDefault(0.0f),
               "Time to ramp up LFO amplitude\n \
               0..10 seconds"),
-    rParamF(fadeout, rShort("fadeout"), rSpecial(disable), rUnit(S),
+    rParamF(fadeout, rShort("f.out"), rSpecial(disable), rUnit(S),
               rLinear(0.001, 10.0f), rDefault(10.0f),
               "Time to ramp down LFO amplitude on key release\n  \
               0..10 seconds, 10=off (default)"),
-- 
2.47.0

