From 3dfdeb8f954a2f3235ebc28cf49a1da398282538 Mon Sep 17 00:00:00 2001
From: Friedolino <mkirchn@freenet.de>
Date: Thu, 24 Jun 2021 17:29:00 +0200
Subject: [PATCH 1754/1936] try better Q mapping

---
 src/DSP/MoogFilter.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/DSP/MoogFilter.cpp b/src/DSP/MoogFilter.cpp
index b3828f1a..f7a271e0 100644
--- a/src/DSP/MoogFilter.cpp
+++ b/src/DSP/MoogFilter.cpp
@@ -136,7 +136,9 @@ void MoogFilter::setfreq(float ff)
 void MoogFilter::setq(float q)
 {
     // flattening the Q input
-    feedbackGain = cbrtf(q/1000.0f)*4.15f + 0.1f;
+    // self oscillation begins around 4.0
+    // mapped to match the ANALOG filter class
+    feedbackGain = cbrtf(q/1000.0f)*4.0f + 0.3f;
     // compensation factor for passband reduction by the negative feedback
     passbandCompensation = 1.0f + limit(feedbackGain, 0.0f, 1.0f);
 }
-- 
2.47.0

