From 24f490d2c05cf8e9c85061c7da5fc90c08581ddb Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Sat, 14 Aug 2021 21:54:56 -0400
Subject: [PATCH 1759/1936] Add Github Actions to PRs

---
 .github/workflows/ccpp.yml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/.github/workflows/ccpp.yml b/.github/workflows/ccpp.yml
index 44023605..700235fa 100644
--- a/.github/workflows/ccpp.yml
+++ b/.github/workflows/ccpp.yml
@@ -1,6 +1,6 @@
 name: Ubuntu Build CI
 
-on: [push]
+on: [push, pull_request]
 
 jobs:
   build:
-- 
2.47.0

