From cf05301b32c9dec9c439f3bccb416829617382a1 Mon Sep 17 00:00:00 2001
From: Johannes Lorenz <j.git@lorenz-ho.me>
Date: Sat, 8 Jan 2022 18:23:45 +0100
Subject: [PATCH 1807/1936] Fix whitespace

---
 src/Synth/OscilGen.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/Synth/OscilGen.cpp b/src/Synth/OscilGen.cpp
index eab543f5..df28af38 100644
--- a/src/Synth/OscilGen.cpp
+++ b/src/Synth/OscilGen.cpp
@@ -38,9 +38,9 @@ const rtosc::Ports OscilGen::non_realtime_ports = {
     rPaste,
     //TODO ensure min/max
     rOption(Phmagtype, rShort("scale"),
-    rOptions(linear,dB scale (-40),
-    dB scale (-60), dB scale (-80),
-    dB scale (-100)),
+            rOptions(linear,dB scale (-40),
+                     dB scale (-60), dB scale (-80),
+                     dB scale (-100)),
             rDefault(linear),
             "Type of magnitude for harmonics"),
     rOption(Pcurrentbasefunc, rShort("base"),
-- 
2.47.0

