From d535947bc23ef09fd27e7fa5cbcae9f828c6fb1f Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Thu, 10 Feb 2022 22:26:45 -0500
Subject: [PATCH 1824/1936] Update KitTest with automode defaults

---
 src/Tests/KitTest.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/Tests/KitTest.cpp b/src/Tests/KitTest.cpp
index f7c84f92..c24e7162 100644
--- a/src/Tests/KitTest.cpp
+++ b/src/Tests/KitTest.cpp
@@ -1137,6 +1137,7 @@ class KitTest
             auto &pool = part->notePool;
 
             part->ctl.setportamento(127);
+            part->ctl.portamento.automode = false;
 
             // The point here is to verify that even when a note is released,
             // its portamento offset is retained for subsequent notes to
@@ -1339,6 +1340,7 @@ class KitTest
 
             part->ctl.setportamento(127);
             part->ctl.portamento.pitchthreshtype = 0;
+            part->ctl.portamento.automode = 0;
 
             // Test that we get portamento when playing staccato notes
             // that are closer than the default threshold of 3, when the
-- 
2.47.0

