From 5b52108f1e950ef06e26ce11f7d69fa1020764ec Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Tue, 22 Feb 2022 21:25:43 -0500
Subject: [PATCH 1826/1936] Fix wavefile serialization

---
 src/Nio/WavEngine.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/Nio/WavEngine.cpp b/src/Nio/WavEngine.cpp
index 1410a756..316b695f 100644
--- a/src/Nio/WavEngine.cpp
+++ b/src/Nio/WavEngine.cpp
@@ -63,6 +63,7 @@ void WavEngine::Stop()
     work.post();
     pthread_join(*tmp, NULL);
     delete pThread;
+    destroyFile();
 }
 
 void WavEngine::push(Stereo<float *> smps, size_t len)
-- 
2.47.0

