From 6e25116ee8dded8d5ebd22fc5d8ef922fae11265 Mon Sep 17 00:00:00 2001
From: Johannes Lorenz <j.git@lorenz-ho.me>
Date: Sat, 19 Mar 2022 10:30:12 +0100
Subject: [PATCH 1834/1936] check-ports.rb: Use dummy MIDI engine

---
 src/Tests/check-ports.rb | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Tests/check-ports.rb b/src/Tests/check-ports.rb
index b39a1c0a..b1df8e11 100755
--- a/src/Tests/check-ports.rb
+++ b/src/Tests/check-ports.rb
@@ -9,7 +9,7 @@ my_path=File.dirname(__FILE__)
 # NOTE: If you add too much debug output to zyn, it will be blocked when writing into the pipe,
 #       leading to MiddleWare to not reply and to port-checker to fail.
 #       This script should be rewritten to keep reading (and printing) zyn's messages.
-Open3.popen3(my_path + "/../zynaddsubfx -O null --no-gui") do |stdin, stdout, stderr, wait_thr|
+Open3.popen3(my_path + "/../zynaddsubfx -O null -I null --no-gui") do |stdin, stdout, stderr, wait_thr|
   pid = wait_thr[:pid]
   while line=stderr.gets do 
     # print "line: " + line;
-- 
2.47.0

