From 3b7e7fdccecde504113d2c5bad2614b5950d1581 Mon Sep 17 00:00:00 2001
From: Plagiat Gang Of Bros <plagiat@LaMachineACashDesBros.plagiat.org>
Date: Mon, 31 Jan 2022 17:34:13 +0100
Subject: [PATCH 1847/1936] Use lo_send_message_from in sendToRemote method

---
 src/Misc/MiddleWare.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/Misc/MiddleWare.cpp b/src/Misc/MiddleWare.cpp
index f790462e..643da4eb 100644
--- a/src/Misc/MiddleWare.cpp
+++ b/src/Misc/MiddleWare.cpp
@@ -2236,8 +2236,8 @@ void MiddleWareImpl::sendToRemote(const char *rtmsg, std::string dest)
 
         //Send to known url
         lo_address addr = lo_address_new_from_url(dest.c_str());
-        if(addr)
-            lo_send_message(addr, rtmsg, msg);
+        if(addr && server)
+            lo_send_message_from(addr, server, rtmsg, msg);
         lo_address_free(addr);
         lo_message_free(msg);
     }
-- 
2.47.0

