From b8fcf7994cfbb36419d4d66d0ba2e1ca0185e040 Mon Sep 17 00:00:00 2001
From: Friedolino <mkirchn@freenet.de>
Date: Sun, 15 May 2022 11:56:41 +0200
Subject: [PATCH 1851/1936] fix of combfilter initialization

---
 src/DSP/CombFilter.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/DSP/CombFilter.cpp b/src/DSP/CombFilter.cpp
index 912adbda..5dd48a61 100644
--- a/src/DSP/CombFilter.cpp
+++ b/src/DSP/CombFilter.cpp
@@ -14,7 +14,7 @@ namespace zyn{
 
 CombFilter::CombFilter(Allocator *alloc, unsigned char Ftype, float Ffreq, float Fq,
     unsigned int srate, int bufsize)
-    :Filter(srate, bufsize), gain(1.0f), type(Ftype), memory(*alloc)
+    :Filter(srate, bufsize), gain(1.0f), q(Fq), type(Ftype), memory(*alloc)
 {
     //worst case: looking back from smps[0] at 25Hz using higher order interpolation
     mem_size = (int)ceilf((float)samplerate/25.0) + buffersize + 2; // 2178 at 48000Hz and 256Samples
@@ -23,7 +23,7 @@ CombFilter::CombFilter(Allocator *alloc, unsigned char Ftype, float Ffreq, float
     memset(input, 0, mem_size*sizeof(float));
     memset(output, 0, mem_size*sizeof(float));
 
-    setfreq_and_q(Ffreq, Fq);
+    setfreq_and_q(Ffreq, q);
     settype(type);
 }
 
-- 
2.47.0

