From 255d8442407842841712f4cae0966969129a76a9 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Tue, 13 Dec 2022 22:13:31 -0500
Subject: [PATCH 1873/1936] Don't consider notes from disabled kits valid

---
 src/Misc/Part.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Misc/Part.cpp b/src/Misc/Part.cpp
index 2638955a..a8342412 100644
--- a/src/Misc/Part.cpp
+++ b/src/Misc/Part.cpp
@@ -1506,7 +1506,7 @@ uint8_t Part::Kit::sendto(void) const
 
 bool Part::Kit::validNote(char note) const
 {
-    return !Pmuted && inRange((uint8_t)note, Pminkey, Pmaxkey);
+    return !Pmuted && Penabled && inRange((uint8_t)note, Pminkey, Pmaxkey);
 }
 
 }
-- 
2.47.0

