From 883293a34df81a6088c0b1c40855c413c9f3c394 Mon Sep 17 00:00:00 2001
From: Johannes Lorenz <j.git@lorenz-ho.me>
Date: Fri, 28 Apr 2023 21:09:46 +0200
Subject: [PATCH 1886/1936] Update rtosc: bugfixes

---
 rtosc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rtosc b/rtosc
index 5cae2dfd..54a523fe 160000
--- a/rtosc
+++ b/rtosc
@@ -1 +1 @@
-Subproject commit 5cae2dfdc00b9fe223c62dfebceb020f066baaee
+Subproject commit 54a523feb73847eba5d1a2304dbc53ea80a34af0
-- 
2.47.0

