From df1bc0dda96b1ebf8b014610cf740d112495e9c0 Mon Sep 17 00:00:00 2001
From: Philippe Gervais <philippe.n.gervais@free.fr>
Date: Sat, 10 Jun 2023 14:32:22 +0200
Subject: [PATCH 1912/1936] Added more doc files to .gitignore.

---
 .gitignore | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/.gitignore b/.gitignore
index a6cf454d..fd6cfb2f 100644
--- a/.gitignore
+++ b/.gitignore
@@ -16,4 +16,7 @@ CMakeFiles
 build
 doc/gen/
 doc/zynaddsubfx.html
+doc/zynaddsubfx.pdf
+doc/zynaddsubfx.1
+doc/zynaddsubfx.chunked
 .DS_Store
-- 
2.47.0

