From 1fbe7c6552486f9bef9e7b286540c95592fe2936 Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Wed, 21 Jun 2023 17:13:22 -0400
Subject: [PATCH 1919/1936] SaveOSC: Fix compiler warning on struct init

---
 src/Tests/SaveOSC.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Tests/SaveOSC.cpp b/src/Tests/SaveOSC.cpp
index 6bea0a0b..7190c9e2 100644
--- a/src/Tests/SaveOSC.cpp
+++ b/src/Tests/SaveOSC.cpp
@@ -475,7 +475,7 @@ class SaveOSCTest
         {
             git_libgit2_init();
 
-            git_buf root_path = {0};
+            git_buf root_path = {0, 0, 0};
             git_repository_discover(&root_path, ".", 0, NULL);
 
             git_repository *repo = NULL;
-- 
2.47.0

