From c75762a193db906bd181cd58867c97834a7721f9 Mon Sep 17 00:00:00 2001
From: Johannes Lorenz <j.git@lorenz-ho.me>
Date: Sat, 17 Jun 2023 22:46:13 +0200
Subject: [PATCH 1924/1936] Savefiles: Show diff in case of failure

---
 src/Misc/MiddleWare.cpp | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/src/Misc/MiddleWare.cpp b/src/Misc/MiddleWare.cpp
index 58763532..264f1a25 100644
--- a/src/Misc/MiddleWare.cpp
+++ b/src/Misc/MiddleWare.cpp
@@ -844,9 +844,12 @@ public:
                             std::cout << savefile << std::endl;
                             std::cerr << "Cannot write OSC savefile!! (see tmp1.txt and tmp2.txt)"
                                       << std::endl;
-                            std::ofstream tmp1("tmp1.txt"), tmp2("tmp2.txt");
-                            tmp1 << xml;
-                            tmp2 << xml2;
+                            {
+                                std::ofstream tmp1("tmp1.txt"), tmp2("tmp2.txt");
+                                tmp1 << xml;
+                                tmp2 << xml2;
+                            }
+                            system("diff tmp1.txt tmp2.txt");
                             res = -1;
                         }
 
-- 
2.47.0

