From d533aa456099bd9e03b1a89e1db023dc9fa6a70a Mon Sep 17 00:00:00 2001
From: fundamental <mark.d.mccurry@gmail.com>
Date: Fri, 4 Aug 2023 20:22:55 -0400
Subject: [PATCH 1927/1936] Add missing stdint.h include to FilterParams.h

---
 src/Params/FilterParams.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/Params/FilterParams.h b/src/Params/FilterParams.h
index 656c4182..201080a8 100644
--- a/src/Params/FilterParams.h
+++ b/src/Params/FilterParams.h
@@ -17,6 +17,7 @@
 #include "../globals.h"
 #include "../Misc/XMLwrapper.h"
 #include "PresetsArray.h"
+#include <stdint.h>
 
 namespace zyn {
 
-- 
2.47.0

