From 13a38b80fbf383d9b5baf323a8e6033510c23768 Mon Sep 17 00:00:00 2001
From: Friedolino <mkirchn@freenet.de>
Date: Sun, 3 Mar 2024 20:43:12 +0100
Subject: [PATCH 1932/1936] fix lfo rnd behaviour

---
 src/Synth/LFO.cpp | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/src/Synth/LFO.cpp b/src/Synth/LFO.cpp
index 19aed227..4a7d69f6 100644
--- a/src/Synth/LFO.cpp
+++ b/src/Synth/LFO.cpp
@@ -279,7 +279,7 @@ float LFO::lfoout()
         phase += phaseInc;
     else {
         const float tmp = (incrnd * (1.0f - phase) + nextincrnd * phase);
-        phase += phaseInc * limit(tmp, 0.0f, 1.0f);
+        phase += phaseInc * tmp;
     }
     if(phase >= 1) {
         phase    = fmod(phase, 1.0f);
@@ -309,7 +309,13 @@ void LFO::computeNextFreqRnd()
     if(deterministic)
         return;
     incrnd     = nextincrnd;
-    nextincrnd = powf(0.5f, lfofreqrnd) + RND * (powf(2.0f, lfofreqrnd) - 1.0f);
+    lfofreqrnd = powf(lfopars.Pfreqrand / 127.0f, 2.0f) * 4.0f;
+    // old implementation:
+    // nextincrnd = powf(0.5f, lfofreqrnd) + RND * (powf(2.0f, lfofreqrnd) - 1.0f);
+    // problem with that old implementation is that it changes the center frequency.
+    // the new one doesn't
+    const float rndValue = lfofreqrnd*(RND*2.0f - 1.0);
+    nextincrnd = powf(2.0f, rndValue);
 }
 
 }
-- 
2.47.0

